<div class="container">

<table style="width: 100%;"><tr>
<td>bbt_run_from_model</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run BEAST2</h2>

<h3>Description</h3>

<p>Do a full run: create a 'BEAST2' configuration file (like 'BEAUti 2'),
run 'BEAST2', parse results (like 'Tracer')
</p>


<h3>Usage</h3>

<pre><code class="language-R">bbt_run_from_model(
  fasta_filename,
  inference_model = beautier::create_inference_model(),
  beast2_options = beastier::create_beast2_options()
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model,
as returned by create_inference_model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beast2_options</code></td>
<td>
<p>'BEAST2'  options,
as can be created by create_beast2_options</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list with the following elements:<br></p>

<ul>
<li>
<p><code>estimates</code>: a data frame with 'BEAST2'
parameter estimates

</p>
</li>
<li>
<p><code>[alignment_id]_trees</code>: a <code>multiPhylo</code>
containing the phylogenies
in the 'BEAST2' posterior. <code>[alignment_id]</code>
is the ID of the alignment. For example,
when running <code>bbt_run_from_model</code> with
<code>anthus_aco.fas</code>, this element will have
name <code>anthus_aco_trees</code>

</p>
</li>
<li>
<p><code>operators</code>: a data frame with the
'BEAST2' MCMC operator acceptances

</p>
</li>
<li>
<p><code>output</code>: a numeric vector with the output
sent to standard output and error streams

</p>
</li>
<li>
<p><code>ns</code>: (optional) the results of a marginal likelihood estimation,
will exist only when <code>create_ns_mcmc</code> was
used for <code>mcmc</code>.
This structure will contain the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li>
</ul>
</li>
</ul>
<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code>remove_burn_ins</code>
to remove the burn-ins from
the posterior's estimates (<code>posterior$estimates</code>)
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Simple short inference
  inference_model &lt;- create_test_inference_model()

  # Default BEAST2 options
  beast2_options &lt;- create_beast2_options()

  bbt_run_from_model(
    fasta_filename = get_babette_path("anthus_aco.fas"),
    inference_model = inference_model,
    beast2_options = beast2_options
  )

  # Cleanup
  bbt_delete_temp_files(
    inference_model = inference_model,
    beast2_options = beast2_options
  )
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>


</div>