<div class="container">

<table style="width: 100%;"><tr>
<td>bbt_run</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Run BEAST2</h2>

<h3>Description</h3>

<p>Do a full BEAST2 run: create a 'BEAST2' configuration file (like 'BEAUti 2'),
run 'BEAST2', parse results (like 'Tracer')
</p>


<h3>Usage</h3>

<pre><code class="language-R">bbt_run(
  fasta_filename,
  tipdates_filename = NA,
  site_model = beautier::create_jc69_site_model(),
  clock_model = beautier::create_strict_clock_model(),
  tree_prior = beautier::create_yule_tree_prior(),
  mrca_prior = NA,
  mcmc = beautier::create_mcmc(),
  beast2_input_filename = beastier::create_temp_input_filename(),
  rng_seed = 1,
  beast2_output_state_filename = beastier::create_temp_state_filename(),
  beast2_path = beastier::get_default_beast2_path(),
  overwrite = TRUE,
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fasta_filename</code></td>
<td>
<p>a FASTA filename</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tipdates_filename</code></td>
<td>
<p>name of the file containing tip dates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>site_model</code></td>
<td>
<p>one site model,
see create_site_models</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_model</code></td>
<td>
<p>one clock model,
see create_clock_model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_prior</code></td>
<td>
<p>one tree priors,
as created by create_tree_prior</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrca_prior</code></td>
<td>
<p>one Most Recent Common Ancestor prior,
as returned by <code>create_mrca_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mcmc</code></td>
<td>
<p>the MCMC options,
see create_mcmc</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beast2_input_filename</code></td>
<td>
<p>path of the 'BEAST2'  configuration file.
By default, this file is put in a temporary folder with a random
filename, as the user needs not read it: it is used as input of 'BEAST2'.
Specifying a <code>beast2_input_filename</code> allows to store that file
in a more permanently stored location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rng_seed</code></td>
<td>
<p>the random number generator seed. Must be either
<code>NA</code> or a positive non-zero value. An RNG seed of <code>NA</code>
results in 'BEAST2'  picking a random seed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beast2_output_state_filename</code></td>
<td>
<p>name of the final state file created
by 'BEAST2', containing the operator acceptances. By default, this
file is put a temporary folder with a random
filename, as the user needs not read it: its content
is parsed and returned by this function.
Specifying a <code>beast2_output_state_filename</code> allows to store that file
in a more permanently stored location.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beast2_path</code></td>
<td>
<p>name of either a 'BEAST2'  binary file
(usually simply <code>beast</code>)
or a 'BEAST2'  jar file
(usually has a <code>.jar</code> extension).
Use <code>get_default_beast2_bin_path</code> to get
the default BEAST binary file's path
Use <code>get_default_beast2_jar_path</code> to get
the default BEAST jar file's path</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overwrite</code></td>
<td>
<p>will 'BEAST2'  overwrite files? Like 'BEAST2',
this is set to TRUE by default.
If TRUE, 'BEAST2'  will overwrite the
<code>beast2_options$output_state_filename</code> if its present.
If FALSE, 'BEAST2'  will not overwrite the
<code>beast2_options$output_state_filename</code> if its present
and babette will give an error message.
Note that if <code>overwrite</code> is set to FALSE when
a <code>tracelog</code> (see create_tracelog),
<code>screenlog</code> (see create_screenlog)
or <code>treelog</code> (see create_treelog)
file already exists,
'BEAST2'  (and thus babette) will freeze.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>set to TRUE for more output</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Prefer using <code>bbt_run_from_model</code>, as it has a cleaner interface.
</p>


<h3>Value</h3>

<p>a list with the following elements:<br></p>

<ul>
<li>
<p><code>estimates</code>: a data frame with 'BEAST2'
parameter estimates

</p>
</li>
<li>
<p><code>[alignment_id]_trees</code>: a <code>multiPhylo</code>
containing the phylogenies
in the 'BEAST2' posterior. <code>[alignment_id]</code>
is the ID of the alignment. For example,
when running <code>bbt_run</code> with
<code>anthus_aco.fas</code>, this element will have
name <code>anthus_aco_trees</code>

</p>
</li>
<li>
<p><code>operators</code>: a data frame with the
'BEAST2' MCMC operator acceptances

</p>
</li>
<li>
<p><code>output</code>: a numeric vector with the output
sent to standard output and error streams

</p>
</li>
<li>
<p><code>ns</code>: (optional) the results of a marginal likelihood estimation,
will exist only when <code>create_ns_mcmc</code> was
used for the MCMC.
This structure will contain the following elements:
</p>

<ul>
<li> <p><code>marg_log_lik</code> the marginal log likelihood estimate
</p>
</li>
<li> <p><code>marg_log_lik_sd</code> the standard deviation around the estimate
</p>
</li>
<li> <p><code>estimates</code> the parameter estimates
created during the marginal likelihood estimation
</p>
</li>
<li> <p><code>trees</code> the trees
created during the marginal likelihood estimation
</p>
</li>
</ul>
</li>
</ul>
<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


<h3>See Also</h3>

<p>Use <code>remove_burn_ins</code>
to remove the burn-ins from
the posterior's estimates (<code>posterior$estimates</code>)
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (beautier::is_on_ci() &amp;&amp; is_beast2_installed()) {
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()

  # Setup for a short run
  mcmc &lt;- create_test_mcmc()

  # Store filenames for cleanup.
  # Note that 'bbt_run_from_model allows for easier cleanup
  mcmc$tracelog$filename &lt;- tempfile()
  mcmc$treelog$filename &lt;- tempfile()
  mcmc$screenlog$filename &lt;- tempfile()
  beast2_input_filename &lt;- tempfile()
  beast2_output_state_filename &lt;- tempfile()

  bbt_run(
    fasta_filename = get_babette_path("anthus_aco.fas"),
    beast2_input_filename = beast2_input_filename,
    beast2_output_state_filename = beast2_output_state_filename,
    mcmc = mcmc
  )

  # Cleanup
  # Again, note that 'bbt_run_from_model allows for easier cleanup
  file.remove(mcmc$tracelog$filename)
  file.remove(mcmc$treelog$filename)
  file.remove(mcmc$screenlog$filename)
  file.remove(beast2_input_filename)
  file.remove(beast2_output_state_filename)
  beastier::remove_beaustier_folders()
  beastier::check_empty_beaustier_folders()
}
</code></pre>


</div>