<div class="container">

<table style="width: 100%;"><tr>
<td>bd.gwas.test</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fast K-sample Ball Divergence Test for GWAS Data</h2>

<h3>Description</h3>

<p>Fast K-sample Ball Divergence Test for GWAS Data
</p>


<h3>Usage</h3>

<pre><code class="language-R">bd.gwas.test(
  x,
  snp,
  screening.method = c("permute", "spectrum"),
  refine = TRUE,
  num.permutations,
  distance = FALSE,
  alpha,
  screening.result = NULL,
  verbose = TRUE,
  seed = 1,
  num.threads = 0,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a numeric vector, matrix, data.frame, dist object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snp</code></td>
<td>
<p>a numeric matrix recording the values of single nucleotide polymorphism (SNP). Each column must be an integer vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>screening.method</code></td>
<td>
<p>if <code>screening.method = "spectrum"</code>, the spectrum method is applied to
screening the candidate SNPs, or otherwise, the permutation method is applied. Default: <code>screening.method = "permute"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refine</code></td>
<td>
<p>a logical value. If <code>refine = TRUE</code>, a <code class="reqn">p</code>-values refining process is applied to
the SNPs which passes the pre-screening process. Default: <code>refine = TRUE</code> (At present, <code>refine = FALSE</code> is not available).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.permutations</code></td>
<td>
<p>the number of permutation replications. When <code>num.permutations = 0</code>,
the function just returns the Ball Divergence statistic. Default: <code>num.permutations = 100 * ncol(snp)</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distance</code></td>
<td>
<p>if <code>distance = TRUE</code>, the elements of <code>x</code> will be considered as a distance matrix. Default: <code>distance = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>the significance level. Default: <code>0.05 / ncol(snp)</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>screening.result</code></td>
<td>
<p>A object return by <code>bd.gwas.test</code> that
preserving the pre-screening result.
It works only if the pre-screening is available.
Default: <code>screening.result = NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Show computation status and estimated runtimes. Default: <code>verbose = FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>the random seed. Default <code>seed = 1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.threads</code></td>
<td>
<p>number of threads. If <code>num.threads = 0</code>, then all of available cores will be used. Default <code>num.threads = 0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>further arguments to be passed to or from methods.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>bd.gwas.test returns a list containing the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>
<p>ball divergence statistics vector.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>permuted.statistic</code></td>
<td>
<p>a data.frame containing permuted ball divergence statistic for pre-screening SNPs.
If <code>refine = FALSE</code>, it takes value <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eigenvalue</code></td>
<td>
<p>the eigenvalue of spectrum decomposition. If <code>refine = TRUE</code>, it takes value <code>NULL</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p.value</code></td>
<td>
<p>the p-values of ball divergence test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refined.snp</code></td>
<td>
<p>the SNPs have been refined.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refined.p.value</code></td>
<td>
<p>the refined <code class="reqn">p</code>-value of significant snp.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refined.permuted.statistic</code></td>
<td>
<p>a data.frame containing permuted ball divergence statistics for refining <code class="reqn">p</code>-values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>screening.result</code></td>
<td>
<p>a list containing the result of screening.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Jin Zhu
</p>


<h3>References</h3>

<p>Yue Hu, Haizhu Tan, Cai Li, and Heping Zhang. (2021). Identifying genetic risk variants associated with brain volumetric phenotypes via K-sample Ball Divergence method. Genetic Epidemiology, 1â€“11. https://doi.org/10.1002/gepi.22423
</p>


<h3>See Also</h3>

<p><code>bd</code>, <code>bd.test</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
library(Ball)
set.seed(1234)
num &lt;- 200
snp_num &lt;- 500
p &lt;- 5
x &lt;- matrix(rnorm(num * p), nrow = num)
snp &lt;- sapply(1:snp_num, function(i) {
  sample(0:2, size = num, replace = TRUE)
})
snp1 &lt;- sapply(1:snp_num, function(i) {
  sample(1:2, size = num, replace = TRUE)
})
snp &lt;- cbind(snp, snp1)
res &lt;- Ball::bd.gwas.test(x = x, snp = snp)
mean(res[["p.value"]] &lt; 0.05)
mean(res[["p.value"]] &lt; 0.005)

## only return the test statistics;
res &lt;- Ball::bd.gwas.test(x = x, snp = snp, num.permutation = 0)

## save pre-screening process results:
x &lt;- matrix(rnorm(num * p), nrow = num)
snp &lt;- sapply(1:snp_num, function(i) {
  sample(0:2, size = num, replace = TRUE, prob = c(1/2, 1/4, 1/4))
})
snp_screening &lt;- Ball::bd.gwas.test(x = x, snp = snp,
                                    alpha = 5*10^-4, 
                                    num.permutations = 19999)
mean(res[["p.value"]] &lt; 0.05)
mean(res[["p.value"]] &lt; 0.005)
mean(res[["p.value"]] &lt; 0.0005)
## refine p-value according to the pre-screening process result:
res &lt;- Ball::bd.gwas.test(x = x, snp = snp, alpha = 5*10^-4,
                          num.permutations = 19999,
                          screening.result = snp_screening[["screening.result"]])

</code></pre>


</div>