<div class="container">

<table style="width: 100%;"><tr>
<td>diffexprm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Measure for performance of differential expression analysis (after batch effect adjustment)
</h2>

<h3>Description</h3>

<p>This metric is similar to the idea presented in Lazar et al (2012) which consists 
in comparing the list of the most differentially expressed genes obtained using a batch 
effect adjusted dataset to the list obtained using an independent dataset. For each 
batch the following is done by <code>diffexprm</code>: 1) the respective batch is left out
and batch effect adjustment 
is performed using the remaining batches; 2) differential expression analysis is performed 
once using the left-out batch and once using the remaining batch-effect adjusted data; 
3) the overlap between the two lists of genes found differentially expressed in the 
two subsets is measured. See below for further details.
</p>


<h3>Usage</h3>

<pre><code class="language-R">diffexprm(x, batch, y, method = c("fabatch", "combat", "sva", 
  "meancenter", "standardize", "ratioa", "ratiog", "none"))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>matrix. The covariate matrix. Observations in rows, variables in columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>factor. Binary target variable. Has to have two factor levels, where each of them correponds to one of the two classes of the target variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>character. Method for batch effect adjustment. The following are supported: <code>fabatch</code>, <code>combat</code>, <code>fsva</code>, <code>meancenter</code>, <code>standardize</code>, <code>ratioa</code>, <code>ratiog</code> and <code>none</code>
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The following procedure is performed: 1) For each batch j leave this batch out and 
perform batch effect adjustment on the rest of the dataset. Derive two lists of the 5 percent 
of variables which are most differentially expressed (see next paragraph): one using the batch 
effect adjusted dataset - where batch j was left out - and one using the data from batch j. 
Calculate the number of variables appearing in both lists and divide this number by the common 
length of the lists. 2) Calculate a weighted average of the values obtained in 1) with weights 
proportional to the number of observations in the corresponding left-out batches. 
</p>
<p>Differential expression is measured as follows. For each variable a randomized p-value out 
of the Whitney-Wilcoxon rank sum test is drawn, see Geyer and Meeden (2005) for details. 
Then those 5 percent variables are considered differentially expressed, which are associated 
with the smallest p-values.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The larger the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>
<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>
<p>Geyer, C. J., Meeden, G., D. (2005). Fuzzy and randomized confidence intervals and p-values (with discussion). Statistical Science 20(4):358-387, &lt;doi: <a href="https://doi.org/10.1214/088342305000000340">10.1214/088342305000000340</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



diffexprm(x=Xsub, batch=batchsub, y=ysub, method = "ratiog")
diffexprm(x=Xsub, batch=batchsub, y=ysub, method = "none")
</code></pre>


</div>