<div class="container">

<table style="width: 100%;"><tr>
<td>svabaaddon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Addon batch effect adjustment using frozen SVA
</h2>

<h3>Description</h3>

<p>Performs addon batch effect adjustment using frozen SVA. Takes the output of performing <code>svaba</code> on a training data set and new batch data and correspondingly adjusts the test data to better match the adjusted training data according to the SVA model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">svabaaddon(params, x)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>params</code></td>
<td>

<p>object of class <code>svatrain</code>. Contains parameters necessary for addon batch effect adjustment with frozen SVA.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>It is <b>not recommended</b> to perform frozen SVA in cross-study prediction settings, because it assumes similarity between training and test set and has been observed to (strongly) impair prediction performance in cases where this assumption is not given. Given a not too small test set, the following methods are recommended (Hornung et al., 2016): <code>combatba</code>, <code>meancenter</code>, <code>ratioa</code>, <code>ratiog</code>.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Leek, J. T., Storey, J. D. (2007). Capturing Heterogeneity in Gene Expression Studies by Surrogate Variable Analysis. PLoS Genetics 3:1724-1735, &lt;doi: <a href="https://doi.org/10.1371/journal.pgen.0030161">10.1371/journal.pgen.0030161</a>&gt;.
</p>
<p>Parker, H. S., Bravo, H. C., Leek, J. T. (2014). Removing batch effects for prediction problems with frozen surrogate variable analysis. PeerJ 2:e561, &lt;doi: <a href="https://doi.org/10.7717/peerj.561">10.7717/peerj.561</a>&gt;.
</p>
<p>Hornung, R., Causeur, D., Bernau, C., Boulesteix, A.-L. (2017). Improving cross-study prediction through addon batch effect adjustment and addon normalization. Bioinformatics 33(3):397â€“404, &lt;doi: <a href="https://doi.org/10.1093/bioinformatics/btw650">10.1093/bioinformatics/btw650</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(autism)

# Random subset of 150 variables:
set.seed(1234)
Xsub &lt;- X[,sample(1:ncol(X), size=150)]

# In cases of batches with more than 20 observations
# select 20 observations at random:
subinds &lt;- unlist(sapply(1:length(levels(batch)), function(x) {
  indbatch &lt;- which(batch==x)
  if(length(indbatch) &gt; 20)
    indbatch &lt;- sort(sample(indbatch, size=20))
  indbatch
}))
Xsub &lt;- Xsub[subinds,]
batchsub &lt;- batch[subinds]
ysub &lt;- y[subinds]



trainind &lt;- which(batchsub %in% c(1,2))

Xsubtrain &lt;- Xsub[trainind,]
ysubtrain &lt;- ysub[trainind]
batchsubtrain &lt;- factor(as.numeric(batchsub[trainind]), levels=c(1,2))


testind &lt;- which(batchsub %in% c(3,4))

Xsubtest &lt;- Xsub[testind,]
ysubtest &lt;- ysub[testind]

batchsubtest &lt;- as.numeric(batchsub[testind])
batchsubtest[batchsubtest==3] &lt;- 1
batchsubtest[batchsubtest==4] &lt;- 2
batchsubtest &lt;- factor(batchsubtest, levels=c(1,2))



params &lt;- svaba(x=Xsubtrain, y=ysubtrain, batch=batchsubtrain)

Xsubtestaddon &lt;- svabaaddon(params, x=Xsubtest)
</code></pre>


</div>