<div class="container">

<table style="width: 100%;"><tr>
<td>kldist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Kullback-Leibler divergence between density of within and between batch pairwise distances
</h2>

<h3>Description</h3>

<p>This metric estimates the Kullback-Leibler divergences between the distributions 
of the within and that of the between batch euclidian distances of pairs of 
observations. These distributions should be similar in the abscence of stronger 
batch effects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">kldist(xba, batch)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) the distances between all pairs of observations in batch j - denoted as {dist_j} - and 
the distances between all such pairs in batch j* - denoted as {dist_j*} - are calculated; 
2) for each observation in j the distances to all observations in j* are calculated, 
resulting in n_j x n_j* distances denoted as {dist_jj*}; calculate the 
Kullback-Leibler divergence between the densities of {dist_j} and {dist_jj*} and that 
between the densities of {dist_j*} and {dist_jj*} - using the k-nearest neighbours based 
method by Boltz et al (2009) with k=5; 3) take the weighted mean of the values of 
these two divergences with weights proportional to n_j and n_j*. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches. 
</p>
<p>The variables are standardized before the calculation to make the metric independent of scale.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Lee, J. A., Dobbin, K. K., Ahn, J. (2014). Covariance adjustment for batch effect in gene expression data. Statistics in Medicine 33:2681-2695, &lt;doi: <a href="https://doi.org/10.1002/sim.6157">10.1002/sim.6157</a>&gt;.
</p>
<p>Boltz, S., Debreuve, E., Barlaud, M. (2009). High-dimensional statistical measure for region-of-interest tracking. Transactions in Image Processing 18(6):1266-1283, &lt;doi: <a href="https://doi.org/10.1109/TIP.2009.2015158">10.1109/TIP.2009.2015158</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(autism)

kldist(xba=X, batch=batch)
</code></pre>


</div>