<div class="container">

<table style="width: 100%;"><tr>
<td>ratioaaddon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Addon batch effect adjustment for Ratio-A
</h2>

<h3>Description</h3>

<p>Performs addon batch effect adjustment for Ratio-A: 
1) takes the output of <code>ratioa</code> applied to a training 
data set together with new batch data; 
2) checks whether the training data was also adjusted using Ratio-A and whether the same number of variables is present in training and new data;
3) performs Ratio-A on the new batch data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">ratioaaddon(params, x, batch)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>params</code></td>
<td>

<p>object of class <code>ratioa</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>

<p>matrix. The covariate matrix of the new data. Observations in rows, variables in columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch</code></td>
<td>

<p>factor. Batch variable of the new data. Currently has to have levels: '1', '2', '3' and so on.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The adjusted covariate matrix of the test data.
</p>


<h3>Note</h3>

<p>Because Ratio-A is performed "batch by batch" the "addon procedure" for Ratio-A consists of plain Ratio-A on the new test batches.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Luo, J., Schumacher, M., Scherer, A., Sanoudou, D., Megherbi, D., Davison, T., Shi, T., Tong, W., Shi, L., Hong, H., Zhao, C., Elloumi, F., Shi, W., Thomas, R., Lin, S., Tillinghast, G., Liu, G., Zhou, Y., Herman, D., Li, Y., Deng, Y., Fang, H., Bushel, P., Woods, M., Zhang, J. (2010). A comparison of batch effect removal methods for enhancement of prediction performance using maqc-ii microarray gene expression data. The Pharmacogenomics Journal 10:278-291, &lt;doi: <a href="https://doi.org/10.1038/tpj.2010.57">10.1038/tpj.2010.57</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(autism)

trainind &lt;- which(batch %in% c(1,2))

Xtrain &lt;- X[trainind,]
ytrain &lt;- y[trainind]
batchtrain &lt;- factor(as.numeric(batch[trainind]), levels=c(1,2))


testind &lt;- which(batch %in% c(3,4))

Xtest &lt;- X[testind,]
ytest &lt;- y[testind]

batchtest &lt;- as.numeric(batch[testind])
batchtest[batchtest==3] &lt;- 1
batchtest[batchtest==4] &lt;- 2
batchtest &lt;- factor(batchtest, levels=c(1,2))


params &lt;- ratioa(x=Xtrain, batch=batchtrain)

Xtestaddon &lt;- ratioaaddon(params=params, x=Xtest, batch=batchtest)
</code></pre>


</div>