<div class="container">

<table style="width: 100%;"><tr>
<td>avedist</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Average minimal distance between batches
</h2>

<h3>Description</h3>

<p>This metric is concerned with the minimal distances between pairs of batches.
</p>


<h3>Usage</h3>

<pre><code class="language-R">avedist(xba, batch)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xba</code></td>
<td>

<p>matrix. The covariate matrix, raw or after batch effect adjustment. observations in rows, variables in columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>batch</code></td>
<td>

<p>factor. Batch variable. Each factor level (or 'category') corresponds to one of the batches. For example, if there are four batches, this variable would have four factor levels and observations with the same factor level would belong to the same batch.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For two batches j and j* (see next paragraph for the case with more batches): 1) for each observation in batch j the euclidean 
distance to the nearest observation in batch j* is calculated; 2) the roles 
of j and j* are switched and 1) is re-performed; 3) the average is taken over all n_j + n_j* minimal distances. 
</p>
<p>For more than two batches: 1) for all possible pairs of batches: calculate the metric as described above; 2) calculate
the weighted average of the values in 1) with weights proportional to the sum of the sample sizes in the two respective batches. 
</p>
<p>The variables are standardized before the calculation to make the metric independent of scale.
</p>


<h3>Value</h3>

<p>Value of the metric
</p>


<h3>Note</h3>

<p>The smaller the values of this metric, the better.
</p>


<h3>Author(s)</h3>

<p>Roman Hornung
</p>


<h3>References</h3>

<p>Lazar, C., Meganck, S., Taminau, J., Steenhoff, D., Coletta, A., Molter,C., Weiss-Solís, D. Y., Duque, R., Bersini, H., Nowé, A. (2012). Batch effect removal methods for microarray gene expression data integration: a survey. Briefings in Bioinformatics 14(4):469-490, &lt;doi: <a href="https://doi.org/10.1093/bib/bbs037">10.1093/bib/bbs037</a>&gt;.
</p>
<p>Hornung, R., Boulesteix, A.-L., Causeur, D. (2016). Combining location-and-scale batch effect adjustment with data cleaning by latent factor adjustment. BMC Bioinformatics 17:27, &lt;doi: <a href="https://doi.org/10.1186/s12859-015-0870-z">10.1186/s12859-015-0870-z</a>&gt;.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(autism)

avedist(xba=X, batch=batch)
</code></pre>


</div>