<div class="container">

<table style="width: 100%;"><tr>
<td>summary.bayesCureModel</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Summary method.
</h2>

<h3>Description</h3>

<p>This function produces all summaries after fitting a cure rate model. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'bayesCureModel'
summary(object, burn = NULL, gamma_mix = TRUE, 
	K_gamma = 3,  K_max = 3, fdr = 0.1, covariate_levels = NULL, 
	yRange = NULL, alpha = 0.1, quantiles = c(0.05, 0.5, 0.95), 
	verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>

<p>An object of class <code>bayesCureModel</code>
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>burn</code></td>
<td>

<p>Positive integer corresponding to the number of mcmc iterations to discard as burn-in period
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma_mix</code></td>
<td>

<p>Boolean. If TRUE, the density of the marginal posterior distribution of the <code class="reqn">\gamma</code> parameter is estimated from the sampled MCMC values by fitting a normal mixture  model.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K_gamma</code></td>
<td>

<p>Used only when <code>gamma_mix = TRUE</code> and corresponds to the number of normal mixture components used to estimate the marginal posterior density of the <code class="reqn">\gamma</code> parameter. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K_max</code></td>
<td>

<p>Maximum number of components in order to cluster the (univariate) values of the joint posterior distribution across the MCMC run. Used to identify the main mode of the posterior distribution. See details. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fdr</code></td>
<td>

<p>The target value for controlling the False Discovery Rate when classifying subjects as cured or not. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>covariate_levels</code></td>
<td>

<p>Optional <code>data.frame</code> with new data for the covariates. It is only required when the user wishes to obtain a vector with the estimated posterior cured probabilities for a given combination of covariates. The column names should be exactly the same with the ones used in the input data. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yRange</code></td>
<td>

<p>Optional range (a vector of two non-negative values) for computing the sequence of posterior probabilities for the given values in <code>covariate_levels</code>. 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>

<p>Scalar between 0 and 1 corresponding to 1 - confidencel level for computing Highest Density Intervals. If set to NULL, the confidence intervals are not computed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quantiles</code></td>
<td>

<p>A vector of quantiles to evaluate for each variable.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Boolean: if TRUE the function prints the summary.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>ignored.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The values of the posterior draws are clustered according to a (univariate) normal mixture model, and the main mode corresponds to the cluster with the largest mean. The maximum number of mixture components corresponds to the <code>K_max</code> argument. The <span class="pkg">mclust</span> library is used for this purpose. The inference for the latent cure status of each (censored) observation is based on the MCMC draws corresponding to the main mode of the posterior distribution. The FDR is controlled according to the technique proposed in Papastamoulis and Rattray (2018). 
</p>
<p>In case where <code>covariate_levels</code> is set to <code>TRUE</code>, the <code>summary</code> function also returns a list named <code>p_cured_output</code> with the following entries
</p>

<dl>
<dt>mcmc</dt>
<dd>
<p>It is returned only in the case where the argument <code>covariate_values</code> is not <code>NULL</code>. A vector of posterior cured probabilities for the given values in <code>covariate_values</code>, per retained MCMC draw.</p>
</dd>
<dt>map</dt>
<dd>
<p>It is returned only in the case where the argument <code>covariate_values</code> is not <code>NULL</code>. The cured probabilities computed at the MAP estimate of the parameters, for the given values <code>covariate_values</code>.</p>
</dd>
<dt>tau_values</dt>
<dd>
<p>tau values</p>
</dd>
<dt>covariate_levels</dt>
<dd>
<p>covariate levels</p>
</dd>
<dt>index_of_main_mode</dt>
<dd>
<p>the subset of MCMC draws allocated to the main mode of the posterior distribution.</p>
</dd>
</dl>
<h3>Value</h3>

<p>A list with the following entries
</p>
<table>
<tr style="vertical-align: top;">
<td><code>map_estimate</code></td>
<td>
<p>Maximum A Posteriori (MAP) estimate of the parameters of the model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>highest_density_intervals</code></td>
<td>
<p>Highest Density Interval per parameter</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>latent_cured_status</code></td>
<td>
<p>Estimated posterior probabilities of the latent cure status per censored subject.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cured_at_given_FDR</code></td>
<td>
<p>Classification as cured or not, at given FDR level.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p_cured_output</code></td>
<td>
<p>It is returned only in the case where the argument <code>covariate_values</code> is not <code>NULL</code>. See details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main_mode_index</code></td>
<td>
<p>The retained MCMC iterations which correspond to the main mode of the posterior distribution.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Panagiotis Papastamoulis
</p>


<h3>References</h3>

<p>Papastamoulis and Milienos (2024). Bayesian inference and cure rate modeling for event history data. TEST doi: 10.1007/s11749-024-00942-w.
</p>
<p>Papastamoulis and Rattray (2018). A Bayesian Model Selection Approach for Identifying Differentially Expressed Transcripts from RNA Sequencing Data, Journal of the Royal Statistical Society Series C: Applied Statistics, Volume 67, Issue 1.
</p>
<p>Scrucca L, Fraley C, Murphy TB, Raftery AE (2023). Model-Based Clustering, Classification, and Density Estimation Using mclust in R. Chapman and Hall/CRC. ISBN 978-1032234953
</p>


<h3>See Also</h3>

<p><code>cure_rate_MC3</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R"># simulate toy data just for cran-check purposes        
	set.seed(10)
        n = 4
        # censoring indicators
        stat = rbinom(n, size = 1, prob = 0.5)
        # covariates
        x &lt;- matrix(rnorm(2*n), n, 2)
        # observed response variable 
        y &lt;- rexp(n)
#	define a data frame with the response and the covariates        
        my_data_frame &lt;- data.frame(y, stat, x1 = x[,1], x2 = x[,2])
# run a weibull model with default prior setup
# considering 2 heated chains 
	fit1 &lt;- cure_rate_MC3(survival::Surv(y, stat) ~ x1 + x2, 
		data = my_data_frame, 
		promotion_time = list(distribution = 'exponential'),
		nChains = 2, 
		nCores = 1, 
		mcmc_cycles = 3, sweep=2)
	mySummary &lt;- summary(fit1, burn = 0)

</code></pre>


</div>