<div class="container">

<table style="width: 100%;"><tr>
<td>liver</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Mus Musculus HDL QTL data from Leduc et. al. (2012)</h2>

<h3>Description</h3>

<p>Liver QTL data was obtained from a F2 inner-cross between inbred MRL/MpJ and SM/J 
strains of mice.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(liver)
</code></pre>


<h3>Format</h3>

<p>The data set <code>liver</code> contains three objects: the data, a learned Bayesian network structure 
and <code>vector</code> specifying node type. The fields are described as follows:  
</p>

<dl>
<dt><code>data</code></dt>
<dd>
<p>a <code>data.frame</code> object that contains 280 samples (rows) and 15 variables: genotype data 
(genotype states at 5 SNP markers) and phenotype data (HDL levels and normalized expression values of 10 genes).  
Three of these phenotypes are dichotomized, including Cyp2b10, Spgl1 and HDL.  Genotypes and dichotomized phenotypes 
are of class <code>factor</code> and continuous phenotypes are of class <code>numeric</code>.</p>
</dd>
<dt><code>dag</code></dt>
<dd>
<p>a <code>graphNEL</code> object, which is the network structure learned by <code>qtlnet</code> package.</p>
</dd>
<dt><code>node.class</code></dt>
<dd>
<p>a named <code>vector</code> of <code>logical</code> values indicating whether each node is discrete.</p>
</dd>
</dl>
<h3>References</h3>

<p>Leduc MS, Blair RH, Verdugo RA, Tsaih SW, Walsh K, Churchill GA, Paigen B.(2012). 
"Using bioinformatics and systems genetics to dissect HDL-cholesterol genetics in an MRL/MpJ 
x SM/J intercross." J Lipid Res., 6, 1163-75.
</p>


</div>