<div class="container">

<table style="width: 100%;"><tr>
<td>exploreBic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The exploreBic function</h2>

<h3>Description</h3>

<p>Provides exploratory plots for biclustered and clustered data.</p>


<h3>Usage</h3>

<pre><code class="language-R">exploreBic(dset, bres, gby ="genes", pfor ="mean", mname ="biclust", bnum =1, 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dset</code></td>
<td>
<p>data matrix.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bres</code></td>
<td>
<p>bicluster result.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gby</code></td>
<td>
<p>dimension to plot; 'genes' or 'conditions'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pfor</code></td>
<td>
<p>plot for 'mean', 'median', 'variance', 'mad', 'all', or 'quant' (quantile).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bnum</code></td>
<td>
<p>existing biclusters; '1','2'...</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The exploreBic function is mainly used for exploratory data analysis. It provides summary plots for mean, median, variance, MAD and quantile plot.
</p>
<p>The <code>exploreBic</code> function checks if the parameters are appropriately submitted and then identifies the biclusters submatrix and calculates its summary statistics. Finally, the results are displayed on the required plot.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Value</h3>

<p>Summary plot will display according to the user specification.
</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay <a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Hochreiter, S., Bodenhofer, U., Heusel, M.<em>et al.</em> (2010).FABIA: factor analysis for bicluster acquisition. <em>Bioinformatices</em>, 26, 1520-1527.
</p>


<h3>See Also</h3>

<p><code>exploreOnlybic</code></p>


<h3>Examples</h3>

<pre><code class="language-R">data(breastc)
# find bicluster using biclust package
library(biclust)
bic &lt;- biclust(breastc,method=BCPlaid())
# Plot the mean of biclusterd and clustered genes parallely.
exploreBic(dset=breastc,bres=bic,gby="conditions",pfor="mean",mname="biclust")
</code></pre>


</div>