<div class="container">

<table style="width: 100%;"><tr>
<td>writeBic</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>The writeBic function</h2>

<h3>Description</h3>

<p>Provides a summary output in a text format, extracted from 'biclust','isa2' and 'fabia' bicluster algorithms. </p>


<h3>Usage</h3>

<pre><code class="language-R">writeBic(dset, fileName, bicResult, bicname, 
mname = c("fabia", "isa2", "biclust","bicare"), append = TRUE, delimiter = " ", 
fabia.thresZ=0.5,fabia.thresL=NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dset</code></td>
<td>
<p>data matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fileName</code></td>
<td>
<p>the name of the bicluster file to be saved.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bicResult</code></td>
<td>
<p>bicluster result obtained from 'biclust','isa2' or 'fabia'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bicname</code></td>
<td>
<p>the title to be given for the biclustered data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mname</code></td>
<td>
<p>method name; 'biclust', 'isa2', 'fabia' or 'bicare'</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p>logical value; TRUE as default</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>delimiter</code></td>
<td>
<p>delimiter in created output file; default value is " ".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fabia.thresZ</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for sample belonging to bicluster; default 0.5.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fabia.thresL</code></td>
<td>
<p>Bicluster threshold for <code>mname="fabia"</code>. Threshold for loading belonging to bicluster (if not given it is estimated).</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The original function was developed in 'biclust' package by Kaiser <em>et.al</em> (2008). We extend the function to be used for further bicluster algorithms, such as; 'isa2', 'fabia' and 'bicare'.
</p>
<p>Note that the <code>"biclust"</code> option for <code>mname</code> will also accept results from the packages <span class="pkg">iBBiG</span> and <span class="pkg">rqubic</span>.
</p>


<h3>Value</h3>

<p>Biclustered text file with title, total number of biclustered, dimension and name of the biclustered genes(rows) or conditions(columns).</p>


<h3>Author(s)</h3>

<p>Mengsteab Aregay 
</p>
<p><a href="mailto:mycs.zab@gmail.com">mycs.zab@gmail.com</a></p>


<h3>References</h3>

<p>Van't Veer, L.J., Dai, H., van de Vijver, M.J., He, Y.D., Hart, A.A. et al. (2002). Gene expression profiling predicts clinical outcome of breast cancer,<em>Nature</em>, 415, 530-536.
</p>
<p>Kaiser S. and Leisch F. (2008). A Toolbox for Bicluster Analysis in R. <em>Ludwigstrasse.</em> 33.
</p>
<p>Csardi G., Kutalik Z., and Bergmann S.(2010). Modular analysis of gene expression data with R. <em>Bioinformatics,</em> 26, 1376-7
</p>


<h3>See Also</h3>

<p><span class="pkg">biclust</span></p>


<h3>Examples</h3>

<pre><code class="language-R"># create the biclustering result
data(breastc)
library(fabia)
fab&lt;- fabia(breastc)
# write the biclustering result into a text file
writeBic(dset=breastc,fileName="fabiabreast.txt",
bicResult=fab, bicname="Biclust results for fabia",
mname="fabia")
</code></pre>


</div>