<div class="container">

<table style="width: 100%;"><tr>
<td>default_params_doc</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.</h2>

<h3>Description</h3>

<p>Documentation of general function arguments.
This function does nothing.
It is intended to inherit function argument documentation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">default_params_doc(
  alignment_id,
  allow_empty_str,
  allow_na,
  alpha_parameter,
  b_pop_sizes_param,
  b_pop_sizes_parameter,
  bd_tree_prior,
  beautier_folder,
  cbs_tree_prior,
  beast2_version,
  beauti_options,
  beta_parameter,
  ccp_tree_prior,
  cep_tree_prior,
  chain_length,
  clock_model,
  clock_model_name,
  clock_model_names,
  clock_models,
  clock_prior_distr_id,
  clock_rate_param,
  crown_age,
  crown_ages,
  distr_id,
  fasta_filename,
  fasta_filenames,
  filename,
  fixed_crown_age,
  fixed_crown_ages,
  freq_param,
  gamma_distr,
  gamma_site_model,
  group_sizes_dimension,
  gtr_site_model,
  has_non_strict_clock_model,
  has_tip_dating,
  hky_site_model,
  id,
  ids,
  inference_model,
  inference_models,
  initial_phylogenies,
  input_filename,
  input_filenames,
  is_monophyletic,
  jc69_site_model,
  kappa_param,
  log_every,
  m_param,
  mcmc,
  mode,
  mrca_prior,
  mrca_priors,
  mrca_prior_name,
  n_init_attempts,
  output_filename,
  param,
  param_id,
  phylogeny,
  pop_sizes_scaler_scale_factor,
  pre_burnin,
  rate_scaler_factor,
  rename_fun,
  rln_clock_model,
  sample_from_prior,
  sanitise_headers,
  screenlog,
  sequence_length,
  site_model,
  site_model_name,
  site_model_names,
  site_models,
  sort,
  store_every,
  strict_clock_model,
  taxa_names,
  tipdates_filename,
  tn93_site_model,
  tracelog,
  treelog,
  tree_prior,
  tree_prior_name,
  tree_prior_names,
  tree_priors,
  verbose,
  yule_tree_prior
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignment_id</code></td>
<td>
<p>ID of the alignment,
as returned by get_alignment_id.
Keep at <code>NA</code> to have it initialized automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allow_empty_str</code></td>
<td>
<p>allow a string to be empty</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>allow_na</code></td>
<td>
<p>allow NA</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha_parameter</code></td>
<td>
<p>an alpha parameter,
as created by create_alpha_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b_pop_sizes_param</code></td>
<td>
<p>a Bayesian population size parameter,
as created by create_b_pop_sizes_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b_pop_sizes_parameter</code></td>
<td>
<p>a Bayesian population size parameter,
as created by create_b_pop_sizes_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bd_tree_prior</code></td>
<td>
<p>a Birth-Death tree prior, as created
by <code>create_bd_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beautier_folder</code></td>
<td>
<p>the path to
the beautier temporary files folder</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cbs_tree_prior</code></td>
<td>
<p>a Coalescent Bayesian Skyline tree prior,
as returned by <code>create_cbs_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beast2_version</code></td>
<td>
<p>BEAST2 version, for example, <code>"2.5"</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beauti_options</code></td>
<td>
<p>one BEAUti options object,
as returned by <code>create_beauti_options</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta_parameter</code></td>
<td>
<p>a beta parameter,
as created by create_beta_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ccp_tree_prior</code></td>
<td>
<p>a Coalescent Constant Population tree prior,
as returned by <code>create_ccp_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cep_tree_prior</code></td>
<td>
<p>a Coalescent Exponential Population tree prior,
as returned by <code>create_cep_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain_length</code></td>
<td>
<p>length of the MCMC chain</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_model</code></td>
<td>
<p>a clock model,
as returned by <code>create_clock_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_model_name</code></td>
<td>
<p>name of a clock model,
must be a name as returned by <code>get_clock_model_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_model_names</code></td>
<td>
<p>one or more names of a clock model,
must be name among those returned by <code>get_clock_model_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_models</code></td>
<td>
<p>a list of one or more clock models,
as returned by <code>create_clock_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_prior_distr_id</code></td>
<td>
<p>ID of an MRCA clock model's distribution.
Keep at <code>NA</code> to have it initialized automatically</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>clock_rate_param</code></td>
<td>
<p>a <code>clockRate</code> parameter,
a numeric value,
as created by create_clock_rate_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crown_age</code></td>
<td>
<p>the crown age of the phylogeny</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>crown_ages</code></td>
<td>
<p>the crown ages of the phylogenies. Set to NA
if the crown age needs to be estimated</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>distr_id</code></td>
<td>
<p>a distributions' ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta_filename</code></td>
<td>
<p>a FASTA filename.
Use <code>get_fasta_filename</code> to obtain a testing FASTA filename.
Note that BEAST2 also supports missing data,
by using a dash (<code>-</code>) or question mark (<code>?</code>)
as a sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code>get_fasta_filename</code> to obtain a testing FASTA filename.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>a filename, as can be checked by check_filename</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixed_crown_age</code></td>
<td>
<p>determines if the phylogeny's crown age is
fixed. If FALSE, crown age is estimated by BEAST2. If TRUE,
the crown age is fixed to the crown age
of the initial phylogeny.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixed_crown_ages</code></td>
<td>
<p>one or more booleans to determine if the
phylogenies' crown ages are fixed.
If FALSE, crown age is estimated by BEAST2. If TRUE,
the crown age is fixed to the crown age
of the initial phylogeny.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>freq_param</code></td>
<td>
<p>a 'freq' parameter,
as created by create_freq_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma_distr</code></td>
<td>
<p>a gamma distribution,
as created by <code>create_gamma_distr</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gamma_site_model</code></td>
<td>
<p>a site model's gamma site model,
as returned by <code>create_gamma_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group_sizes_dimension</code></td>
<td>
<p>the group sizes' dimension,
as used by the CBS tree prior (see <code>create_cbs_tree_prior</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gtr_site_model</code></td>
<td>
<p>a GTR site model,
as returned by <code>create_gtr_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>has_non_strict_clock_model</code></td>
<td>
<p>boolean to indicate that the is
already at least one non-strict (i.e. relaxed log-normal) clock model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>has_tip_dating</code></td>
<td>
<p>TRUE if the user has supplied tip dates,
FALSE otherwise</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hky_site_model</code></td>
<td>
<p>an HKY site model,
as returned by <code>create_hky_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>an alignment's IDs.
An ID can be extracted from its FASTA filename
with <code>get_alignment_ids_from_fasta_filenames</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p>one or more alignments' IDs.
IDs can be extracted from their FASTA filenames
with <code>get_alignment_ids_from_fasta_filenames</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inference_model</code></td>
<td>
<p>a Bayesian phylogenetic inference model.
An inference model is the complete model setup in which a site model,
clock model, tree prior and more are specified.
Use create_inference_model to create an inference model.
Use check_inference_model to check if  an inference model is valid.
Use rename_inference_model_filenames to rename the files in an
inference model.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inference_models</code></td>
<td>
<p>a list of one or more inference models,
as can be created by create_inference_model</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>initial_phylogenies</code></td>
<td>
<p>one or more MCMC chain's initial phylogenies.
Each one set to <code>NA</code> will result in BEAST2 using a random phylogeny.
Else the phylogeny is assumed to be of class <code>phylo</code> from the
<code>ape</code> package</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input_filename</code></td>
<td>
<p>A FASTA filename.
Use <code>get_fasta_filename</code> to obtain a testing FASTA filename.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input_filenames</code></td>
<td>
<p>One or more FASTA filenames.
Use <code>get_fasta_filename</code> to obtain a testing FASTA filename.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is_monophyletic</code></td>
<td>
<p>boolean to indicate monophyly is assumed in
a Most Recent Common Ancestor prior,
as returned by <code>create_mrca_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>jc69_site_model</code></td>
<td>
<p>a JC69 site model,
as returned by <code>create_jc69_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kappa_param</code></td>
<td>
<p>a kappa parameter,
as created by create_kappa_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_every</code></td>
<td>
<p>number of MCMC states between writing to file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>m_param</code></td>
<td>
<p>an m parameter,
as created by create_m_param</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mcmc</code></td>
<td>
<p>one MCMC.
Use <code>create_mcmc</code> to create an MCMC.
Use <code>create_ns_mcmc</code> to create an MCMC
for a Nested Sampling run.
Use <code>check_mcmc</code> to check if an MCMC is valid.
Use <code>rename_mcmc_filenames</code> to rename the filenames in an MCMC.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p>mode how to log.
Valid values are the ones returned by get_log_modes</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrca_prior</code></td>
<td>
<p>a Most Recent Common Ancestor prior,
as returned by <code>create_mrca_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrca_priors</code></td>
<td>
<p>a list of one or more Most Recent Common Ancestor priors,
as returned by <code>create_mrca_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mrca_prior_name</code></td>
<td>
<p>the unique name of the MRCA prior,
for example a genus, family,
order or even class name.
Leave at NA to have it named automatically.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_init_attempts</code></td>
<td>
<p>number of initialization attempts before failing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_filename</code></td>
<td>
<p>Name of the XML parameter file created by this
function. BEAST2 uses this file as input.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param</code></td>
<td>
<p>a parameter, as can be created by <code>create_param</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>param_id</code></td>
<td>
<p>a parameter's ID</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phylogeny</code></td>
<td>
<p>a phylogeny of type <code>phylo</code> from the <code>ape</code>
package</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pop_sizes_scaler_scale_factor</code></td>
<td>
<p>the scale factor used by the
population sizes scaler operator</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pre_burnin</code></td>
<td>
<p>number of burn in samples taken before entering
the main loop</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rate_scaler_factor</code></td>
<td>
<p>the strict clock model's operator scaler
for the rate.
Use an empty string to indicate the default.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rename_fun</code></td>
<td>
<p>a function to rename a filename,
as can be checked by check_rename_fun. This function should
have one argument, which will be a filename or NA. The
function should return one filename (when passed one filename) or
one NA (when passed one NA).
Example rename functions are:
</p>

<ul>
<li> <p>get_remove_dir_fun get a function that removes the directory
paths from the filenames, in effect turning these into local files
</p>
</li>
<li> <p>get_replace_dir_fun get a function that replaces the directory
paths from the filenames
</p>
</li>
<li> <p>get_remove_hex_fun get a function that removes the
hex string from filenames.
For example, <code>tracelog_82c1a522040.log</code> becomes <code>tracelog.log</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rln_clock_model</code></td>
<td>
<p>a Relaxed Log-Normal clock model,
as returned by <code>create_rln_clock_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample_from_prior</code></td>
<td>
<p>set to TRUE to sample from the prior</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sanitise_headers</code></td>
<td>
<p>set to TRUE to sanitise the headers of the
log file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>screenlog</code></td>
<td>
<p>a <code>screenlog</code>,
as created by create_screenlog</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence_length</code></td>
<td>
<p>a DNA sequence length, in base pairs</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>site_model</code></td>
<td>
<p>a site model,
as returned by <code>create_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>site_model_name</code></td>
<td>
<p>name of a site model,
must be a name as returned by <code>get_site_model_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>site_model_names</code></td>
<td>
<p>one or more names of a site model,
must be name among those returned by <code>get_site_model_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>site_models</code></td>
<td>
<p>one or more site models,
as returned by <code>create_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sort</code></td>
<td>
<p>how to sort the log.
Valid values are the ones returned by get_log_sorts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>store_every</code></td>
<td>
<p>number of states the MCMC will process
before the posterior's state will be saved to file.
Use -1 or <code>NA</code> to use the default frequency.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strict_clock_model</code></td>
<td>
<p>a strict clock model,
as returned by <code>create_strict_clock_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxa_names</code></td>
<td>
<p>names of the taxa,
as returned by <code>get_taxa_names</code>.
Keep at <code>NA</code> to have it initialized automatically,
using all taxa in the alignment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tipdates_filename</code></td>
<td>
<p>name of the file containing the tip dates.
This file is assumed to have two columns, separated by a tab.
The first column contains the taxa names, the second column contains
the date.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tn93_site_model</code></td>
<td>
<p>a TN93 site model,
as returned by <code>create_tn93_site_model</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tracelog</code></td>
<td>
<p>a <code>tracelog</code>,
as created by create_tracelog</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>treelog</code></td>
<td>
<p>a <code>treelog</code>,
as created by create_treelog</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_prior</code></td>
<td>
<p>a tree priors,
as returned by <code>create_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_prior_name</code></td>
<td>
<p>name of a tree prior,
must be a name as returned by <code>get_tree_prior_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_prior_names</code></td>
<td>
<p>one or more names of a tree prior,
must be a name among those returned by <code>get_tree_prior_names</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree_priors</code></td>
<td>
<p>one or more tree priors,
as returned by <code>create_tree_prior</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>if TRUE, additional information is displayed, that
is potentially useful in debugging</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>yule_tree_prior</code></td>
<td>
<p>a Yule tree_prior,
as created by <code>create_yule_tree_prior</code></p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>This is an internal function, so it should be marked with
<code>@export</code>. This is not done, as this will disallow all
functions to find the documentation parameters
</p>


<h3>Author(s)</h3>

<p>Rich√®l J.C. Bilderbeek
</p>


</div>