<div class="container">

<table style="width: 100%;"><tr>
<td>getEnsemblTranscriptIds</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Feeding BED: Download Ensembl DB and load transcript information in BED</h2>

<h3>Description</h3>

<p>Not exported to avoid unintended modifications of the DB.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getEnsemblTranscriptIds(organism, release, gv, ddir, dbCref, canChromosomes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>character vector of 1 element corresponding to the organism
of interest (e.g. "Homo sapiens")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>release</code></td>
<td>
<p>the Ensembl release of interest (e.g. "83")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gv</code></td>
<td>
<p>the genome version (e.g. "38")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ddir</code></td>
<td>
<p>path to the directory where the data should be saved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dbCref</code></td>
<td>
<p>a named vector of characters providing cross-reference DB of
interest. These DB are also used to find indirect ID associations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>canChromosomes</code></td>
<td>
<p>canonical chromosmomes to be considered as preferred
ID (e.g. c(1:22, "X", "Y", "MT") for human)</p>
</td>
</tr>
</table>
</div>