<div class="container">

<table style="width: 100%;"><tr>
<td>firstCommonUpstreamBe</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>First common upstream BE</h2>

<h3>Description</h3>

<p>Returns the first common Biological Entity (BE) upstream a set of BE.
</p>


<h3>Usage</h3>

<pre><code class="language-R">firstCommonUpstreamBe(beList = listBe(), uniqueOrg = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>beList</code></td>
<td>
<p>a character vector containing BE</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>uniqueOrg</code></td>
<td>
<p>a logical value indicating if as single organism is under
focus. If false "Gene" is returned.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is used to identified the level at which different BE should be
compared. Peptides and transcripts should be compared at the level of
transcripts whereas transcripts and objects should be compared at the
level of genes. BE from different organism should be compared at the level
of genes using homologs.
</p>


<h3>See Also</h3>

<p>listBe
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
firstCommonUpstreamBe(c("Object", "Transcript"))
firstCommonUpstreamBe(c("Peptide", "Transcript"))
firstCommonUpstreamBe(c("Peptide", "Transcript"), uniqueOrg=FALSE)

## End(Not run)

</code></pre>


</div>