<div class="container">

<table style="width: 100%;"><tr>
<td>calculate.pairwise.Fst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculates unbiased pairwise Fst between a pair of populations
</h2>

<h3>Description</h3>

<p>This function calculates unbiased <code class="reqn">F_{ST}</code> (based on Weir and Hill's 
<code class="reqn">\theta</code>, 2002), between a pair of populations/individuals. Loci for which either 
of the populations/individuals has missing data (i.e. - the sample size is zero) are 
excluded.  
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculate.pairwise.Fst(allele.counts, sample.sizes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>allele.counts</code></td>
<td>

<p>A matrix of allele counts of dimensions <code>nrow = </code> 2, one for each of the two 
populations/individuals between which pairwise <code class="reqn">F_{ST}</code> is being calculated, and 
<code>ncol = </code> the number of loci genotyped.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.sizes</code></td>
<td>

<p>A matrix of sample sizes of dimensions <code>nrow = </code> 2, one for each of the two 
populations/individuals between which pairwise <code class="reqn">F_{ST}</code> is being calculated, and 
<code>ncol = </code> the number of loci genotyped (i.e. - the dimensions of
<code>sample.sizes</code> must match those of <code>counts</code>).  Each cell gives the number 
of chromosomes successfully genotyped at each locus in each population.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Pairwise unbiased <code class="reqn">F_{ST}</code> between a pair of populations/individuals
</p>


<h3>Author(s)</h3>

<p>Gideon Bradburd
</p>


<h3>References</h3>

<p>Weir,B.S. and W.G. Hill. 2002. Estimating F-statistics. Ann.Rev.Gen. 36:949-952.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#With the HGDP dataset
	data(HGDP.bedassle.data)

#Draw 2 populations at random from the Eurasian HGDP dataset
	pop1 &lt;- sample(nrow(HGDP.bedassle.data$allele.counts),1)
	pop2 &lt;- sample(nrow(HGDP.bedassle.data$allele.counts),1)

#Calculate unbiased Fst between them
	pairwise.Fst &lt;- calculate.pairwise.Fst(
		HGDP.bedassle.data$allele.counts[c(pop1,pop2),],
		HGDP.bedassle.data$sample.sizes[c(pop1,pop2),]	
	)

#Print that Fst to the console
	print(sprintf("Fst between the %s population and the %s population is %s",
		HGDP.bedassle.data$hgdp.metadata[pop1,1],
		HGDP.bedassle.data$hgdp.metadata[pop2,1],
		round(pairwise.Fst,3))
	 )
</code></pre>


</div>