<div class="container">

<table style="width: 100%;"><tr>
<td>HGDP.bedassle.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
The Eurasian subset of the HGDP dataset used in example BEDASSLE analyses 
</h2>

<h3>Description</h3>

<p>The allelic counts, sample sizes, geographic distances, ecological distances, and 
population metadata from the 38 human populations used in example BEDASSLE 
analyses, subsetted from the Human Genome Diversity Panel (HGDP) dataset.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(HGDP.bedassle.data)</code></pre>


<h3>Format</h3>

<p>The format is:
List of 7
</p>

<dl>
<dt>$ allele.counts        :</dt>
<dd>
<p>int [1:38, 1:1000] 12 16 5 17 4 14 20 5 34 ...</p>
</dd>
</dl>
<ul>
<li>
<p>..- attr(*, "dimnames")=List of 2
</p>
</li>
<li>
<p>.. ..$ : chr [1:38] "Adygei" "Basque" "Italian" "French" ...
</p>
</li>
<li>
<p>.. ..$ : chr [1:1000] "rs13287637" "rs17792496" "rs1968588" ...
</p>
</li>
</ul>
<dl>
<dt>$ sample.sizes         :</dt>
<dd>
<p>int [1:38, 1:1000] 34 48 24 56 30 50 56 ...</p>
</dd>
</dl>
<ul>
<li>
<p>..- attr(*, "dimnames")=List of 2
</p>
</li>
<li>
<p>.. ..$ : chr [1:38] "Adygei" "Basque" "Italian" "French" ...
</p>
</li>
<li>
<p>.. ..$ : chr [1:1000] "rs13287637" "rs17792496" "rs1968588" ...
</p>
</li>
</ul>
<dl>
<dt>$ GeoDistance          :</dt>
<dd>
<p>num [1:38, 1:38] 0 1.187 0.867 1.101 1.247 ...</p>
</dd>
<dt>$ EcoDistance          :</dt>
<dd>
<p>num [1:38, 1:38] 0 0 0 0 0 0 0 0 0 0 ...</p>
</dd>
</dl>
<ul>
<li>
<p>..- attr(*, "dimnames")=List of 2
</p>
</li>
<li>
<p>.. ..$ : chr [1:38] "1" "2" "3" "4" ...
</p>
</li>
<li>
<p>.. ..$ : chr [1:38] "1" "2" "3" "4" ...
</p>
</li>
</ul>
<dl>
<dt>$ number.of.populations:</dt>
<dd>
<p>int 38</p>
</dd>
<dt>$ number.of.loci       :</dt>
<dd>
<p>int 1000</p>
</dd>
<dt>$ hgdp.metadata        :</dt>
<dd>
<p>'data.frame':	38 obs. of  3 variables:</p>
</dd>
</dl>
<dl>
<dt>..$ Population:</dt>
<dd>
<p> chr [1:38] "Adygei" "Basque" "Italian" ...</p>
</dd>
<dt>..$ Latitude  :</dt>
<dd>
<p> chr [1:38] "44" "43" "46" "46" ...</p>
</dd>
<dt>..$ Longitude :</dt>
<dd>
<p> chr [1:38] "39" "0" "10" "2" ...</p>
</dd>
</dl>
<h3>Details</h3>


<dl>
<dt>allele.counts</dt>
<dd>
<p>A matrix of allelic count data, for which <code>nrow = </code> 
the number of populations and <code>ncol = </code> the number of bi-allelic loci 
sampled.  Each cell gives the number of times allele ‘1’ is observed in each 
population.  The choice of which allele is allele ‘1’ is arbitrary, but must 
be consistent across all populations at a locus.</p>
</dd>
<dt>sample.sizes</dt>
<dd>
<p>A matrix of sample sizes, for which <code>nrow = </code> the number 
of populations and <code>ncol = </code> the number of bi-allelic loci sampled 
(i.e. - the dimensions of <code>sample.sizes</code> must match those of 
<code>counts</code>).  Each cell gives the number of chromosomes successfully 
genotyped at each locus in each population.</p>
</dd>
<dt>Geo.Distance</dt>
<dd>
<p>Pairwise geographic distance (<code class="reqn">D_{i,j}</code>).  This may be 
Euclidean, or, if the geographic scale of sampling merits it, great-circle 
distance.  In the case of this dataset, it is great-circle distance.</p>
</dd>
<dt>Eco.Distance</dt>
<dd>
<p>Pairwise ecological distance(s) (<code class="reqn">E_{i,j}</code>), which may 
be continuous (e.g. - difference in elevation) or binary (same or opposite 
side of some hypothesized barrier to gene flow).  In this case, the 
ecological distance is binary, representing whether a pair of populations 
occurs on the same side, or on opposite sides, of the Himalayas.</p>
</dd>
<dt>number.of.populations</dt>
<dd>
<p>The number of populations in the analysis.  
This should be equal to <code>nrow(</code>counts<code>)</code>.  In this dataset, there
are 38 populations sampled.</p>
</dd>
<dt>number.of.loci</dt>
<dd>
<p>The number of loci in the analysis.  This should be equal 
to <code>ncol(</code>counts<code>)</code>.  In this dataset, there are 1000 loci 
sampled.</p>
</dd>
<dt>hgdp.metadata</dt>
<dd>
<p>This data frame contains the metadata on the populations
included in the analysis, including:
</p>

<ul>
<li>
<p>Population name
</p>
</li>
<li>
<p>Latitude
</p>
</li>
<li>
<p>Longitude
</p>
</li>
</ul>
</dd>
</dl>
<h3>Source</h3>


<ul>
<li>
<p>Conrad et al. A worldwide survey of haplotype variation and linkage
disequilibrium in the human genome. <em>Nature Genetics</em> 2008.
</p>
</li>
<li>
<p>Li et al. Worldwide human relationships inferred from genome-wide 
patterns of variation. <em>Science</em> 2008.
</p>
</li>
</ul>
<h3>References</h3>

<p>Bradburd, G.S., Ralph, P.L., and Coop, G.M. Disentangling the effects of 
geographic and ecological isolation on genetic differentiation. 
<em>Evolution</em> 2013.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## see \command{MCMC}, \command{MCMC_BB}, \command{calculate.pariwise.Fst}, 
## \command{calculate.all.pairwise.Fst}, and \command{Covariance} for usage.
</code></pre>


</div>