<div class="container">

<table style="width: 100%;"><tr>
<td>calculate.all.pairwise.Fst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Calculates unbiased pairwise Fst between all sampled populations
</h2>

<h3>Description</h3>

<p>This function calculates unbiased <code class="reqn">F_{ST}</code> (based on Weir and Hill's <code class="reqn">\theta</code>, 
2002), between all populations/individuals included in the <code>counts</code> matrix, and 
returns the results in a <code>k</code> by <code>k</code> matrix, where <code>k = nrow(counts)</code>.  
Loci for which either of the populations/individuals has missing data (i.e. - the sample 
size is zero) are excluded.
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculate.all.pairwise.Fst(allele.counts, sample.sizes)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>allele.counts</code></td>
<td>

<p>A matrix of allelic count data, for which <code>nrow = </code> the number of populations
and <code>ncol = </code> the number of bi-allelic loci sampled.  Each cell gives the number 
of times allele ‘1’ is observed in each population.  The choice of which allele is 
allele ‘1’ is arbitrary, but must be consistent across all populations at a locus.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.sizes</code></td>
<td>

<p>A matrix of sample sizes, for which <code>nrow = </code> the number of populations
and <code>ncol = </code> the number of bi-allelic loci sampled (i.e. - the dimensions of
<code>sample.sizes</code> must match those of <code>counts</code>).  Each cell gives the number 
of chromosomes successfully genotyped at each locus in each population.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A matrix of pairwise, unbiased <code class="reqn">F_{ST}</code>.
</p>


<h3>Author(s)</h3>

<p>Gideon Bradburd
</p>


<h3>References</h3>

<p>Weir,B.S. and W.G. Hill. 2002. Estimating F-statistics. Ann.Rev.Gen. 36:949-952.
</p>


<h3>Examples</h3>

<pre><code class="language-R">#With the HGDP dataset
	data(HGDP.bedassle.data)

#Calculate pairwise Fst between all population pairs
	hgdp.pairwise.Fst &lt;- calculate.all.pairwise.Fst(
		HGDP.bedassle.data$allele.counts,
		HGDP.bedassle.data$sample.sizes
	)	

#Plot pairwise Fst against geographic distance
	plot(HGDP.bedassle.data$GeoDistance,
		hgdp.pairwise.Fst,
		pch=19,
		col=HGDP.bedassle.data$EcoDistance+1,
		ylab="pairwise Fst",
		xlab="geographic distance",
		main="isolation by distance")
			legend(x="bottomright",pch=19,col=c(1,2),
				legend=c("same side of Himalayas",
						"opposite sides of Himalayas"))
</code></pre>


</div>