<div class="container">

<table style="width: 100%;"><tr>
<td>BEDMatrix</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a BEDMatrix Object from a PLINK .bed File</h2>

<h3>Description</h3>

<p>This function constructs a new <code>BEDMatrix</code> object by mapping the
specified <a href="https://www.cog-genomics.org/plink2/formats#bed">PLINK
.bed</a> file into memory.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BEDMatrix(path, n = NULL, p = NULL, simple_names = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>

<p>Path to the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file (with
or without extension).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>The number of samples. If <code>NULL</code> (the default), this number will
be determined from the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> file (of
the same name as the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file). If
a positive integer, the .fam file is not read and <code>rownames</code> will
be set to <code>NULL</code> and have to be provided manually.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>

<p>The number of variants. If <code>NULL</code> (the default) the number of
variants will be determined from the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> file (of
the same name as the
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> file). If
a positive integer, the .bim file is not read and <code>colnames</code> will
be set to <code>NULL</code> and have to be provided manually.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>simple_names</code></td>
<td>

<p>Whether to simplify the format of the dimension names. If <code>FALSE</code>
(the default), row names are concatenations of family IDs, <code>_</code>,
and within-family IDs, while column names are concatenations of variant
names, <code>_</code>, and allele codes for the first allele of the
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> file
(<code>A1</code>). If <code>TRUE</code>, row names are within-family IDs only and
column names are variant names only.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p><a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files must be
accompanied by <a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a>
and <a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files:
.fam files contain sample information, and .bim files contain variant
information. If the name of the .bed file is <em>plink</em>.bed then the
names of the .fam and .bim files have to be <em>plink</em>.fam and
<em>plink</em>.bim, respectively. The .fam and .bim files are used to extract
the number and names of samples and variants.
</p>
<p>For very large .bed files, reading the .fam and .bim files can take a long
time. If <code>n</code> and <code>p</code> are provided, these files are not read and
<code>dimnames</code> have to be provided manually.
</p>
<p>Currently, only the variant-major mode of .bed files is supported.
<a href="https://www.cog-genomics.org/plink2/">PLINK2</a> "dropped" support for
the sample-major mode by automatically converting files in this format to
the variant-major mode. Therefore, it is recommended to run files in
sample-major mode through PLINK2 first.
</p>


<h3>Value</h3>

<p>A <code>BEDMatrix</code> object.
</p>


<h3>See Also</h3>

<p><code>BEDMatrix-package</code> to learn more about the <code>BEDMatrix</code>
package, <code>BEDMatrix-class</code> to learn more about the
<code>BEDMatrix</code> class.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Get the path to the example .bed file
path &lt;- system.file("extdata", "example.bed",
                    package = "BEDMatrix")

# Create a BEDMatrix object the example .bed file
m1 &lt;- BEDMatrix(path)

# Create a BEDMatrix object the example .bed file without loading
# the .fam and .bim files
m2 &lt;- BEDMatrix(path, n = 50, p = 1000)
</code></pre>


</div>