<div class="container">

<table style="width: 100%;"><tr>
<td>BEDMatrix-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Package to Extract Genotypes from a PLINK .bed File</h2>

<h3>Description</h3>

<p>The <code>BEDMatrix</code> package provides a matrix-like wrapper around
<a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files, one of
the genotype/phenotype file formats of
<a href="https://www.cog-genomics.org/plink2">PLINK</a>, the whole genome
association analysis toolset. <code>BEDMatrix</code> objects are created by
simply providing the path to a .bed file and once created, they behave
similarly to regular matrices with the advantage that genotypes are
retrieved on demand without loading the entire file into memory. This
allows handling of very large files with limited use of memory.
</p>


<h3>.bed Files</h3>

<p><a href="https://www.cog-genomics.org/plink2/formats#bed">.bed</a> files
(sometimes referred to as binary .ped files) are binary representations of
genotype calls at biallelic variants. This very compact file format (2 bits
per genotype call) is used and generated by
<a href="https://www.cog-genomics.org/plink2/">PLINK</a>. .bed files should not
be confused with the
<a href="https://genome.ucsc.edu/FAQ/FAQformat.html#format1">UCSC Genome
Browser's BED format</a>, which is totally different.
</p>
<p>A .bed file can be created from a
<a href="https://www.cog-genomics.org/plink2/formats#ped">.ped</a> file with
<a href="https://www.cog-genomics.org/plink2">PLINK</a> using <code>plink --file
    myfile --make-bed</code>.
</p>


<h3>See Also</h3>

<p><code>BEDMatrix-class</code> to learn more about the <code>BEDMatrix</code>
class.
</p>


</div>