<div class="container">

<table style="width: 100%;"><tr>
<td>BEDMatrix-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Class to Extract Genotypes from a PLINK .bed File</h2>

<h3>Description</h3>

<p><code>BEDMatrix</code> is a class that maps a
<a href="https://www.cog-genomics.org/plink2/formats#bed">PLINK .bed</a> file
into memory and behaves similarly to a regular <code>matrix</code> by
implementing key methods such as <code>[</code>, <code>dim</code>, and <code>dimnames</code>.
Subsets are extracted directly and on-demand from the .bed file without
loading the entire file into memory.
</p>


<h3>Details</h3>

<p>The subsets extracted from a <code>BEDMatrix</code> object are coded as the
allelic dosages of the first allele in the .bim file (A1), similarly to
<a href="https://www.cog-genomics.org/plink2/formats#raw">.raw</a> files
generated with the <code>--recode A</code> argument in
<a href="https://www.cog-genomics.org/plink2/">PLINK</a>).
</p>
<p>Internally, this class is an S4 class with the following slots that should
not be relied upon in actual code: <code>xptr</code>, <code>dims</code>, <code>dnames</code>,
and <code>path</code>. The .bed file is mapped into memory using <code>mmap</code> on
Unix and <code>MapViewOfFile</code> on Windows.
</p>


<h3>Slots</h3>


<dl>
<dt>
<code>xptr</code>:</dt>
<dd>
<p>An external pointer to the underlying <code>C</code> code.
</p>
</dd>
<dt>
<code>dims</code>:</dt>
<dd>
<p>An integer vector specifying the number of samples and variants as
determined by the the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> and
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files
or by the <code>n</code> and <code>p</code> parameters of the <code>BEDMatrix</code>
constructor function.
</p>
</dd>
<dt>
<code>dnames</code>:</dt>
<dd>
<p>A list describing the row names and column names of the object as
determined by the accompanying
<a href="https://www.cog-genomics.org/plink2/formats#fam">.fam</a> and
<a href="https://www.cog-genomics.org/plink2/formats#bim">.bim</a> files,
or <code>NULL</code> if the <code>n</code> and <code>p</code> parameters of the
<code>BEDMatrix</code> constructor function were provided.
</p>
</dd>
<dt>
<code>path</code>:</dt>
<dd>
<p>A character string containing the path to the .bed file.
</p>
</dd>
</dl>
<h3>Methods</h3>


<dl>
<dt>
<code>[</code>:</dt>
<dd>
<p>Extract parts of an object</p>
</dd>
<dt>
<code>dim</code>:</dt>
<dd>
<p>Retrieve the dimension of an object</p>
</dd>
<dt>
<code>dimnames</code>:</dt>
<dd>
<p>Retrieve the dimnames of an object</p>
</dd>
<dt>
<code>dimnames&lt;-</code>:</dt>
<dd>
<p>Set the dimnames of an object</p>
</dd>
<dt>
<code>as.matrix</code>:</dt>
<dd>
<p>Turn the object into a matrix</p>
</dd>
<dt>
<code>is.matrix</code>:</dt>
<dd>
<p>Test if the object is a matrix</p>
</dd>
<dt>
<code>length</code>:</dt>
<dd>
<p>Get the length of an object</p>
</dd>
<dt>
<code>str</code>:</dt>
<dd>
<p>Display the internal structure of an object</p>
</dd>
<dt>
<code>show</code>:</dt>
<dd>
<p>Display the object</p>
</dd>
</dl>
<h3>See Also</h3>

<p><code>BEDMatrix</code> to create a <code>BEDMatrix</code> object from a .bed
file, <code>BEDMatrix-package</code> to learn more about the
<code>BEDMatrix</code> package,
<code>LinkedMatrix</code> to link
several <code>BEDMatrix</code> objects together.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Get the path to the example .bed file
path &lt;- system.file("extdata", "example.bed",
                    package = "BEDMatrix")

# Create a BEDMatrix object the example .bed file
m &lt;- BEDMatrix(path)

# Get the dimensions of the BEDMatrix object
dim(m)

# Get the row names of the BEDMatrix object
rownames(m)

# Get the column names of the BEDMatrix object
colnames(m)

# Extract genotypes for the specified sample(s)
m[1, ]
m[1:3, ]
m["per0_per0", ]
m[c("per0_per0", "per1_per1", "per2_per2"), ]

# Extract genotypes for a particular variant
m[, 1]
m[, c("snp0_A", "snp1_C", "snp2_G")]

# Extract genotypes for the specified samples and variants
m[
    c("per0_per0", "per1_per1", "per2_per2"),
    c("snp0_A", "snp1_C", "snp2_G")
]
</code></pre>


</div>