<div class="container">

<table style="width: 100%;"><tr>
<td>bedr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Main bedtools wrapper function.
</h2>

<h3>Description</h3>

<p>Main bedtools wrapper function.
</p>


<h3>Usage</h3>

<pre><code class="language-R">bedr(engine = "bedtools", 
	params = NULL,
	input = list(),
	method = NULL,
	tmpDir = NULL,
	deleteTmpDir = TRUE,
	outputDir = NULL,
	outputFile = NULL,
	check.chr = TRUE,
	check.zero.based = TRUE,
	check.valid = TRUE,
	check.sort = TRUE,
	check.merge = TRUE,
	verbose = TRUE
	);

</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>engine</code></td>
<td>
<p>What analytical engine to use i.e. bedtools, bedops, tabix, unix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>params</code></td>
<td>
<p>A string that includes all the extra parameters and arguments to the bedtools commmand.  For example if you wanted to do a left outer join you would specificy method as intersect and use params = c("-loj -header"). If you leave input and method as defaults then this is this string represents the full command.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>input</code></td>
<td>
<p>A list of input items to be used by bedtools.  Each item should be named by its parameter name in bedtools for example input = list(a=xxx, b=yyy, i=zzz).  Items can be R objects or external files.  R objects need to be in bed format i.e. have chr, start, stop as the first three columns, or, have an position index as the first column or rowname i.e. chr1:100-1000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>What bedtools method.  This can be intersect, sort, merge etc.  See bedtools documentation for specifcis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tmpDir</code></td>
<td>
<p>The directory to be used for writing files</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>deleteTmpDir</code></td>
<td>
<p>Should tmp files be deleted.  helpful for diagnostics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputDir</code></td>
<td>
<p>The output directory.  Only used if outputFile is specified.  It defaults to the current working directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputFile</code></td>
<td>
<p>The name of the output file.  If this is specified the output will be sent to a file not an R object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.chr</code></td>
<td>
<p>check for chr prefix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.zero.based</code></td>
<td>
<p>check for zero based coordinates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.valid</code></td>
<td>
<p>do all region integrity checks</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.sort</code></td>
<td>
<p>check if region is sorted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.merge</code></td>
<td>
<p>check if region is  merged</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Should messages be printed to screen.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The output of command with some parsing to keep it consistent with the input. 
</p>


<h3>Author(s)</h3>

<p>Daryl Waggott
</p>


<h3>See Also</h3>

<p>iranges
</p>


<h3>Examples</h3>

<pre><code class="language-R">if (check.binary("bedtools")) {
set.seed(666)

index &lt;- get.example.regions();
a &lt;- index[[1]];
b &lt;- index[[2]];

### check
is.a.valid  &lt;- is.valid.region(a);
is.b.valid  &lt;- is.valid.region(b);
a &lt;- a[is.a.valid];
b &lt;- b[is.b.valid];

### sort
is.sorted &lt;- is.sorted.region(a);
a.sort1 &lt;- bedr(engine = "bedtools", input = list(i = a), method = "sort", params = "");
b.sort1 &lt;- bedr(engine = "bedtools", input = list(i = b), method = "sort", params = "");
a.sort2 &lt;- bedr(engine = "bedops",   input = list(i = a), method = "sort", params = "");

a.sort3 &lt;- bedr.sort.region(a);
a.sort4 &lt;- bedr.sort.region(a, engine = "unix", method = "natural");
a.sort5 &lt;- bedr.sort.region(a, engine = "R", method = "natural");

### merge
is.merged &lt;- is.merged.region(a.sort1);
a.merge1 &lt;- bedr(engine = "bedtools", input = list(i = a.sort1), method = "merge", params = "");
b.merge1 &lt;- bedr(engine = "bedtools", input = list(i = b.sort1), method = "merge", params = "");
a.merge2 &lt;- bedr(engine = "bedops",   input = list(i = a.sort1), method = "merge", params = "");
# a.merge3 &lt;- bedr.merge.region(a); this will throw an error b/c region is not sorted

### subtract
a.sub1 &lt;- bedr(input = list(a = a.merge1, b = b.merge1), method = "subtract", params="");
a.sub2 &lt;- bedr.subtract.region(a.merge1, b.merge1);

### in.region
is.region &lt;- in.region(a.merge1, b.merge1);
#is.region &lt;- a.merge1 %in.region% b.merge1
### intersect 
# note for bedtools its recommended to bedr.sort.regions before intersect for faster processing
# also if regions are not merged this can cause unexpected behaviour
a.int1 &lt;- bedr(input = list(a = a.sort1, b = b.sort1), method = "intersect", params = "-loj");
a.int1 &lt;- bedr(input = list(a = a.sort1, b = b.sort1), method="intersect",params ="-loj -sorted");
a.int2 &lt;- bedr(input = list(a = a.merge1, b = b.merge1), method="intersect",params="-loj -sorted");
a.int3 &lt;- bedr.join.region(a.merge1, b.merge1);

### multiple join
d &lt;- get.random.regions(100, chr="chr1", sort = TRUE);
a.mult &lt;- bedr.join.multiple.region(x = list(a.merge1,b.merge1,bedr.sort.region(d)));

## Not run: 
  ### groupby 
  # note the "g" column number is based on bed format i.e. first three columns chr, start, stop
  # note the use of first, first, last on the region columns i.e. the union of the regions
  # note currently missing values are not dealt with in bedtools.  also the 5th column is 
  # assumed to be "score" and gets a default "-1" not a "."
  cnv.gene &lt;- bedr(
    input = list(i=cnv.gene), method = "groupby", params = paste(
      "-g 16 -c ", 
      paste(1:15, collapse = ","), 
      " -o ", "first,first,last, ", 
      paste(rep("sum",12), collapse = ","), 
      sep = ""
      )
    );

  ### example 1
  ###  workflow adding gene names to exome sequencing target file
  # download refseq genes from ucsc or query biomart for ensemble gene names.  
  # format them in basic bed format.
  # sort, merge target
  # sort, merge -nms target.  Overlapping genes/features get merged.  
  # this may not be ideal if there are some really big features.
  # intersect -loj target, genes.
  # alternatively, do not merge the target and apply the merge after the intersect.  
  # this will provide precision at the level of the exon.
  
## End(Not run)
}
</code></pre>


</div>