<div class="container">

<table style="width: 100%;"><tr>
<td>get.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Query fasta sequence
</h2>

<h3>Description</h3>

<p>Query fasta sequence using bedtools get.fasta
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.fasta(
	x,
	fasta = NULL,
	bed12 = FALSE,
	strand = FALSE,
	output.fasta = FALSE,
	use.name.field = FALSE,
	check.zero.based = TRUE,
	check.chr = TRUE,
	check.valid = TRUE,
	check.sort = TRUE,
	check.merge = TRUE,
	verbose = TRUE
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>region or index</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fasta</code></td>
<td>
<p>a fasta file defaults to mini example hg19 human</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bed12</code></td>
<td>
<p>should bed12 format be used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strand</code></td>
<td>
<p>strand specific i.e. reverse complement negative</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.fasta</code></td>
<td>
<p>output a fasta defaults to a data.frame for easier parsing</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use.name.field</code></td>
<td>
<p>should the name field be used for</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.zero.based</code></td>
<td>
<p>check for zero based region</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.chr</code></td>
<td>
<p>check for "chr" prefix </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.valid</code></td>
<td>
<p>check for valid regions i.e. start &lt; end</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.sort</code></td>
<td>
<p>check if region is sorted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check.merge</code></td>
<td>
<p>check if region is merged</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>print progress</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Uses bedtools getFasta to query a fasta file and load into R as a data.frame for easy parsing.
</p>
<p>Note that the hg19 reference genome fasta is large and requires on the order of 4 GB RAM to avoid a segfault happens.
</p>


<h3>Value</h3>

<p>A data.frame or fasta. The data.frame has is two columns corresponding to the region and the sequence.
</p>


<h3>Author(s)</h3>

<p>Daryl Waggott, Syed Haider
</p>


<h3>References</h3>

<p><a href="http://bedtools.readthedocs.org/en/latest/content/tools/getfasta.html">http://bedtools.readthedocs.org/en/latest/content/tools/getfasta.html</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">

if (check.binary("bedtools")) {

## Not run: 

  # get the sequence for a set of regions as a data.frame
  index &lt;- get.example.regions();
  a &lt;- index[[1]];
  b &lt;- get.fasta(bedr.sort.region(a));

  # this time output a fasta
  d &lt;- get.fasta(b, output.fasta = TRUE);
  writeLines(d[[1]], con = "test.fasta");

  # get the region adjacent to a set of mutations in a vcf
  clinvar.vcf.example &lt;- system.file(
    "extdata/clinvar_dbSNP138_example.vcf.gz", 
    package = "bedr"
    );
  clinvar &lt;- read.vcf(clinvar.vcf.example);

  # note that clinvar uses ncbi fasta which does not use "chr" and codes chrM as MT
  clinvar.bed &lt;- data.frame(
    chr = paste0("chr", gsub("MT", "M", clinvar$vcf$CHROM)),
    start = clinvar$vcf$POS - 2,
    end = clinvar$vcf$POS + 1,
    stringsAsFactors = FALSE
    );

  # get trinucleotide sequences of variants on chr M only
  mutation.triplet &lt;- get.fasta(
  	clinvar.bed[which(clinvar.bed$chr == "chrM"), ], 
  	fasta = system.file("extdata/ucsc.hg19.chrM.fasta", package = "bedr"),
  	check.chr = FALSE
  	);
  
## End(Not run)
}
</code></pre>


</div>