<div class="container">

<table style="width: 100%;"><tr>
<td>pca.methylation.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>DNA methylation data from patients with prostate cancer</h2>

<h3>Description</h3>

<p>A dataset containing pre-processed beta methylation values from <code class="reqn">R=2</code> sample types, collected from <code class="reqn">N=4</code> patients with prostate cancer.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(pca.methylation.data)
</code></pre>


<h3>Format</h3>

<p>A data frame with 5,067 rows and 9 columns. The data contain no missing values.
</p>

<ul>
<li>
<p>IlmnID: The unique identifier from the Illumina CG database, i.e. the probe ID.
</p>
</li>
<li>
<p>Benign_Patient_1: Methylation values from benign prostate tissue from patient 1.
</p>
</li>
<li>
<p>Benign_Patient_2: Methylation values from benign prostate tissue from patient 2.
</p>
</li>
<li>
<p>Benign_Patient_3: Methylation values from benign prostate tissue from patient 3.
</p>
</li>
<li>
<p>Benign_Patient_4: Methylation values from benign prostate tissue from patient 4.
</p>
</li>
<li>
<p>Tumour_Patient_1: Methylation values from tumor prostate tissue from patient 1.
</p>
</li>
<li>
<p>Tumour_Patient_2: Methylation values from tumor prostate tissue from patient 2.
</p>
</li>
<li>
<p>Tumour_Patient_3: Methylation values from tumor prostate tissue from patient 3.
</p>
</li>
<li>
<p>Tumour_Patient_4: Methylation values from tumor prostate tissue from patient 4.
</p>
</li>
</ul>
<h3>Details</h3>

<p>The raw methylation array data was first quality controlled and preprocessed using the <a href="https://rnbeads.org/">RnBeads</a> package. The array data were then normalized and and probes located outside of CpG sites and on the sex chromosome were filtered out. The CpG sites with missing values were removed from the resulting dataset. A subset of the complete dataset has been uploaded in the package for testing purposes. The complete dataset is available on <a href="https://github.com/koyelucd/betaclust">GitHub</a>.
</p>


<h3>References</h3>

<p>Mueller F, Scherer M, Assenov Y, Lutsik P, Walter J, Lengauer T, Bock C (2019). “RnBeads 2.0: comprehensive analysis of DNA methylation data.” Genome Biology, 20(55).
</p>
<p>Assenov Y, Mueller F, Lutsik P, Walter J, Lengauer T, Bock C (2014). “Comprehensive Analysis of DNA Methylation Data with RnBeads.” Nature Methods, 11(11), 1138–1140.
</p>


<h3>See Also</h3>

<p><code>legacy.data</code>
</p>


</div>