<div class="container">

<table style="width: 100%;"><tr>
<td>subset_variants</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Remove variants with no data for pathogenicity</h2>

<h3>Description</h3>

<p>Subset an allele count matrix given a minimum allele count threshold for pathogenicity per individual so that only variants for which data relevant to pathogencity are retained. This is useful to apply before running <code>bevimed</code> as it reduces the size of the parameter space used in the inference.
</p>


<h3>Usage</h3>

<pre><code class="language-R">subset_variants(G, min_ac = 1L, return_variants = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>Integer matrix of variant counts per individual, one row per individual and one column per variant.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_ac</code></td>
<td>
<p>Integer vector with a length equalling the number of individuals or length <code>1</code> (in which case the given value is used for all individuals) giving the minimum number of alleles at pathogenic variant sites each individual requires in order to classify as having a ‘pathogenic allele configuration’. Thus, this parameter encodes the mode of inheritance. For instance, setting this parameter to <code>1</code> corresponds to dominant inheritance. If there are differences in ploidy between individuals in the locus, it is necessary to set it on an sample level basis - e.g. to ensure sex is accounted for if the locus lies on the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_variants</code></td>
<td>
<p>Logical value determining whether to return an integer vector of indices of retained variants or the subsetted allele count matrix</p>
</td>
</tr>
</table>
</div>