<div class="container">

<table style="width: 100%;"><tr>
<td>BeviMed-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Bayesian Evaluation of Variant Involvement in Mendelian Disease 
</h2>

<h3>Description</h3>

<p>A fast integrative genetic association test for rare diseases.
</p>


<h3>Details</h3>

<p>BeviMed estimates a probability of association between a case/control label and allele counts at rare variant sites in a genomic locus and also, given that there is an association, the probabilities that each variant is involved in the disease. It does so by estimating the evidence for a model where the case/control label is independent of the allele configurations, and a model in which the probability of the case/control label depends on the corresponding allele configuration and a latent partition of variants into pathogenic and non-pathogenic groups.
</p>


<h3>Author(s)</h3>

<p>Daniel Greene.
</p>
<p>Maintainer: Daniel Greene &lt;dg333@cam.ac.uk&gt;
</p>


<h3>References</h3>

<p>Greene et al., A Fast Association Test for Identifying Pathogenic Variants Involved in Rare Diseases, The American Journal of Human Genetics (2017), http://dx.doi.org/10.1016/j.ajhg.2017.05.015.
</p>


<h3>See Also</h3>

<p><code>bevimed</code>
</p>


</div>