<div class="container">

<table style="width: 100%;"><tr>
<td>prob_association_m</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate probability of association for one mode of inheritance</h2>

<h3>Description</h3>

<p>Equivalent to <code>prob_association</code> where the prior probability of one mode of inheritance is 1. This function is faster, as it only calls <code>bevimed_m</code> once.
</p>


<h3>Usage</h3>

<pre><code class="language-R">prob_association_m(y, min_ac = 1L, prior_prob_association = 0.01, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>Logical vector of case (<code>TRUE</code>) control (<code>FALSE</code>) status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_ac</code></td>
<td>
<p>Integer vector with a length equalling the number of individuals or length <code>1</code> (in which case the given value is used for all individuals) giving the minimum number of alleles at pathogenic variant sites each individual requires in order to classify as having a ‘pathogenic allele configuration’. Thus, this parameter encodes the mode of inheritance. For instance, setting this parameter to <code>1</code> corresponds to dominant inheritance. If there are differences in ploidy between individuals in the locus, it is necessary to set it on an sample level basis - e.g. to ensure sex is accounted for if the locus lies on the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prior_prob_association</code></td>
<td>
<p>The prior probability of association.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Other arguments to pass to <code>log_BF</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Probability value.
</p>


<h3>See Also</h3>

<p><code>log_BF</code>, <code>prob_association</code>, <code>bevimed_m</code>
</p>


</div>