<div class="container">

<table style="width: 100%;"><tr>
<td>call_cpp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>R interface to BeviMed c++ MCMC procedure</h2>

<h3>Description</h3>

<p>Allows other functions in the package to call the c++ function passing arguments more succinctly and by name.
</p>


<h3>Usage</h3>

<pre><code class="language-R">call_cpp(
  samples_per_chain,
  y,
  block_starts,
  block_ends,
  cases,
  counts,
  min_ac,
  tau_shape,
  pi_shape,
  omega_shape,
  temperatures,
  z0_matrix,
  estimate_omega,
  logit_omegas,
  logit_omega_proposal_sds,
  variant_weights,
  estimate_phi,
  log_phis,
  log_phi_mean,
  log_phi_sd,
  log_phi_proposal_sds,
  chain_swaps_per_cycle,
  annealing,
  tandem_variant_updates,
  comphet_variant_block_starts,
  comphet_variant_block_ends,
  comphet_variants,
  return_z_trace,
  return_x_trace,
  vec_sums = FALSE,
  burn = 0,
  check = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>samples_per_chain</code></td>
<td>
<p>Number of samples to draw from each chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>Logical vector of subject affectedness status.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block_starts</code></td>
<td>
<p>Integer vector of k 0-indexed start positions (with respect to <code>cases</code> and <code>counts</code>) for contiguous blocks relating to the k variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block_ends</code></td>
<td>
<p>Integer vector of (exclusive) k 0-indexed end positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cases</code></td>
<td>
<p>0 based vector of case indices with respect to y.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>counts</code></td>
<td>
<p>Vector of variant counts.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_ac</code></td>
<td>
<p>Integer vector with a length equalling the number of individuals or length <code>1</code> (in which case the given value is used for all individuals) giving the minimum number of alleles at pathogenic variant sites each individual requires in order to classify as having a ‘pathogenic allele configuration’. Thus, this parameter encodes the mode of inheritance. For instance, setting this parameter to <code>1</code> corresponds to dominant inheritance. If there are differences in ploidy between individuals in the locus, it is necessary to set it on an sample level basis - e.g. to ensure sex is accounted for if the locus lies on the X chromosome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau_shape</code></td>
<td>
<p>Beta distribution parameterisation of benign variant configuration rate of affection, q.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pi_shape</code></td>
<td>
<p>Beta distribution parameterisation of pathogenic variant configuration rate of affection, p.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omega_shape</code></td>
<td>
<p>Beta distribution of global rate of pathogenicty of variants in gene given pathogenicity of gene, omega.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>temperatures</code></td>
<td>
<p>Numeric vector of temperatures of power posteriors. One chain will be created for each element of the vector at the corresponding temperature.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>z0_matrix</code></td>
<td>
<p>Matrix of logicals, where the rows are used as an initial zs for the chains.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_omega</code></td>
<td>
<p>Logical value determining whether to estimate the parameter omega.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logit_omegas</code></td>
<td>
<p>Numeric vector of logit omega values, one value per chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>logit_omega_proposal_sds</code></td>
<td>
<p>Numeric vector of proposal standard deviations for Metropolis-Hastings sampling of logit omega parameter, one value per chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant_weights</code></td>
<td>
<p>Vector of log-odds off-sets for rates of pathogenicity of individual variants relative to the global rate, omega.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>estimate_phi</code></td>
<td>
<p>Logical value determining whether to estimate a scaling factor of <code>variant_weights</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_phis</code></td>
<td>
<p>Numeric vector of log phi values, one value per chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_phi_mean</code></td>
<td>
<p>Mean for normal prior on scaling factor phi.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_phi_sd</code></td>
<td>
<p>SD for normal prior on scaling factor phi.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>log_phi_proposal_sds</code></td>
<td>
<p>Numeric vector of proposal standard deviations for Metropolis-Hastings sampling of log phi parameter, one value per chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain_swaps_per_cycle</code></td>
<td>
<p>Number of chain swaps to propose per update cycle.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annealing</code></td>
<td>
<p>Logical value determining whether to anneal the chains, e.g. for optimisation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tandem_variant_updates</code></td>
<td>
<p>Number of tandem variant updates to make per update cycle.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comphet_variant_block_starts</code></td>
<td>
<p>0-indexed start positions for contiguous blocks of variants in <code>comphet_variants</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comphet_variant_block_ends</code></td>
<td>
<p>As <code>comphet_variant_block_starts</code> for (exclusive) stop positions.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>comphet_variants</code></td>
<td>
<p>Integer vector giving variant numbers (0-based, i.e. between 0 and k-1). Used to pick pairs of variants for tandem updates from.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_z_trace</code></td>
<td>
<p>Logical value determining whether to store the z-vectors for each chain, which uses alot of memory, particularly if <code>samples_per_chain</code>, k and <code>length(temperatures)</code> are large.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_x_trace</code></td>
<td>
<p>Logical value determining whether to store the x variable determined by success samples of z. Potentially uses alot of memory, particularly if <code>samples_per_chain</code>, k and <code>length(temperatures)</code> are large.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vec_sums</code></td>
<td>
<p>Logical value determining whether to calculate vector summary statistics.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>burn</code></td>
<td>
<p>Number of samples to drop from the start of the chain.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>check</code></td>
<td>
<p>Logical value indicating whether to perform validation on the arguments before calling the c++ function.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Object of class <code>BeviMed_raw</code>, containing the output of the MCMC sampling.
</p>


</div>