<div class="container">

<table style="width: 100%;"><tr>
<td>readRAW</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Creates a BGData Object From a .raw File or a .ped-Like File</h2>

<h3>Description</h3>

<p>Creates a <code>BGData</code> object from a .raw file (generated with
<code>--recodeA</code> in <a href="https://www.cog-genomics.org/plink2">PLINK</a>).
Other text-based file formats are supported as well by tweaking some of the
parameters as long as the records of individuals are in rows, and
phenotypes, covariates and markers are in columns.
</p>


<h3>Usage</h3>

<pre><code class="language-R">readRAW(fileIn, header = TRUE, dataType = integer(), n = NULL,
  p = NULL, sep = "", na.strings = "NA", nColSkip = 6L,
  idCol = c(1L, 2L), nNodes = NULL, linked.by = "rows",
  folderOut = paste0("BGData_", sub("\\.[[:alnum:]]+$", "",
  basename(fileIn))), outputType = "byte", dimorder = if (linked.by ==
  "rows") 2L:1L else 1L:2L, verbose = FALSE)

readRAW_matrix(fileIn, header = TRUE, dataType = integer(), n = NULL,
  p = NULL, sep = "", na.strings = "NA", nColSkip = 6L,
  idCol = c(1L, 2L), verbose = FALSE)

readRAW_big.matrix(fileIn, header = TRUE, dataType = integer(),
  n = NULL, p = NULL, sep = "", na.strings = "NA", nColSkip = 6L,
  idCol = c(1L, 2L), folderOut = paste0("BGData_",
  sub("\\.[[:alnum:]]+$", "", basename(fileIn))), outputType = "char",
  verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>fileIn</code></td>
<td>

<p>The path to the plaintext file.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>header</code></td>
<td>

<p>Whether <code>fileIn</code> contains a header. Defaults to <code>TRUE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dataType</code></td>
<td>

<p>The coding type of genotypes in <code>fileIn</code>. Use <code>integer()</code> or
<code>double()</code> for numeric coding. Alpha-numeric coding is currently
not supported for <code>readRAW</code> and <code>readRAW_big.matrix</code>: use the
<code>--recodeA</code> option of PLINK to convert the .ped file into a .raw
file. Defaults to <code>integer()</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>The number of individuals. Auto-detect if <code>NULL</code>. Defaults to
<code>NULL</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>

<p>The number of markers. Auto-detect if <code>NULL</code>. Defaults to
<code>NULL</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sep</code></td>
<td>

<p>The field separator character. Values on each line of the file are
separated by this character. If <code>sep = ""</code> (the default for
<code>readRAW</code> the separator is "white space", that is one or more
spaces, tabs, newlines or carriage returns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.strings</code></td>
<td>

<p>The character string used in the plaintext file to denote missing
value. Defaults to <code>NA</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nColSkip</code></td>
<td>

<p>The number of columns to be skipped to reach the genotype information
in the file. Defaults to <code>6</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>idCol</code></td>
<td>

<p>The index of the ID column. If more than one index is given, both
columns will be concatenated with "_". Defaults to <code>c(1, 2)</code>, i.e.
a concatenation of the first two columns.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nNodes</code></td>
<td>

<p>The number of nodes to create. Auto-detect if <code>NULL</code>. Defaults to
<code>NULL</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>linked.by</code></td>
<td>

<p>If <code>columns</code> a column-linked matrix (<code>ColumnLinkedMatrix</code>) is
created, if <code>rows</code> a row-linked matrix (<code>RowLinkedMatrix</code>).
Defaults to <code>rows</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>folderOut</code></td>
<td>

<p>The path to the folder where to save the binary files. Defaults to the
name of the input file (<code>fileIn</code>) without extension prefixed with
"BGData_".
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outputType</code></td>
<td>

<p>The <code>vmode</code> for <code>ff</code> and <code>type</code> for <code>big.matrix</code>
objects. Default to <code>byte</code> for <code>ff</code> and <code>char</code> for
<code>big.matrix</code> objects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dimorder</code></td>
<td>

<p>The physical layout of the underlying <code>ff</code> object of each node.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Whether progress updates will be posted. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The data included in the first couple of columns (up to <code>nColSkip</code>) is
used to populate the sample information of a <code>BGData</code> object, and the
remaining columns are used to fill the genotypes. If the first row contains
a header (<code>header = TRUE</code>), data in this row is used to determine the
column names for sample information and genotypes.
</p>
<p>The genotypes can take several forms, depending on the function that is
called (<code>readRAW</code>, <code>readRAW_matrix</code>, or
<code>readRAW_big.matrix</code>). The following sections illustrate each function
in detail.
</p>


<h3>readRAW</h3>

<p>Genotypes are stored in a <code>LinkedMatrix</code> object where each node is an
<code>ff</code> instance. Multiple <code>ff</code> files are used because the array
size in <code>ff</code> is limited to the largest integer which can be
represented on the system (<code>.Machine$integer.max</code>) and for genetic
data this limitation is often exceeded. The <code>LinkedMatrix</code> package
makes it possible to link several <code>ff</code> files together by columns or by
rows and treat them similarly to a single matrix. By default a
<code>ColumnLinkedMatrix</code> is used for the genotypes, but the user can
modify this using the <code>linked.by</code> argument. The number of nodes to
generate is either specified by the user using the <code>nNodes</code> argument
or determined internally so that each <code>ff</code> object has a number of
cells that is smaller than <code>.Machine$integer.max / 1.2</code>. A folder (see
<code>folderOut</code>) that contains the binary flat files (named
<code>geno_*.bin</code>) and an external representation of the <code>BGData</code>
object in <code>BGData.RData</code> is created.
</p>


<h3>readRAW_matrix</h3>

<p>Genotypes are stored in a regular <code>matrix</code> object. Therefore, this
function will only work if the .raw file is small enough to fit into
memory.
</p>


<h3>readRAW_big.matrix</h3>

<p>Genotypes are stored in a filebacked <code>big.matrix</code> object. A folder
(see <code>folderOut</code>) that contains the binary flat file (named
<code>BGData.bin</code>), a descriptor file (named <code>BGData.desc</code>), and an
external representation of the <code>BGData</code> object in <code>BGData.RData</code>
are created.
</p>


<h3>Reloading a BGData object</h3>

<p>To reload a <code>BGData</code> object, it is recommended to use the
<code>load.BGData</code> function instead of the <code>load</code> function as
<code>load</code> does not initialize <code>ff</code> objects or attach
<code>big.matrix</code> objects.
</p>


<h3>See Also</h3>

<p><code>load.BGData()</code> to load a previously saved
<code>BGData</code> object, <code>as.BGData()</code> to create
<code>BGData</code> objects from non-text files (e.g. .bed files).
<code>BGData-class</code>,
<code>ColumnLinkedMatrix-class</code>,
<code>RowLinkedMatrix-class</code>,
<code>big.matrix-class</code>, and <code>ff</code> for
more information on the above mentioned classes.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Path to example data
path &lt;- system.file("extdata", package = "BGData")

# Convert RAW files of chromosome 1 to a BGData object
bg &lt;- readRAW(fileIn = paste0(path, "/chr1.raw"))

unlink("BGData_chr1", recursive = TRUE)
</code></pre>


</div>