<div class="container">

<table style="width: 100%;"><tr>
<td>segments</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Find non-overlapping segments based on a summary statistic</h2>

<h3>Description</h3>

<p>Given a summary statistic and a threshold, this function computes the
number of non-overlapping segments, each defined as a discovery (i.e.,
<code>statistic[i] &lt;= threshold)</code> +/- a gap, in the same units as <code>bp</code>
(often base-pair position).
</p>


<h3>Usage</h3>

<pre><code class="language-R">segments(statistic, chr, bp, threshold, gap, trim = FALSE, verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>statistic</code></td>
<td>

<p>A statistic (e.g., BFDR or p-values).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>

<p>A vector containing the chromosome for each value of <code>statistic</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bp</code></td>
<td>

<p>A vector containing the base-pair positions for each value of
<code>statistic</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>threshold</code></td>
<td>

<p>The threshold to determine 'significance' (e.g., <code>1e-5</code> for
p-values).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>

<p>1/2 of the length of the desired segments.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>trim</code></td>
<td>

<p>Whether to collapse segments that were artifically inflated by
<code>gap</code>. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Whether progress updates will be posted. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A data frame containing the following information:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>chr</code></td>
<td>

<p>Chromosome
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>start</code></td>
<td>

<p>Index where segment starts within <code>statistic</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>end</code></td>
<td>

<p>Index where segment ends within <code>statistic</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>length</code></td>
<td>

<p>Length of segment.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bpStart</code></td>
<td>

<p>Base-pair position where segment starts.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bpEnd</code></td>
<td>

<p>Base-pair position where segment ends.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bpLength</code></td>
<td>

<p>Length of segment in base-pair positions.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minValue</code></td>
<td>

<p>Smallest value of <code>statistic</code> within segment.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minValuePos</code></td>
<td>

<p>Position of variant with the smallest value of <code>statistic</code> within
segment.
</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">library(BGData)

# Load example data
bg &lt;- BGData:::loadExample()

# Perform GWAS
pValues &lt;- GWAS(
    formula = FT10 ~ 1,
    data = bg,
    method = "rayOLS"
)

# Determine segments within +/- 1MB from a significant variant
segments &lt;- segments(
    statistic = pValues[, 4],
    chr = map(bg)$chromosome,
    bp = map(bg)$base_pair_position,
    threshold = 1e-5,
    gap = 1e6,
    trim = FALSE,
    verbose = FALSE
)
</code></pre>


</div>