<div class="container">

<table style="width: 100%;"><tr>
<td>BGData-class</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Container for Phenotype and Genotype Data</h2>

<h3>Description</h3>

<p>The BGData class is a container for genotypes, sample information, and
variant information. The class is inspired by the <code>.bed/.fam/.bim</code>
(binary) and <code>.ped/.fam/.map</code> (text) phenotype/genotype file formats
of <a href="https://www.cog-genomics.org/plink2">PLINK</a>. It is used by several
functions of this package such as <code>GWAS</code> for performing a Genome Wide
Association Study or <code>getG</code> for calculating a genomic relationship
matrix.
</p>


<h3>Details</h3>

<p>There are several ways to create an instance of this class:
</p>

<ul>
<li>
<p> from arbitrary phenotype/genotype data using the <code>BGData</code>
constructor function.
</p>
</li>
<li>
<p> from a .bed file using <code>as.BGData</code> and <code>BEDMatrix</code>.
</p>
</li>
<li>
<p> from a previously saved <code>BGData</code> object using
<code>load.BGData</code>.
</p>
</li>
<li>
<p> from multiple files (even a mixture of different file types)
using <code>LinkedMatrix</code>.
</p>
</li>
<li>
<p> from a .raw file (or a .ped-like file) using
<code>readRAW</code>, <code>readRAW_matrix</code>, or
<code>readRAW_big.matrix</code>.
</p>
</li>
</ul>
<p>A .ped file can be recoded to a .raw file in
<a href="https://www.cog-genomics.org/plink2">PLINK</a> using <code>plink --file
    myfile --recodeA</code>, or converted to a .bed file using <code>plink --file
    myfile --make-bed</code>. Conversely, a .bed file can be transformed back to a
.ped file using <code>plink --bfile myfile --recode</code> or to a .raw file
using <code>plink --bfile myfile --recodeA</code> without losing information.
</p>


<h3>Accessors</h3>

<p>In the following code snippets, <code>x</code> is a BGData object.
</p>

<dl>
<dt>
<code>geno(x)</code>, <code>geno(x) &lt;- value</code>:</dt>
<dd>
<p>Get or set genotypes.
</p>
</dd>
<dt>
<code>pheno(x)</code>, <code>pheno(x) &lt;- value</code>:</dt>
<dd>
<p>Get or set sample information.
</p>
</dd>
<dt>
<code>map(x)</code>, <code>map(x) &lt;- value</code>:</dt>
<dd>
<p>Get or set variant information.
</p>
</dd>
</dl>
<h3>See Also</h3>

<p><code>BGData</code>, <code>as.BGData</code>, <code>load.BGData</code>,
<code>readRAW</code> to create <code>BGData</code> objects.
</p>
<p><code>LinkedMatrix-class</code> and
<code>BEDMatrix-class</code> for more information on the above
mentioned classes.
</p>


<h3>Examples</h3>

<pre><code class="language-R">X &lt;- matrix(data = rnorm(100), nrow = 10, ncol = 10)
Y &lt;- data.frame(y = runif(10))
MAP &lt;- data.frame(means = colMeans(X), freqNA = colMeans(is.na(X)))
DATA &lt;- BGData(geno = X, pheno = Y, map = MAP)

dim(geno(DATA))
head(pheno(DATA))
head(map(DATA))
</code></pre>


</div>