<div class="container">

<table style="width: 100%;"><tr>
<td>GWAS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Performs Single Marker Regressions Using BGData Objects</h2>

<h3>Description</h3>

<p>Implements single marker regressions. The regression model includes all the
covariates specified in the right-hand-side of the <code>formula</code> plus one
column of the genotypes at a time. The data from the association tests is
obtained from a <code>BGData</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GWAS(formula, data, method = "lsfit", i = seq_len(nrow(geno(data))),
  j = seq_len(ncol(geno(data))), chunkSize = 5000L,
  nCores = getOption("mc.cores", 2L), verbose = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>formula</code></td>
<td>

<p>The formula for the GWAS model without the variant, e.g. <code>y ~ 1</code>
or <code>y ~ factor(sex) + age</code>. The variables included in the formula
must be column names in the sample information of the <code>BGData</code>
object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>

<p>A <code>BGData</code> object.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>The regression method to be used. Currently, the following methods are
implemented: <code>rayOLS</code> (see below), <code>lsfit</code>, <code>lm</code>,
<code>lm.fit</code>, <code>glm</code>, <code>lmer</code>, and <code>SKAT</code>. Defaults to
<code>lsfit</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>i</code></td>
<td>

<p>Indicates which rows of the genotypes should be used. Can be integer,
boolean, or character. By default, all rows are used.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>j</code></td>
<td>

<p>Indicates which columns of the genotypes should be used. Can be
integer, boolean, or character. By default, all columns are used.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chunkSize</code></td>
<td>

<p>The number of columns of the genotypes that are brought into physical
memory for processing per core. If <code>NULL</code>, all elements in
<code>j</code> are used. Defaults to 5000.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nCores</code></td>
<td>

<p>The number of cores (passed to <code>mclapply</code>). Defaults to the number
of cores as detected by <code>detectCores</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>

<p>Whether progress updates will be posted. Defaults to <code>FALSE</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments for chunkedApply and regression method.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>rayOLS</code> method is a regression through the origin that can only
be used with a <code>y ~ 1</code> formula, i.e. it only allows for one
quantitative response variable <code>y</code> and one variant at a time as an
explanatory variable (the variant is not included in the formula, hence
<code>1</code> is used as a dummy). If covariates are needed, consider
preadjustment of <code>y</code>. Among the provided methods, it is by far the
fastest.
</p>
<p>Some regression methods may require the data to not contain columns with
variance 0 or too many missing values. We suggest running <code>summarize</code>
to detect variants that do not clear the desired minor-allele frequency and
rate of missing genotype calls, and filtering these variants out using the
<code>j</code> parameter of the <code>GWAS</code> function (see example below).
</p>


<h3>Value</h3>

<p>The same matrix that would be returned by <code>coef(summary(model))</code>.
</p>


<h3>See Also</h3>

<p><code>file-backed-matrices</code> for more information on file-backed
matrices. <code>multi-level-parallelism</code> for more information on
multi-level parallelism. <code>BGData-class</code> for more information on
the <code>BGData</code> class. <code>lsfit</code>,
<code>lm</code>, <code>lm.fit</code>,
<code>glm</code>, <code>lmer</code>, and
<code>SKAT</code> for more information on regression methods.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Restrict number of cores to 1 on Windows
if (.Platform$OS.type == "windows") {
    options(mc.cores = 1)
}

# Load example data
bg &lt;- BGData:::loadExample()

# Detect variants that do not pass MAF and missingness thresholds
summaries &lt;- summarize(geno(bg))
maf &lt;- ifelse(summaries$allele_freq &gt; 0.5, 1 - summaries$allele_freq,
    summaries$allele_freq)
exclusions &lt;- maf &lt; 0.01 | summaries$freq_na &gt; 0.05

# Perform a single marker regression
res1 &lt;- GWAS(formula = FT10 ~ 1, data = bg, j = !exclusions)

# Draw a Manhattan plot
plot(-log10(res1[, 4]))

# Use lm instead of lsfit (the default)
res2 &lt;- GWAS(formula = FT10 ~ 1, data = bg, method = "lm", j = !exclusions)

# Use glm instead of lsfit (the default)
y &lt;- pheno(bg)$FT10
pheno(bg)$FT10.01 &lt;- y &gt; quantile(y, 0.8, na.rm = TRUE)
res3 &lt;- GWAS(formula = FT10.01 ~ 1, data = bg, method = "glm", j = !exclusions)

# Perform a single marker regression on the first 50 markers (useful for
# distributed computing)
res4 &lt;- GWAS(formula = FT10 ~ 1, data = bg, j = 1:50)
</code></pre>


</div>