<div class="container">

<table style="width: 100%;"><tr>
<td>BGData-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>A Suite of Packages for Analysis of Big Genomic Data</h2>

<h3>Description</h3>

<p>Modern genomic datasets are big (large <em>n</em>), high-dimensional (large
<em>p</em>), and multi-layered. The challenges that need to be addressed are
memory requirements and computational demands. Our goal is to develop
software that will enable researchers to carry out analyses with big
genomic data within the R environment.
</p>


<h3>Details</h3>

<p>We have identified several approaches to tackle those challenges within R:
</p>

<ul>
<li>
<p> File-backed matrices: The data is stored in on the hard drive and
users can read in smaller chunks when they are needed.
</p>
</li>
<li>
<p> Linked arrays: For very large datasets a single file-backed array
may not be enough or convenient. A linked array is an array whose
content is distributed over multiple file-backed nodes.
</p>
</li>
<li>
<p> Multiple dispatch: Methods are presented to users so that they
can treat these arrays pretty much as if they were RAM arrays.
</p>
</li>
<li>
<p> Multi-level parallelism: Exploit multi-core and multi-node
computing.
</p>
</li>
<li>
<p> Inputs: Users can create these arrays from standard formats
(e.g., PLINK .bed).
</p>
</li>
</ul>
<p>The <code>BGData</code> package is an umbrella package that comprises several
packages: <code>BEDMatrix</code>, <code>LinkedMatrix</code>, and <code>symDMatrix</code>. It
features scalable and efficient computational methods for large genomic
datasets such as genome-wide association studies (GWAS) or genomic
relationship matrices (G matrix). It also contains a container class called
<code>BGData</code> that holds genotypes, sample information, and variant
information.
</p>


<h3>Example dataset</h3>

<p>The <code>extdata</code> folder contains example files that were generated from
the 250k SNP and phenotype data in
<a href="https://www.nature.com/articles/nature08800">Atwell et al. (2010)</a>.
Only the first 300 SNPs of chromosome 1, 2, and 3 were included to keep the
size of the example dataset small.
<a href="https://www.cog-genomics.org/plink2">PLINK</a> was used to convert the
data to <a href="https://www.cog-genomics.org/plink2/input#bed">.bed</a> and
<a href="https://www.cog-genomics.org/plink2/input#raw">.raw</a> files.
<code>FT10</code> has been chosen as a phenotype and is provided as an
<a href="https://www.cog-genomics.org/plink2/input#pheno">alternate phenotype
file</a>. The file is intentionally shuffled to demonstrate that the
additional phenotypes are put in the same order as the rest of the
phenotypes.
</p>


<h3>See Also</h3>

<p><code>BEDMatrix-package</code>,
<code>LinkedMatrix-package</code>, and
<code>symDMatrix-package</code> for an introduction to the
respective packages.
</p>
<p><code>file-backed-matrices</code> for more information on file-backed
matrices. <code>multi-level-parallelism</code> for more information on
multi-level parallelism.
</p>


</div>