<div class="container">

<table style="width: 100%;"><tr>
<td>read_ped</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>read_ped</h2>

<h3>Description</h3>

<p>This function reads genotype information stored in  PED format used in plink.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  read_ped(ped_file)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>ped_file</code></td>
<td>
<p>ASCII file with genotype information.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The PED file is a white-space (space or tab) delimited file: the first six columns are mandatory:
</p>
<p>Family ID
Individual ID
Paternal ID
Maternal ID
Sex (1=male; 2=female; other=unknown)
Phenotype
</p>
<p>The IDs are alphanumeric: the combination of family and individual ID should uniquely identify a person. 
A PED file must have 1 and only 1 phenotype in the sixth column. The phenotype can be either a quantitative 
trait or an affection status column.
</p>


<h3>Value</h3>

<p>The routine will return a vector of dimension n*p (n=number of individuals, p=number of snps), with the snps stacked.
</p>
<p>The vector contains integer codes:
</p>

<table>
<tr>
<td style="text-align: left;">
   Integer code </td>
<td style="text-align: left;"> Genotype </td>
</tr>
<tr>
<td style="text-align: left;">
   0 </td>
<td style="text-align: left;"> 00 Homozygote "1"/"1" </td>
</tr>
<tr>
<td style="text-align: left;">
   1 </td>
<td style="text-align: left;"> 01 Heterozygote </td>
</tr>
<tr>
<td style="text-align: left;">
   2 </td>
<td style="text-align: left;"> 10 Missing genotype </td>
</tr>
<tr>
<td style="text-align: left;">
   3 </td>
<td style="text-align: left;"> 11 Homozygote "2"/"2" </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Gustavo de los Campos, Paulino Perez Rodriguez,
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 

library(BGLR)
demo(read_ped)


## End(Not run)

</code></pre>


</div>