<div class="container">

<table style="width: 100%;"><tr>
<td>read_bed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>read_bed</h2>

<h3>Description</h3>

<p>This function reads genotype information stored in  binary PED (BED) files used in plink.
These files save space and time. The pedigree/phenotype information is stored in a separate file (*.fam) 
and the map information is stored in an extededed MAP file (*.bim) that contains information about the allele names, 
which would otherwise be lost in the BED file. More details <a href="http://zzz.bwh.harvard.edu/plink/binary.shtml">http://zzz.bwh.harvard.edu/plink/binary.shtml</a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  read_bed(bed_file,bim_file,fam_file,na.strings,verbose)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bed_file</code></td>
<td>
<p>binary file with genotype information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bim_file</code></td>
<td>
<p>text file with pedigree/phenotype information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fam_file</code></td>
<td>
<p>text file with extended map information.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.strings</code></td>
<td>
<p>missing value indicators, default=c("0","-9").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>logical, if true print hex dump of bed file.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The routine will return a vector of dimension n*p (n=number of individuals, p=number of snps), with the snps(individuals) stacked, 
depending whether the BED file is in SNP-major or individual-major mode.
</p>
<p>The vector contains integer codes:
</p>

<table>
<tr>
<td style="text-align: left;">
   Integer code </td>
<td style="text-align: left;"> Genotype </td>
</tr>
<tr>
<td style="text-align: left;">
   0 </td>
<td style="text-align: left;"> 00 Homozygote "1"/"1" </td>
</tr>
<tr>
<td style="text-align: left;">
   1 </td>
<td style="text-align: left;"> 01 Heterozygote </td>
</tr>
<tr>
<td style="text-align: left;">
   2 </td>
<td style="text-align: left;"> 10 Missing genotype </td>
</tr>
<tr>
<td style="text-align: left;">
   3 </td>
<td style="text-align: left;"> 11 Homozygote "2"/"2" </td>
</tr>
<tr>
<td style="text-align: left;">
 </td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Gustavo de los Campos, Paulino Perez Rodriguez,
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 

library(BGLR)
demo(read_bed)


## End(Not run)

</code></pre>


</div>