<div class="container">

<table style="width: 100%;"><tr>
<td>calculateRelatedness</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Calculate Relatedness Coefficient</h2>

<h3>Description</h3>

<p>This function calculates the relatedness coefficient between two individuals based on their shared ancestry, as described by Wright (1922).
</p>


<h3>Usage</h3>

<pre><code class="language-R">calculateRelatedness(
  generations = 2,
  path = NULL,
  full = TRUE,
  maternal = FALSE,
  empirical = FALSE,
  segregating = TRUE,
  total_a = 6800 * 1e+06,
  total_m = 16500,
  weight_a = 1,
  weight_m = 1,
  denom_m = FALSE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>generations</code></td>
<td>
<p>Number of generations back of common ancestors the pair share.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>Traditional method to count common ancestry, which is twice the number of generations removed from common ancestors. If not provided, it is calculated as 2*generations.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full</code></td>
<td>
<p>Logical. Indicates if the kin share both parents at the common ancestor's generation. Default is TRUE.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maternal</code></td>
<td>
<p>Logical. Indicates if the maternal lineage should be considered in the calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>empirical</code></td>
<td>
<p>Logical. Adjusts the coefficient based on empirical data, using the total number of nucleotides and other parameters.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segregating</code></td>
<td>
<p>Logical. Adjusts for segregating genes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total_a</code></td>
<td>
<p>Numeric. Represents the total size of the autosomal genome in terms of nucleotides, used in empirical adjustment. Default is 6800*1000000.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>total_m</code></td>
<td>
<p>Numeric. Represents the total size of the mitochondrial genome in terms of nucleotides, used in empirical adjustment. Default is 16500.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight_a</code></td>
<td>
<p>Numeric. Represents the weight of phenotypic influence from additive genetic variance, used in empirical adjustment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>weight_m</code></td>
<td>
<p>Numeric. Represents the weight of phenotypic influence from mitochondrial effects, used in empirical adjustment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>denom_m</code></td>
<td>
<p>Logical. Indicates if 'total_m' and 'weight_m' should be included in the denominator of the empirical adjustment calculation.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Further named arguments that may be passed to another function.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The relatedness coefficient between two people (b &amp; c) is defined in relation to their common ancestors:
<code class="reqn">r_{bc} = \sum \left(\frac{1}{2}\right)^{n+n'+1} (1+f_a)</code>
</p>


<h3>Value</h3>

<p>Relatedness Coefficient ('coef'): A measure of the genetic relationship between two individuals.
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# For full siblings, the relatedness coefficient is expected to be 0.5:
calculateRelatedness(generations = 1, full = TRUE)
# For half siblings, the relatedness coefficient is expected to be 0.25:
calculateRelatedness(generations = 1, full = FALSE)

## End(Not run)
</code></pre>


</div>