<div class="container">

<table style="width: 100%;"><tr>
<td>DEprobs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Signed probabilities of differential expression
</h2>

<h3>Description</h3>

<p>The <code>DEprobs</code> function is an application of mixture modeling to differential gene expression analysis. The function takes as input a two- or three-component model of one-dimensional gene expression data. The data is assumed to represent log fold change expression values and be negative when the corresponding genes are down-regulated. The function calculates probabilities of differential expression for the data and gives them a sign according to the sign of the data.
</p>


<h3>Usage</h3>

<pre><code class="language-R">DEprobs(model, verbose=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>model</code></td>
<td>
<p>an object of the class <code>mModel</code>,   </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>indicates whether log messeges should be prited out.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Given the input <code>model</code>, the function identifies the component which corresponds to the differentially expressed genes as the one which looks differential according to the posterior probabilities. 
</p>
<p>For input models with two Gaussian components the differential component should be the one with a broader range (encompassing the other), or the one with higher deviation from 0 (we assume the data are centered around 0).
</p>
<p>For input models with three Gaussian components there are two differential components: one corresponding to the down-regulated genes, and one corresponding to the up-regulated genes. Those components are identified as the ones with the lowest and the highest mean, respectively.
</p>
<p>For <code>verbose=TRUE</code> the index of the differential component is printed out. 
</p>


<h3>Value</h3>

<p>An list with the following elements:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>diff.p.X</code></td>
<td>
<p>a vector with the calculated signed differential expression probabilities for the unlabeled observations in the dataset <code>model$X</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diff.p.knowns</code></td>
<td>
<p> a vector with the calculated signed differential expression probabilities for the unlabeled observations in the dataset <code>model$knowns</code>. For <code>model$knowns=NULL</code> <code>diff.p.knowns</code> is also NULL(null).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>diff.c</code></td>
<td>
<p>the index (or two indexes, in case of a three-component input model) of the identified differential component.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Przemyslaw Biecek
</p>


<h3>References</h3>

<p>Przemyslaw Biecek, Ewa Szczurek, Martin Vingron, Jerzy Tiuryn (2012), The R Package bgmm: Mixture Modeling with Uncertain Knowledge, Journal of Statistical Software.
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(Ste12)
X = Ste12Data[ match(names(Ste12Data), rownames(Ste12Beliefs), nomatch = 0) ==0 ]
knowns = Ste12Data[rownames(Ste12Beliefs)]
model = belief(X=X, knowns=knowns, B=Ste12Beliefs)
dep=DEprobs(model)
str(dep)
</code></pre>


</div>