<div class="container">

<table style="width: 100%;"><tr>
<td>A.wrapper</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Amino Acid Wrapper</h2>

<h3>Description</h3>

<p>Wrapper function for amino acid analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">A.wrapper(
  loci,
  loci.ColNames,
  genos,
  grp,
  Exon,
  EPL,
  Cores,
  Strict.Bin,
  Output,
  Verbose
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>loci</code></td>
<td>
<p>Loci being analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>loci.ColNames</code></td>
<td>
<p>The column names of the loci being analyzed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genos</code></td>
<td>
<p>Genotype table.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp</code></td>
<td>
<p>Case/Control or Phenotype groupings.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Exon</code></td>
<td>
<p>Exon(s)for targeted analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EPL</code></td>
<td>
<p>Protein Alignment List.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Cores</code></td>
<td>
<p>Number of cores to use for analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Output</code></td>
<td>
<p>Data return carryover from main BIGDAWG function</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Verbose</code></td>
<td>
<p>Summary display carryover from main BIGDAWG function</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>This function is for internal BIGDAWG use only.
</p>


</div>