<div class="container">

<table style="width: 100%;"><tr>
<td>BIGDAWG</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BIGDAWG Main Wrapper Function</h2>

<h3>Description</h3>

<p>This is the main wrapper function for each analysis.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BIGDAWG(
  Data,
  HLA = TRUE,
  Run.Tests,
  Loci.Set,
  Exon,
  All.Pairwise = FALSE,
  Trim = FALSE,
  Res = 2,
  EVS.rm = FALSE,
  Missing = 2,
  Strict.Bin = FALSE,
  Cores.Lim = 1L,
  Results.Dir,
  Return = FALSE,
  Output = TRUE,
  Merge.Output = FALSE,
  Verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Data</code></td>
<td>
<p>Name of the genotype data file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HLA</code></td>
<td>
<p>Logical Indicating whether data is HLA class I/II genotyping data only.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Run.Tests</code></td>
<td>
<p>Specifics which tests to run.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Loci.Set</code></td>
<td>
<p>Input list defining which loci to use for analyses (combinations permitted).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Exon</code></td>
<td>
<p>Numeric Exon(s) for targeted amino acid analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>All.Pairwise</code></td>
<td>
<p>Logical indicating whether all pairwise loci should be analyzed in haplotype analysis.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Trim</code></td>
<td>
<p>Logical indicating if HLA alleles should be trimmed to a set resolution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Res</code></td>
<td>
<p>Numeric setting what desired resolution to trim HLA alleles.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>EVS.rm</code></td>
<td>
<p>Logical indicating if expression variant suffixes should be removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Missing</code></td>
<td>
<p>Numeric setting allowable missing data for running analysis (may use "ignore").</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Strict.Bin</code></td>
<td>
<p>Logical specify if strict rare cell binning should be used in ChiSq test.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Cores.Lim</code></td>
<td>
<p>Integer setting the number of cores accessible to BIGDAWG (Windows limit is 1 core).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Results.Dir</code></td>
<td>
<p>Optional, string of full path directory name for BIGDAWG output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Return</code></td>
<td>
<p>Logical Should analysis results be returned as list.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Output</code></td>
<td>
<p>Logical Should analysis results be written to output directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Merge.Output</code></td>
<td>
<p>Logical Should analysis results be merged into a single file for easy access.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Verbose</code></td>
<td>
<p>Logical Should a summary of each analysis be displayed in console.</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
### The following examples use the synthetic data set bundled with BIGDAWG

# Haplotype analysis with no missing genotypes for two loci sets
# Significant haplotype association with phenotype
# BIGDAWG(Data="HLA_data", Run.Tests="H", Missing=0, Loci.Set=list(c("DRB1","DQB1")))

# Hardy-Weinberg and Locus analysis ignoring missing data
# Significant locus associations with phenotype at all but DQB1
# BIGDAWG(Data="HLA_data", Run.Tests="L", Missing="ignore")

# Hardy-Weinberg analysis trimming data to 2-Field resolution with no output to files (console only)
# Significant locus deviation at DQB1
BIGDAWG(Data="HLA_data", Run.Tests="HWE", Trim=TRUE, Res=2, Output=FALSE)

## End(Not run)
</code></pre>


</div>