<div class="container">

<table style="width: 100%;"><tr>
<td>GLSconvert</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genotype List String Conversion</h2>

<h3>Description</h3>

<p>Main Workhorse wrapper for cross converting columnar table to GL string representaion.
</p>


<h3>Usage</h3>

<pre><code class="language-R">GLSconvert(
  Data,
  Convert,
  File.Output = "txt",
  System = "HLA",
  HZY.Red = FALSE,
  DRB345.Check = FALSE,
  Strip.Prefix = TRUE,
  Abs.Fill = FALSE,
  Cores.Lim = 1L
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Data</code></td>
<td>
<p>String File name or R Data Frame.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Convert</code></td>
<td>
<p>String Direction for conversion.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>File.Output</code></td>
<td>
<p>String Type of File.Output.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>System</code></td>
<td>
<p>String Genetic system (HLA or KIR) of the data being converted</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>HZY.Red</code></td>
<td>
<p>Logical Reduction of homozygote genotypes to single allele.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>DRB345.Check</code></td>
<td>
<p>Logical Check DR haplotypes for consistency and flag unusual haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Strip.Prefix</code></td>
<td>
<p>Logical Should System/Locus prefixes be stripped from table data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Abs.Fill</code></td>
<td>
<p>Logical Should absent loci special designations be used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Cores.Lim</code></td>
<td>
<p>Integer How many cores can be used.</p>
</td>
</tr>
</table>
</div>