<div class="container">

<table style="width: 100%;"><tr>
<td>colon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Gene expression data from colon-cancer patients</h2>

<h3>Description</h3>

<p>The data file contains gene expression data of 62 samples (40 tumor samples,
22 normal samples) from colon-cancer patients analyzed with an Affymetrix
oligonucleotide Hum6000 array.
</p>


<h3>Format</h3>

<p>A list of 2 variables included in <code>colon</code>: </p>

<ul>
<li>
<p><code>X</code>: a 62-by-2000 matrix that records the gene expression data.
Used as design matrix.
</p>
</li>
<li>
<p><code>y</code>: a binary vector of length 62 recording the sample status: 1 =
tumor; 0 = normal. Used as response vector.
</p>
</li>
</ul>
<h3>Source</h3>

<p>The raw data can be found on Bioconductor:
<a href="https://bioconductor.org/packages/release/data/experiment/html/colonCA.html">https://bioconductor.org/packages/release/data/experiment/html/colonCA.html</a>.
</p>


<h3>References</h3>

 <ul><li>
<p> U. Alon et al. (1999): Broad patterns of gene
expression revealed by clustering analysis of tumor and normal colon tissue
probed by oligonucleotide arrays. <em>Proc. Natl. Acad. Sci. USA</em>
<strong>96</strong>, 6745-6750. <a href="https://www.pnas.org/doi/abs/10.1073/pnas.96.12.6745">https://www.pnas.org/doi/abs/10.1073/pnas.96.12.6745</a>.
</p>
</li></ul>
<h3>Examples</h3>

<pre><code class="language-R">data(colon)
X &lt;- colon$X
y &lt;- colon$y
str(X)
dim(X)
X.bm &lt;- as.big.matrix(X, backingfile = "") # convert to big.matrix object
str(X.bm)
dim(X.bm)
</code></pre>


</div>