<div class="container">

<table style="width: 100%;"><tr>
<td>bigQF-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Quadratic Forms in Large Matrices
</h2>

<h3>Description</h3>

<p>A computationally-efficient leading-eigenvalue approximation to tail probabilities and quantiles of large quadratic forms, in particular for the Sequence Kernel Association Test (SKAT) used in genomics &lt;doi:10.1002/gepi.22136&gt;. Also provides stochastic singular value decomposition for dense or sparse matrices.
</p>


<h3>Details</h3>

<p>The DESCRIPTION file:
</p>

<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> bigQF</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Title: </td>
<td style="text-align: left;"> Quadratic Forms in Large Matrices</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 1.6</td>
</tr>
<tr>
<td style="text-align: left;">
Author: </td>
<td style="text-align: left;"> Thomas Lumley</td>
</tr>
<tr>
<td style="text-align: left;">
Maintainer: </td>
<td style="text-align: left;"> Thomas Lumley &lt;t.lumley@auckland.ac.nz&gt;</td>
</tr>
<tr>
<td style="text-align: left;">
Description: </td>
<td style="text-align: left;"> A computationally-efficient leading-eigenvalue approximation to tail probabilities and quantiles of large quadratic forms, in particular for the Sequence Kernel Association Test (SKAT) used in genomics &lt;doi:10.1002/gepi.22136&gt;. Also provides stochastic singular value decomposition for dense or sparse matrices.</td>
</tr>
<tr>
<td style="text-align: left;">
URL: </td>
<td style="text-align: left;"> https://github.com/tslumley/bigQF</td>
</tr>
<tr>
<td style="text-align: left;">
Imports: </td>
<td style="text-align: left;"> svd, CompQuadForm, Matrix, stats, coxme</td>
</tr>
<tr>
<td style="text-align: left;">
Suggests: </td>
<td style="text-align: left;"> knitr, rmarkdown, SKAT</td>
</tr>
<tr>
<td style="text-align: left;">
VignetteBuilder: </td>
<td style="text-align: left;"> knitr</td>
</tr>
<tr>
<td style="text-align: left;">
Depends: </td>
<td style="text-align: left;"> methods</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL-2</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Index of help topics:
</p>
<pre>
SKAT.example            Data example from SKAT package
SKAT.matrixfree         Make 'matrix-free' object for SKAT test
bigQF-package           Quadratic Forms in Large Matrices
famSKAT                 Implicit matrix for family-based SKAT test
pQF                     Tail probabilities for quadratic forms
seigen                  Stochastic singular value decomposition
seqMetaExample          Example data, from seqMeta package
sequence                Simulated human DNA variant sequence
sparse.matrixfree       Make 'matrix-free' object from (sparse) Matrix
</pre>
<p>This package computes tail probabilities for large quadratic forms, with
the motivation being the SKAT test used in DNA sequence association
studies.
</p>
<p>The true distribution is a linear combination of 1-df chi-squared
distributions, where the coefficients are the non-zero eigenvalues of
the matrix <code>A</code> defining the quadratic form <code class="reqn">z^TAz</code>.  The package uses an
approximation to the distribution consisting of the largest <code>neig</code> terms in the
linear combination plus the Satterthwaite approximation to the rest of
the linear combination. 
</p>
<p>The main function is <code>pQF</code>, which has options for how to
compute the leading eigenvalues (Lanczos-type algorithm or stochastic
SVD) and how to compute the linear combination (inverting the
characteristic function or a saddlepoint approximation).  The Lanczos
algorithm is from the <code>svd</code> package; the stochastic SVD can be
called directly via <code>ssvd</code> or <code>seigen</code>
</p>
<p>Given a square matrix, <code>pQF</code> uses it as <code>A</code>. If the input is a
non-square matrix <code>M</code>, then <code>A</code> is <code>crossprod(M)</code>. The
function can also be used matrix-free, given an object containing
functions to compute the product and transpose-product by <code>M</code>. This last option
is described in the <code>"matrix-free"</code> vignette. The matrix-free
algorithm also uses a randomised estimator to estimate
the trace of <code>crossprod(A)</code>. The function <code>sparse.matrixfree</code> constructs a object for
matrix-free use of <code>pQF</code> from a sparse Matrix object. The
algorithms are described in the Lumley et al (2018) reference.
</p>
<p>Finally, there are functions specifically for the SKAT family of genomic
tests. These take a genotype matrix and an adjustment model as arguments
and produce an object that contains the test statistic in its
<code>Q</code> component and which can be used as an argument to <code>pQF</code> to
extract p-values: <code>SKAT.matrixfree</code> and <code>famSKAT</code>. The
vignette <code>"Checking pQF vs SKAT"</code> compares <code>SKAT.matrixfree</code>
to the <code>SKAT</code> package and illustrates how it can be used
</p>


<h3>Author(s)</h3>

<p>Thomas Lumley
</p>
<p>Maintainer: Thomas Lumley &lt;t.lumley@auckland.ac.nz&gt;
</p>


<h3>References</h3>

<p>Tong Chen, Thomas Lumley (2019) Numerical evaluation of methods approximating the distribution of a large quadratic form in normal variables. Computational Statistics &amp; Data Analysis.  139: 75-81,
</p>
<p>Lumley et al. (2018) Sequence kernel association tests for large sets of markers: tail probabilities for large quadratic forms. Genet Epidemiol
. 2018 Sep;42(6):516-527. doi: 10.1002/gepi.22136
</p>
<p>Nathan Halko, Per-Gunnar Martinsson, Joel A. Tropp (2010) Finding structure with randomness: Probabilistic algorithms for constructing approximate matrix decompositions. <a href="https://arxiv.org/abs/0909.4061">https://arxiv.org/abs/0909.4061</a>.
</p>
<p>Lee, S., with contributions from Larisa Miropolsky, and Wu, M. (2015). SKAT: SNP-Set (Sequence) Kernel Association Test. R package version 1.1.2.
</p>
<p>Lee, S., Wu, M. C., Cai, T., Li, Y., Boehnke, M., and Lin,
X. (2011). Rare-variant association testing for sequencing data with the
sequence kernel association test. American Journal of Human Genetics,
89:82-93.
</p>


</div>