<div class="container">

<table style="width: 100%;"><tr>
<td>snp_prodBGEN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BGEN matrix product</h2>

<h3>Description</h3>

<p>Compute a matrix product between BGEN files and a matrix. This removes the
need to read an intermediate FBM object with <code>snp_readBGEN()</code> to compute the
product. Moreover, when using dosages, they are not rounded to two decimal
places anymore.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_prodBGEN(
  bgenfiles,
  beta,
  list_snp_id,
  ind_row = NULL,
  bgi_dir = dirname(bgenfiles),
  read_as = c("dosage", "random"),
  block_size = 1000,
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bgenfiles</code></td>
<td>
<p>Character vector of paths to files with extension ".bgen".
The corresponding ".bgen.bgi" index files must exist.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>beta</code></td>
<td>
<p>A matrix (or a vector), with rows corresponding to <code>list_snp_id</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list_snp_id</code></td>
<td>
<p>List of character vectors of SNP IDs to read, with one
vector per BGEN file. Each SNP ID should be in the form
<code>"&lt;chr&gt;_&lt;pos&gt;_&lt;a1&gt;_&lt;a2&gt;"</code> (e.g. <code>"1_88169_C_T"</code> or <code>"01_88169_C_T"</code>).
If you have one BGEN file only, just wrap your vector of IDs with <code>list()</code>.
<strong>This function assumes that these IDs are uniquely identifying variants.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind_row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used. <strong>Don't use negative indices.</strong>
You can access the sample IDs corresponding to the genotypes from the <em>.sample</em>
file, and use e.g. <code>match()</code> to get indices corresponding to the ones you want.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bgi_dir</code></td>
<td>
<p>Directory of index files. Default is the same as <code>bgenfiles</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_as</code></td>
<td>
<p>How to read BGEN probabilities? Currently implemented:
</p>

<ul>
<li>
<p> as dosages (rounded to two decimal places), the default,
</p>
</li>
<li>
<p> as hard calls, randomly sampled based on those probabilities
(similar to PLINK option '<code style="white-space: pre;">⁠--hard-call-threshold random⁠</code>').
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block_size</code></td>
<td>
<p>Maximum size of temporary blocks (in number of variants).
Default is <code>1000</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The product <code>bgen_data[ind_row, 'list_snp_id'] %*% beta</code>.
</p>


<h3>See Also</h3>

<p><code>snp_readBGEN()</code>
</p>


</div>