<div class="container">

<table style="width: 100%;"><tr>
<td>snp_asGeneticPos</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Interpolate to genetic positions</h2>

<h3>Description</h3>

<p>Use genetic maps available at
<a href="https://github.com/joepickrell/1000-genomes-genetic-maps/">https://github.com/joepickrell/1000-genomes-genetic-maps/</a>
to interpolate physical positions (in bp) to genetic positions (in cM).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_asGeneticPos(
  infos.chr,
  infos.pos,
  dir = tempdir(),
  ncores = 1,
  rsid = NULL,
  type = c("OMNI", "hapmap")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>infos.chr</code></td>
<td>
<p>Vector of integers specifying each SNP's chromosome.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$chromosome⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.pos</code></td>
<td>
<p>Vector of integers specifying the physical position
on a chromosome (in base pairs) of each SNP.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$physical.pos⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>Directory where to download and decompress files.
Default is <code>tempdir()</code>. Directly use <em>uncompressed</em> files there if already
present. You can use <code>R.utils::gunzip()</code> to uncompress local files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rsid</code></td>
<td>
<p>If providing rsIDs, the matching is performed using those
(instead of positions) and variants not matched are interpolated using
spline interpolation of variants that have been matched.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Whether to use the genetic maps interpolated from "OMNI"
(the default), or from "hapmap".</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The new vector of genetic positions.
</p>


</div>