<div class="container">

<table style="width: 100%;"><tr>
<td>snp_readBed</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read PLINK files into a "bigSNP"</h2>

<h3>Description</h3>

<p>Functions to read bed/bim/fam files into a bigSNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_readBed(bedfile, backingfile = sub_bed(bedfile))

snp_readBed2(
  bedfile,
  backingfile = sub_bed(bedfile),
  ind.row = rows_along(obj.bed),
  ind.col = cols_along(obj.bed),
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bedfile</code></td>
<td>
<p>Path to file with extension ".bed" to read.
You need the corresponding ".bim" and ".fam" in the same directory.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>backingfile</code></td>
<td>
<p>The path (without extension) for the backing files
for the cache of the bigSNP object. Default takes the bedfile
without the ".bed" extension.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.col</code></td>
<td>
<p>An optional vector of the column indices (SNPs) that are used.
If not specified, all columns are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For more information on these formats, please visit
<a href="https://www.cog-genomics.org/plink/1.9/formats#bed">PLINK webpage</a>.
For other formats, please use PLINK to convert them in bedfiles,
which require minimal space to store and are faster to read. For example,
to convert from a VCF file, use the <code>--vcf</code> option. See snp_plinkQC.
</p>


<h3>Value</h3>

<p>The path to the RDS file that stores the <code>bigSNP</code> object.
Note that this function creates one other file which stores the values of
the Filebacked Big Matrix.<br><strong>You shouldn't read from PLINK files more than once.</strong> Instead, use
snp_attach to load the "bigSNP" object in any R session from backing files.
</p>


<h3>Examples</h3>

<pre><code class="language-R">(bedfile &lt;- system.file("extdata", "example.bed", package = "bigsnpr"))

# Reading the bedfile and storing the data in temporary directory
rds &lt;- snp_readBed(bedfile, backingfile = tempfile())

# Loading the data from backing files
test &lt;- snp_attach(rds)

str(test)
dim(G &lt;- test$genotypes)
G[1:8, 1:8]
</code></pre>


</div>