<div class="container">

<table style="width: 100%;"><tr>
<td>snp_cor</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Correlation matrix</h2>

<h3>Description</h3>

<p>Get significant (Pearson) correlations between nearby SNPs of the same chromosome
(p-values are computed using a two-sided t-test).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_cor(
  Gna,
  ind.row = rows_along(Gna),
  ind.col = cols_along(Gna),
  size = 500,
  alpha = 1,
  thr_r2 = 0,
  fill.diag = TRUE,
  infos.pos = NULL,
  ncores = 1
)

bed_cor(
  obj.bed,
  ind.row = rows_along(obj.bed),
  ind.col = cols_along(obj.bed),
  size = 500,
  alpha = 1,
  thr_r2 = 0,
  fill.diag = TRUE,
  infos.pos = NULL,
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>Gna</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br>
You can have missing values in these data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.col</code></td>
<td>
<p>An optional vector of the column indices (SNPs) that are used.
If not specified, all columns are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>For one SNP, window size around this SNP to compute correlations.
Default is <code>500</code>. If not providing <code>infos.pos</code> (<code>NULL</code>, the default), this is
a window in number of SNPs, otherwise it is a window in kb (genetic distance).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Type-I error for testing correlations.
Default is <code>1</code> (no threshold is applied).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>thr_r2</code></td>
<td>
<p>Threshold to apply on squared correlations. Default is <code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fill.diag</code></td>
<td>
<p>Whether to fill the diagonal with 1s (the default)
or to keep it as 0s.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.pos</code></td>
<td>
<p>Vector of integers specifying the physical position
on a chromosome (in base pairs) of each SNP.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$physical.pos⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.bed</code></td>
<td>
<p>Object of type bed, which is the mapping of some bed file.
Use <code>obj.bed &lt;- bed(bedfile)</code> to get this object.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The (Pearson) correlation matrix. This is a sparse symmetric matrix.
</p>


<h3>Examples</h3>

<pre><code class="language-R">test &lt;- snp_attachExtdata()
G &lt;- test$genotypes

corr &lt;- snp_cor(G, ind.col = 1:1000)
corr[1:10, 1:10]

# Sparsity
length(corr@x) / length(corr)

</code></pre>


</div>