<div class="container">

<table style="width: 100%;"><tr>
<td>download_genetic_map</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Download a genetic map</h2>

<h3>Description</h3>

<p>This function uses linear interpolation, whereas <code>snp_asGeneticPos()</code> uses
nearest neighbors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">download_genetic_map(
  type = c("hg19_OMNI", "hg19_hapmap", "hg38_price"),
  dir,
  ncores = 1
)

snp_asGeneticPos2(infos.chr, infos.pos, genetic_map)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>Which genetic map to download.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dir</code></td>
<td>
<p>Directory where to download and decompress files.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.chr</code></td>
<td>
<p>Vector of integers specifying each SNP's chromosome.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$chromosome⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.pos</code></td>
<td>
<p>Vector of integers specifying the physical position
on a chromosome (in base pairs) of each SNP.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$physical.pos⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic_map</code></td>
<td>
<p>A data frame with 3 columns: <code>chr</code>, <code>pos</code>, and <code>pos_cM</code>.
You can get it using <code>download_genetic_map()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The hg19 genetic maps are downloaded from
<a href="https://github.com/joepickrell/1000-genomes-genetic-maps/">https://github.com/joepickrell/1000-genomes-genetic-maps/</a>
while the hg38 one is downloaded from
<code style="white-space: pre;">⁠https://alkesgroup.broadinstitute.org/Eagle/downloads/tables/⁠</code>.
</p>


<h3>Value</h3>

<p>A data frame with 3 columns: <code>chr</code>, <code>pos</code>, and <code>pos_cM</code>.
</p>
<p>The new vector of genetic positions.
</p>


</div>