<div class="container">

<table style="width: 100%;"><tr>
<td>snp_MAF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>MAF</h2>

<h3>Description</h3>

<p>Minor Allele Frequency.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_MAF(
  G,
  ind.row = rows_along(G),
  ind.col = cols_along(G),
  nploidy = 2,
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br><strong>You shouldn't have missing values.</strong> Also, remember to do quality control,
e.g. some algorithms in this package won't work if you use SNPs with 0 MAF.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.col</code></td>
<td>
<p>An optional vector of the column indices (SNPs) that are used.
If not specified, all columns are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nploidy</code></td>
<td>
<p>Number of trials, parameter of the binomial distribution.
Default is <code>2</code>, which corresponds to diploidy, such as for the human genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector of MAFs, corresponding to <code>ind.col</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">obj.bigsnp &lt;- snp_attachExtdata()
str(maf &lt;- snp_MAF(obj.bigsnp$genotypes))

</code></pre>


</div>