<div class="container">

<table style="width: 100%;"><tr>
<td>snp_plinkQC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Quality Control</h2>

<h3>Description</h3>

<p>Quality Control (QC) and possible conversion to <em>bed</em>/<em>bim</em>/<em>fam</em> files
using <a href="https://www.cog-genomics.org/plink2"><strong>PLINK 1.9</strong></a>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_plinkQC(
  plink.path,
  prefix.in,
  file.type = "--bfile",
  prefix.out = paste0(prefix.in, "_QC"),
  maf = 0.01,
  geno = 0.1,
  mind = 0.1,
  hwe = 1e-50,
  autosome.only = FALSE,
  extra.options = "",
  verbose = TRUE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>plink.path</code></td>
<td>
<p>Path to the executable of PLINK 1.9.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix.in</code></td>
<td>
<p>Prefix (path without extension) of the dataset to be QCed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file.type</code></td>
<td>
<p>Type of the dataset to be QCed. Default is <code>"--bfile"</code> and
corresponds to bed/bim/fam files. You can also use <code>"--file"</code> for ped/map
files, <code>"--vcf"</code> for a VCF file, or <code>"--gzvcf"</code> for a gzipped VCF.
More information can be found at
<a href="https://www.cog-genomics.org/plink/1.9/input">https://www.cog-genomics.org/plink/1.9/input</a>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix.out</code></td>
<td>
<p>Prefix (path without extension) of the bed/bim/fam dataset
to be created. Default is created by appending <code>"_QC"</code> to <code>prefix.in</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf</code></td>
<td>
<p>Minimum Minor Allele Frequency (MAF) for a SNP to be kept.
Default is <code>0.01</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>geno</code></td>
<td>
<p>Maximum proportion of missing values for a SNP to be kept.
Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mind</code></td>
<td>
<p>Maximum proportion of missing values for a sample to be kept.
Default is <code>0.1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hwe</code></td>
<td>
<p>Filters out all variants which have Hardy-Weinberg equilibrium
exact test p-value below the provided threshold. Default is <code>1e-50</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>autosome.only</code></td>
<td>
<p>Whether to exclude all unplaced and non-autosomal
variants? Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra.options</code></td>
<td>
<p>Other options to be passed to PLINK as a string. More
options can be found at <a href="https://www.cog-genomics.org/plink2/filter">https://www.cog-genomics.org/plink2/filter</a>.
If using PLINK 2.0, you could e.g. use <code>"--king-cutoff 0.0884"</code> to remove
some related samples at the same time of quality controls.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Whether to show PLINK log? Default is <code>TRUE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>The path of the newly created bedfile.
</p>


<h3>References</h3>

<p>Chang, Christopher C, Carson C Chow, Laurent CAM Tellier,
Shashaank Vattikuti, Shaun M Purcell, and James J Lee. 2015.
<em>Second-generation PLINK: rising to the challenge of larger and richer
datasets.</em> GigaScience 4 (1): 7. <a href="https://doi.org/10.1186/s13742-015-0047-8">doi:10.1186/s13742-015-0047-8</a>.
</p>


<h3>See Also</h3>

<p>download_plink snp_plinkIBDQC
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

bedfile &lt;- system.file("extdata", "example.bed", package = "bigsnpr")
prefix  &lt;- sub_bed(bedfile)
plink &lt;- download_plink()
test &lt;- snp_plinkQC(plink.path = plink,
                    prefix.in = prefix,
                    prefix.out = tempfile(),
                    file.type = "--bfile",  # the default (for ".bed")
                    maf = 0.05,
                    geno = 0.05,
                    mind = 0.05,
                    hwe = 1e-10,
                    autosome.only = TRUE)
test

## End(Not run)

</code></pre>


</div>