<div class="container">

<table style="width: 100%;"><tr>
<td>snp_readBGEN</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read BGEN files into a "bigSNP"</h2>

<h3>Description</h3>

<p>Function to read the UK Biobank BGEN files into a bigSNP.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_readBGEN(
  bgenfiles,
  backingfile,
  list_snp_id,
  ind_row = NULL,
  bgi_dir = dirname(bgenfiles),
  read_as = c("dosage", "random"),
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bgenfiles</code></td>
<td>
<p>Character vector of paths to files with extension ".bgen".
The corresponding ".bgen.bgi" index files must exist.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>backingfile</code></td>
<td>
<p>The path (without extension) for the backing files (".bk"
and ".rds") that are created by this function for storing the
bigSNP object.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>list_snp_id</code></td>
<td>
<p>List of character vectors of SNP IDs to read, with one
vector per BGEN file. Each SNP ID should be in the form
<code>"&lt;chr&gt;_&lt;pos&gt;_&lt;a1&gt;_&lt;a2&gt;"</code> (e.g. <code>"1_88169_C_T"</code> or <code>"01_88169_C_T"</code>).
If you have one BGEN file only, just wrap your vector of IDs with <code>list()</code>.
<strong>This function assumes that these IDs are uniquely identifying variants.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind_row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used. <strong>Don't use negative indices.</strong>
You can access the sample IDs corresponding to the genotypes from the <em>.sample</em>
file, and use e.g. <code>match()</code> to get indices corresponding to the ones you want.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bgi_dir</code></td>
<td>
<p>Directory of index files. Default is the same as <code>bgenfiles</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>read_as</code></td>
<td>
<p>How to read BGEN probabilities? Currently implemented:
</p>

<ul>
<li>
<p> as dosages (rounded to two decimal places), the default,
</p>
</li>
<li>
<p> as hard calls, randomly sampled based on those probabilities
(similar to PLINK option '<code style="white-space: pre;">⁠--hard-call-threshold random⁠</code>').
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>For more information on this format, please visit
<a href="https://code.enkre.net/bgen">BGEN webpage</a>.
</p>
<p>This function is designed to read UK Biobank imputation files. This assumes
that variants have been compressed with zlib, that there are only 2 possible
alleles, and that each probability is stored on 8 bits. For example, if you
use <em>qctool</em> to generate your own BGEN files, please make sure you are using
options '<code style="white-space: pre;">⁠-ofiletype bgen_v1.2 -bgen-bits 8 -assume-chromosome⁠</code>'.
</p>
<p>If the format is not the expected one, this will result in an error or even
a crash of your R session. Another common source of error is due to corrupted
files; e.g. if using UK Biobank files, compare the result of <code>tools::md5sum()</code>
with the ones at <a href="https://biobank.ndph.ox.ac.uk/ukb/refer.cgi?id=998">https://biobank.ndph.ox.ac.uk/ukb/refer.cgi?id=998</a>.
</p>
<p>You can look at some example code from my papers on how to use this function:
</p>

<ul>
<li> <p><a href="https://github.com/privefl/paper-infer/blob/main/code/prepare-geno-simu.R">https://github.com/privefl/paper-infer/blob/main/code/prepare-geno-simu.R</a>
</p>
</li>
<li> <p><a href="https://github.com/privefl/paper-misspec/blob/main/code/prepare-genotypes.R">https://github.com/privefl/paper-misspec/blob/main/code/prepare-genotypes.R</a>
</p>
</li>
</ul>
<h3>Value</h3>

<p>The path to the RDS file <code style="white-space: pre;">⁠&lt;backingfile&gt;.rds⁠</code> that stores the <code>bigSNP</code>
object created by this function.<br>
Note that this function creates another file (<em>.bk</em>) which stores the values
of the FBM (<code style="white-space: pre;">⁠$genotypes⁠</code>). The rows corresponds to the order of <code>ind_row</code>;
the columns to the order of <code>list_snp_id</code>. The <code style="white-space: pre;">⁠$map⁠</code> component of the
<code>bigSNP</code> object stores some information on the variants (including allele
frequencies and INFO scores computed from the imputation probabilities).
However, it does not have a <code style="white-space: pre;">⁠$fam⁠</code> component; you should use the individual
IDs in the <em>.sample</em> file (filtered with <code>ind_row</code>) to add external
information on the individuals.<br><strong>You shouldn't read from BGEN files more than once.</strong> Instead, use
snp_attach to load the "bigSNP" object in any R session from backing files.
</p>


</div>