<div class="container">

<table style="width: 100%;"><tr>
<td>snp_fst</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Fixation index (Fst)</h2>

<h3>Description</h3>

<p>Fixation index (Fst), either per variant, or genome-wide
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_fst(list_df_af, min_maf = 0, overall = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>list_df_af</code></td>
<td>
<p>List of data frames with <code style="white-space: pre;">⁠$af⁠</code> (allele frequency per variant)
and <code style="white-space: pre;">⁠$N⁠</code> (sample size per variant). Typically, the outputs of <code>bed_MAF()</code>.
Each new data frame of the list should correspond to a different population.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_maf</code></td>
<td>
<p>Minimum MAF threshold (for the average of populations) to be
included in the final results. Default is <code>0</code> (remove monomorphic variants).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>overall</code></td>
<td>
<p>Whether to compute Fst genome-wide (<code>TRUE</code>) or per variant
(<code>FALSE</code>, the default).</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>If <code>overall</code>, then one value, otherwise a value for each variant
with missing values for the variants not passing <code>min_maf</code>.
This should be equivalent to using '<code>--fst --within</code>' in PLINK.
</p>


<h3>References</h3>

<p>Weir, B. S., &amp; Cockerham, C. C. (1984). Estimating F-statistics for the
analysis of population structure. Evolution, 1358-1370.
</p>


<h3>Examples</h3>

<pre><code class="language-R">bedfile &lt;- system.file("extdata", "example.bed", package = "bigsnpr")
obj.bed &lt;- bed(bedfile)

pop &lt;- rep(1:3, c(143, 167, 207))
ind_pop &lt;- split(seq_along(pop), pop)
list_df_af &lt;- lapply(ind_pop, function(ind) bed_MAF(obj.bed, ind.row = ind))

snp_fst(list_df_af)
snp_fst(list_df_af[c(1, 2)], overall = TRUE)
snp_fst(list_df_af[c(1, 3)], overall = TRUE)
snp_fst(list_df_af[c(3, 2)], overall = TRUE)

</code></pre>


</div>