<div class="container">

<table style="width: 100%;"><tr>
<td>snp_pcadapt</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Outlier detection</h2>

<h3>Description</h3>

<p>Method to detect genetic markers involved in biological adaptation.
This provides a statistical tool for outlier detection based on
Principal Component Analysis. This corresponds to the statistic based
on mahalanobis distance, as implemented in package <strong>pcadapt</strong>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_pcadapt(
  G,
  U.row,
  ind.row = rows_along(G),
  ind.col = cols_along(G),
  ncores = 1
)

bed_pcadapt(
  obj.bed,
  U.row,
  ind.row = rows_along(obj.bed),
  ind.col = cols_along(obj.bed),
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br><strong>You shouldn't have missing values.</strong> Also, remember to do quality control,
e.g. some algorithms in this package won't work if you use SNPs with 0 MAF.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>U.row</code></td>
<td>
<p>Left singular vectors (not scores, <code class="reqn">U^T U = I</code>)
corresponding to <code>ind.row</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.col</code></td>
<td>
<p>An optional vector of the column indices (SNPs) that are used.
If not specified, all columns are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.bed</code></td>
<td>
<p>Object of type bed, which is the mapping of some bed file.
Use <code>obj.bed &lt;- bed(bedfile)</code> to get this object.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An object of classes <code>mhtest</code> and <code>data.frame</code> returning one
score by SNP. See <code>methods(class = "mhtest")</code>.
</p>


<h3>References</h3>

<p>Luu, K., Bazin, E., &amp; Blum, M. G. (2017).
pcadapt: an R package to perform genome scans for selection
based on principal component analysis.
Molecular ecology resources, 17(1), 67-77.
</p>


<h3>See Also</h3>

<p>snp_manhattan, snp_qq and snp_gc.
</p>


<h3>Examples</h3>

<pre><code class="language-R">test &lt;- snp_attachExtdata()
G &lt;- test$genotypes
obj.svd &lt;- big_SVD(G, fun.scaling = snp_scaleBinom(), k = 10)
plot(obj.svd) # there seems to be 3 "significant" components
pcadapt &lt;- snp_pcadapt(G, obj.svd$u[, 1:3])
snp_qq(pcadapt)

</code></pre>


</div>