<div class="container">

<table style="width: 100%;"><tr>
<td>snp_qq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Q-Q plot</h2>

<h3>Description</h3>

<p>Creates a quantile-quantile plot from p-values from a GWAS study.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_qq(gwas, lambdaGC = TRUE, coeff = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>gwas</code></td>
<td>
<p>A <code>mhtest</code> object with the p-values associated with each SNP.
Typically, the output of bigstatsr::big_univLinReg, bigstatsr::big_univLogReg
or snp_pcadapt.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lambdaGC</code></td>
<td>
<p>Add the Genomic Control coefficient as subtitle to the plot?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coeff</code></td>
<td>
<p>Relative size of text. Default is <code>1</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>ggplot2</code> object. You can plot it using the <code>print</code> method.
You can modify it as you wish by adding layers. You might want to read
<a href="https://r4ds.had.co.nz/data-visualisation.html">this chapter</a>
to get more familiar with the package <strong>ggplot2</strong>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">set.seed(9)

test &lt;- snp_attachExtdata()
G &lt;- test$genotypes
y &lt;- rnorm(nrow(G))

gwas &lt;- big_univLinReg(G, y)

snp_qq(gwas)
gwas_gc &lt;- snp_gc(gwas) # this modifies `attr(gwas_gc, "transfo")`
snp_qq(gwas_gc)

# The next plot should be prettier with a real dataset
snp_manhattan(gwas_gc,
              infos.chr = test$map$chromosome,
              infos.pos = test$map$physical.pos) +
  ggplot2::geom_hline(yintercept = -log10(5e-8), linetype = 2, color = "red")

p &lt;- snp_qq(gwas_gc) +
  ggplot2::aes(text = asPlotlyText(test$map)) +
  ggplot2::labs(subtitle = NULL, x = "Expected -log10(p)", y = "Observed -log10(p)")
## Not run: plotly::ggplotly(p, tooltip = "text")
</code></pre>


</div>