<div class="container">

<table style="width: 100%;"><tr>
<td>bigsnpr-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>bigsnpr: Analysis of Massive SNP Arrays</h2>

<h3>Description</h3>

<p>Easy-to-use, efficient, flexible and scalable tools for analyzing massive SNP arrays. Privé et al. (2018) <a href="https://doi.org/10.1093/bioinformatics/bty185">doi:10.1093/bioinformatics/bty185</a>.
</p>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br><strong>You shouldn't have missing values.</strong> Also, remember to do quality control,
e.g. some algorithms in this package won't work if you use SNPs with 0 MAF.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Gna</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br>
You can have missing values in these data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>A bigSNP.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.chr</code></td>
<td>
<p>Vector of integers specifying each SNP's chromosome.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$chromosome⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>infos.pos</code></td>
<td>
<p>Vector of integers specifying the physical position
on a chromosome (in base pairs) of each SNP.<br>
Typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$map$physical.pos⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nploidy</code></td>
<td>
<p>Number of trials, parameter of the binomial distribution.
Default is <code>2</code>, which corresponds to diploidy, such as for the human genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.col</code></td>
<td>
<p>An optional vector of the column indices (SNPs) that are used.
If not specified, all columns are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>is.size.in.bp</code></td>
<td>
<p>Deprecated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.bed</code></td>
<td>
<p>Object of type bed, which is the mapping of some bed file.
Use <code>obj.bed &lt;- bed(bedfile)</code> to get this object.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p><strong>Maintainer</strong>: Florian Privé <a href="mailto:florian.prive.21@gmail.com">florian.prive.21@gmail.com</a>
</p>
<p>Other contributors:
</p>

<ul>
<li>
<p> Michael Blum [thesis advisor]
</p>
</li>
<li>
<p> Hugues Aschard <a href="mailto:hugues.aschard@pasteur.fr">hugues.aschard@pasteur.fr</a> [thesis advisor]
</p>
</li>
<li>
<p> Bjarni Jóhann Vilhjálmsson <a href="mailto:bjv@econ.au.dk">bjv@econ.au.dk</a> [thesis advisor]
</p>
</li>
</ul>
<h3>See Also</h3>

<p>Useful links:
</p>

<ul>
<li> <p><a href="https://privefl.github.io/bigsnpr/">https://privefl.github.io/bigsnpr/</a>
</p>
</li>
<li>
<p> Report bugs at <a href="https://github.com/privefl/bigsnpr/issues">https://github.com/privefl/bigsnpr/issues</a>
</p>
</li>
</ul>
</div>