<div class="container">

<table style="width: 100%;"><tr>
<td>snp_simuPheno</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate phenotypes</h2>

<h3>Description</h3>

<p>Simulate phenotypes using a linear model. When a prevalence is given, the
liability threshold is used to convert liabilities to a binary outcome.
The genetic and environmental liabilities are scaled such that the variance
of the genetic liability is exactly equal to the requested heritability, and
the variance of the total liability is equal to 1.
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_simuPheno(
  G,
  h2,
  M,
  K = NULL,
  alpha = -1,
  ind.row = rows_along(G),
  ind.possible = cols_along(G),
  prob = NULL,
  effects.dist = c("gaussian", "laplace"),
  ncores = 1
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>G</code></td>
<td>
<p>A FBM.code256
(typically <code style="white-space: pre;">⁠&lt;bigSNP&gt;$genotypes⁠</code>).<br><strong>You shouldn't have missing values.</strong> Also, remember to do quality control,
e.g. some algorithms in this package won't work if you use SNPs with 0 MAF.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>h2</code></td>
<td>
<p>Heritability.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>M</code></td>
<td>
<p>Number of causal variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>K</code></td>
<td>
<p>Prevalence. Default is <code>NULL</code>, giving a continuous trait.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>Assumes that the average contribution (e.g. heritability)
of a SNP of frequency <code class="reqn">p</code> is proportional to
<code class="reqn">[2p(1-p)]^{1+\alpha}</code>. Default is <code>-1</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.row</code></td>
<td>
<p>An optional vector of the row indices (individuals) that
are used. If not specified, all rows are used.<br><strong>Don't use negative indices.</strong></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind.possible</code></td>
<td>
<p>Indices of possible causal variants.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prob</code></td>
<td>
<p>Vector of probability weights for sampling causal indices.
It can have 0s (discarded) and is automatically scaled to sum to 1.
Default is <code>NULL</code> (all indices have the same probability).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>effects.dist</code></td>
<td>
<p>Distribution of effects.
Either <code>"gaussian"</code> (the default) or <code>"laplace"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncores</code></td>
<td>
<p>Number of cores used. Default doesn't use parallelism.
You may use <code>bigstatsr::nb_cores()</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with 3 elements:
</p>

<ul>
<li> <p><code style="white-space: pre;">⁠$pheno⁠</code>: vector of phenotypes,
</p>
</li>
<li> <p><code style="white-space: pre;">⁠$set⁠</code>: indices of causal variants,
</p>
</li>
<li> <p><code style="white-space: pre;">⁠$effects⁠</code>: effect sizes (of scaled genotypes) corresponding to <code>set</code>.
</p>
</li>
<li> <p><code style="white-space: pre;">⁠$allelic_effects⁠</code>: effect sizes, but on the allele scale (0|1|2).
</p>
</li>
</ul>
</div>