<div class="container">

<table style="width: 100%;"><tr>
<td>snp_match</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Match alleles</h2>

<h3>Description</h3>

<p>Match alleles between summary statistics and SNP information.
Match by ("chr", "a0", "a1") and ("pos" or "rsid"), accounting for possible
strand flips and reverse reference alleles (opposite effects).
</p>


<h3>Usage</h3>

<pre><code class="language-R">snp_match(
  sumstats,
  info_snp,
  strand_flip = TRUE,
  join_by_pos = TRUE,
  remove_dups = TRUE,
  match.min.prop = 0.2,
  return_flip_and_rev = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>sumstats</code></td>
<td>
<p>A data frame with columns "chr", "pos", "a0", "a1" and "beta".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>info_snp</code></td>
<td>
<p>A data frame with columns "chr", "pos", "a0" and "a1".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strand_flip</code></td>
<td>
<p>Whether to try to flip strand? (default is <code>TRUE</code>)
If so, ambiguous alleles A/T and C/G are removed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>join_by_pos</code></td>
<td>
<p>Whether to join by chromosome and position (default),
or instead by rsid.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_dups</code></td>
<td>
<p>Whether to remove duplicates (same physical position)?
Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>match.min.prop</code></td>
<td>
<p>Minimum proportion of variants in the smallest data
to be matched, otherwise stops with an error. Default is <code style="white-space: pre;">⁠20%⁠</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_flip_and_rev</code></td>
<td>
<p>Whether to return internal boolean variables
<code>"_FLIP_"</code> (whether the alleles must be flipped: A &lt;–&gt; T &amp; C &lt;–&gt; G,
because on the opposite strand) and <code>"_REV_"</code> (whether alleles must be
swapped: <code style="white-space: pre;">⁠$a0⁠</code> &lt;–&gt; <code style="white-space: pre;">⁠$a1⁠</code>, in which case corresponding <code style="white-space: pre;">⁠$beta⁠</code> are multiplied
by -1). Default is <code>FALSE</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A single data frame with matched variants. Values in column <code style="white-space: pre;">⁠$beta⁠</code>
are multiplied by -1 for variants with alleles reversed (i.e. swapped).
New variable <code>"_NUM_ID_.ss"</code> returns the corresponding row indices of the
input <code>sumstats</code> (first argument of this function), and <code>"_NUM_ID_"</code>
corresponding to the input <code>info_snp</code> (second argument).
</p>


<h3>See Also</h3>

<p>snp_modifyBuild
</p>


<h3>Examples</h3>

<pre><code class="language-R">sumstats &lt;- data.frame(
  chr = 1,
  pos = c(86303, 86331, 162463, 752566, 755890, 758144),
  a0 = c("T", "G", "C", "A", "T", "G"),
  a1 = c("G", "A", "T", "G", "A", "A"),
  beta = c(-1.868, 0.250, -0.671, 2.112, 0.239, 1.272),
  p = c(0.860, 0.346, 0.900, 0.456, 0.776, 0.383)
)

info_snp &lt;- data.frame(
  id = c("rs2949417", "rs115209712", "rs143399298", "rs3094315", "rs3115858"),
  chr = 1,
  pos = c(86303, 86331, 162463, 752566, 755890),
  a0 = c("T", "A", "G", "A", "T"),
  a1 = c("G", "G", "A", "G", "A")
)

snp_match(sumstats, info_snp)
snp_match(sumstats, info_snp, strand_flip = FALSE)
</code></pre>


</div>