<div class="container">

<table style="width: 100%;"><tr>
<td>seqidentity</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Percent Identity</h2>

<h3>Description</h3>

<p>Determine the percent identity scores for aligned sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  seqidentity(alignment, normalize=TRUE, similarity=FALSE, ncore=1, nseg.scale=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p> sequence alignment obtained from
<code>read.fasta</code> or an alignment character matrix. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize</code></td>
<td>
<p> logical, if TRUE output is normalized to values
between 0 and 1 otherwise percent identity is returned. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>similarity</code></td>
<td>
<p> logical, if TRUE sequence similarity is calculated
instead of identity. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore</code></td>
<td>
<p> number of CPU cores used to do the calculation.
<code>ncore&gt;1</code> requires package ‘parallel’ installed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nseg.scale</code></td>
<td>
<p> split input data into specified number of segments
prior to running multiple core calculation. See <code>fit.xyz</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The percent identity value is a single numeric score determined for
each pair of aligned sequences.  It measures the number of identical
residues (“matches”) in relation to the length of the alignment.
</p>


<h3>Value</h3>

<p>Returns a numeric matrix with all pairwise identity values.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>read.fasta</code>, <code>filter.identity</code>,
<code>entropy</code>, <code>consensus</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
attach(kinesin)

ide.mat &lt;- seqidentity(pdbs)

# Plot identity matrix
plot.dmat(ide.mat, color.palette=mono.colors,
          main="Sequence Identity", xlab="Structure No.",
          ylab="Structure No.")

# Histogram of pairwise identity values
hist(ide.mat[upper.tri(ide.mat)], breaks=30,xlim=c(0,1),
     main="Sequence Identity", xlab="Identity")

# Compare two sequences
seqidentity( rbind(pdbs$ali[1,], pdbs$ali[15,]) )

detach(kinesin)
</code></pre>


</div>