<div class="container">

<table style="width: 100%;"><tr>
<td>seqaln.pair</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Sequence Alignment of Identical Protein Sequences </h2>

<h3>Description</h3>

<p>Create multiple alignments of amino acid sequences according to the
method of Edgar.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqaln.pair(aln, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> a sequence character matrix, as obtained from
<code>seqbind</code>, or an alignment list object as obtained from
<code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments for the function <code>seqaln</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function is intended for the alignment of identical sequences only.
For standard alignment see the related function <code>seqaln</code>.
</p>
<p>This function is useful for determining the equivalences between
sequences and structures. For example in aligning a PDB sequence to an
existing multiple sequence alignment, where one would first mask the
alignment sequences and then run the alignment to determine equivalences.
</p>


<h3>Value</h3>

<p>A list with two components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>A system call is made to the ‘MUSCLE’ program, which must be
installed on your system and in the search path for executables.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>‘MUSCLE’ is the work of Edgar:
Edgar (2004) <em>Nuc. Acid. Res.</em> <b>32</b>, 1792–1797.
</p>
<p>Full details of the ‘MUSCLE’ algorithm, along with download and
installation instructions can be obtained from:<br><a href="http://www.drive5.com/muscle/">http://www.drive5.com/muscle/</a>.
</p>


<h3>See Also</h3>

 <p><code>seqaln</code>, <code>read.fasta</code>,
<code>read.fasta.pdb</code>, <code>seqbind</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## NOTE: FOLLOWING EXAMPLE NEEDS MUSCLE INSTALLED
if(check.utility("muscle")) {

##- Aligning a PDB sequence to an existing sequence alignment


##- Simple example
aln &lt;- seqbind(c("X","C","X","X","A","G","K"),
               c("C","-","A","X","G","X","X","K"))

seqaln.pair(aln, outfile = tempfile())

}
</code></pre>


</div>