<div class="container">

<table style="width: 100%;"><tr>
<td>core.cmap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Identification of Contact Map Core Positions </h2>

<h3>Description</h3>

<p>Find core positions that have the largest number
of contact with neighboring residues.
</p>


<h3>Usage</h3>

<pre><code class="language-R">core.cmap(pdbs, write.pdb = FALSE, outfile="core.pdb",
          cutoff = NULL, refine = FALSE, ncore = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> an alignment data structure of class ‘pdbs’ 
as obtained with <code>read.fasta.pdb</code> or <code>pdbaln</code>, 
or a numeric matrix of aligned C-alpha xyz Cartesian coordinates. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>write.pdb</code></td>
<td>
<p> logical, if TRUE core coordinate files, containing
only core positions for each iteration, are written to a location
specified by <code>outpath</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p> character string specifying the output directory when
<code>write.pdb</code> is ‘TRUE’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p> numeric value speciyfing the inclusion criteria for
core positions. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refine</code></td>
<td>
<p> logical, if TRUE explore core positions determined by
multiple eigenvectors. By default only the eigenvector describing
the largest variation is used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore </code></td>
<td>
<p> number of CPU cores used to do the calculation.
By default (<code>ncore=NULL</code>) use all cores detected. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calculates eigenvector centrality of the weighted
contact network built based on input structure data and uses it
to determine the core positions.
</p>
<p>In this context, core positions correspond to the most invariant
C-alpha atom positions across an aligned set of protein
structures. Traditionally one would use the <code>core.find</code>
function to for their identification and then use these positions as
the basis for improved structural superposition. This more recent
function utilizes a  much faster approach and is thus preferred in
time sensitive applications such as shiny apps.
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"select"</code> containing ‘atom’ and
‘xyz’ indices.
</p>


<h3>Author(s)</h3>

<p> Xin-Qiu Yao </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>core.find</code>,
<code>read.fasta.pdb</code>, 
<code>fit.xyz</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##-- Generate a small kinesin alignment and read corresponding structures
pdbfiles &lt;- get.pdb(c("1bg2","2ncd","1i6i","1i5s"), URLonly=TRUE)
pdbs &lt;- pdbaln(pdbfiles)

##-- Find 'core' positions
core &lt;- core.cmap(pdbs)
xyz &lt;- pdbfit(pdbs, core, outpath="corefit_structures")

## End(Not run)
</code></pre>


</div>