<div class="container">

<table style="width: 100%;"><tr>
<td>rmsd</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Root Mean Square Deviation </h2>

<h3>Description</h3>

<p>Calculate the RMSD between coordinate sets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rmsd(a, b=NULL, a.inds=NULL, b.inds=NULL, fit=FALSE, ncore=1, nseg.scale=1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p> a numeric vector containing the reference coordinate set for
comparison with the coordinates in <code>b</code>.  Alternatively, if
<code>b=NULL</code> then <code>a</code> can be a matrix or list object
containing multiple coordinates for pairwise comparison. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p> a numeric vector, matrix or list object with an <code>xyz</code>
component, containing one or more coordinate sets to be compared with
<code>a</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>a.inds </code></td>
<td>
<p> a vector of indices that selects the elements of
<code>a</code> upon which the calculation should be based. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b.inds </code></td>
<td>
<p> a vector of indices that selects the elements of
<code>b</code> upon which the calculation should be based. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fit </code></td>
<td>
<p>logical, if TRUE coordinate superposition is performed
prior to RMSD calculation. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore </code></td>
<td>
<p> number of CPU cores used to do the calculation.
<code>ncore&gt;1</code> requires package ‘parallel’ installed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nseg.scale </code></td>
<td>
<p> split input data into specified number of segments
prior to running multiple core calculation. See <code>fit.xyz</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>RMSD is a standard measure of structural distance between
coordinate sets.
</p>
<p>Structure <code>a[a.inds]</code> and <code>b[b.inds]</code> should have the
same length.
</p>
<p>A least-squares fit is performed prior to RMSD calculation by setting
<code>fit=TRUE</code>.  See the function <code>fit.xyz</code> for more
details of the fitting process. 
</p>


<h3>Value</h3>

<p>Returns a numeric vector of RMSD value(s).
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>fit.xyz</code>, <code>rot.lsq</code>,
<code>read.pdb</code>, <code>read.fasta.pdb</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# Redundant testing excluded

# -- Calculate RMSD between two or more structures
aln &lt;- read.fasta(system.file("examples/kif1a.fa",package="bio3d"))
pdbs &lt;- read.fasta.pdb(aln)

# Gap positions
inds &lt;- gap.inspect(pdbs$xyz)

# Superposition before pairwise RMSD 
rmsd(pdbs$xyz, fit=TRUE)

# RMSD between structure 1 and structures 2 and 3
rmsd(a=pdbs$xyz[1,], b=pdbs$xyz[2:3,], a.inds=inds$f.inds, b.inds=inds$f.inds, fit=TRUE)

# RMSD between structure 1 and all structures in alignment
rmsd(a=pdbs$xyz[1,], b=pdbs, a.inds=inds$f.inds, b.inds=inds$f.inds, fit=TRUE)

# RMSD without superposition
rmsd(pdbs$xyz)



</code></pre>


</div>