<div class="container">

<table style="width: 100%;"><tr>
<td>as.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Alignment to FASTA object </h2>

<h3>Description</h3>

<p>Convert alignment/sequence in matrix/vector format to FASTA object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">as.fasta(x, id=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a sequence character matrix/vector (e.g obtained from
<code>get.seq</code> or <code>seqbind</code>). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> a vector of sequence names to serve as sequence
identifers. By default the function will use the row names of the
alignment if they exists, otherwise ids will be generated. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function provides basic functionality to convert a sequence
character matrix/vector to a FASTA object. 
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"fasta"</code> with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call </code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

<p><code>get.seq</code>, <code>seqaln</code>,
<code>seqbind</code>, <code>pdbaln</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">as.fasta(c("A", "C", "D"))
</code></pre>


</div>