<div class="container">

<table style="width: 100%;"><tr>
<td>example.data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bio3d Example Data</h2>

<h3>Description</h3>

<p>These data sets contain the results of running various Bio3D
functions on example kinesin and transducin structural data, and on a short 
coarse-grained MD simulation data for HIV protease. The main 
purpose of including this data (which may be generated by the user by following the extended examples documented within the various Bio3D functions) is to
speed up example execution. It should allow users to more quickly
appreciate the capabilities of functions that would otherwise require
raw data download, input and processing before execution.
</p>
<p>Note that related datasets formed the basis of
the work described in (Grant, 2007) and (Yao &amp; Grant, 2013) for <code>kinesin</code>
and <code>transducin</code> examples, respectively.
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(kinesin)
data(transducin)
data(hivp)
</code></pre>


<h3>Format</h3>

<p>Three objects from analysis of the <code>kinesin</code> and <code>transducin</code> sequence and structure
data:
</p>

<ol>
<li>
<p>pdbs is a list of class <code>pdbs</code> containing aligned PDB
structure data.  In the case of transducin this is the output of running 
<code>pdbaln</code> on a set of 53 G[alpha]i structures from the PDB database (see <code>pdbs$id</code> 
or <code>annotation</code> described below for details). The coordinates
are fitted onto the first structure based on <code>"core"</code> positions obtained from <code>core.find</code> and superposed using the function <code>pdbfit</code>. 
</p>
</li>
<li>
<p>core is a list of class <code>"core"</code> obtained by running the
function <code>core.find</code> on the <code>pdbs</code> object as described above.
</p>
</li>
<li>
<p>annotation is a character matrix describing the nucleotide state and bound 
ligand species for each structure in <code>pdbs</code> as obtained from the function <code>pdb.annotate</code>.
</p>
</li>
</ol>
<p>One object named <code>net</code> in the hivp example data stores the correlation 
network obtained from the analysis of the MD simulation trajectory of HIV 
protease using the <code>cna</code> function. The original trajectory file can be 
accessed by the command ‘system.file("examples/hivp.dcd", package="bio3d")’. 
</p>


<h3>Source</h3>

<p>A related but more extensive dataset formed the basis of
the work described in (Grant, 2007) and (Yao &amp; Grant, 2013) for <code>kinesin</code>
and <code>transducin</code> examples, respectively.
</p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>Grant, B.J. et al. (2007) <em>J. Mol. Biol.</em> <b>368</b>, 1231–1248.
</p>
<p>Yao, X.Q. et al. (2013) <em>Biophys. J.</em> <b>105</b>, L08–L10.
</p>


</div>