<div class="container">

<table style="width: 100%;"><tr>
<td>nma</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Normal Mode Analysis </h2>

<h3>Description</h3>

<p>Perform normal mode analysis (NMA) on either a single or an ensemble
of protein structures.
</p>


<h3>Usage</h3>

<pre><code class="language-R">nma(...)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to the methods
<code>nma.pdb</code>, or <code>nma.pdbs</code>.
</p>
<p>For function <code>nma.pdb</code> this will include an object of
class <code>pdb</code> as obtained from function <code>read.pdb</code>.
</p>
<p>For function <code>nma.pdbs</code> an object of class <code>pdbs</code> as
obtained from function <code>pdbaln</code> or
<code>read.fasta.pdb</code>. </p>
</td>
</tr></table>
<h3>Details</h3>

<p>Normal mode analysis (NMA) is a computational approach for studying and
characterizing protein flexibility. Current functionality entails normal
modes calculation on either a single protein structure or an ensemble
of aligned protein structures. 
</p>
<p>This generic <code>nma</code> function calls the corresponding
methods for the actual calculation, which is determined by the class
of the input argument:
</p>
<p>Function <code>nma.pdb</code> will be used when the input argument is
of class <code>pdb</code>. The function calculates the normal modes of a
C-alpha model of a protein structure.
</p>
<p>Function <code>nma.pdbs</code> will be used when the input argument is
of class <code>pdbs</code>. The function will perform normal mode analysis
of each PDB structure stored in the <code>pdbs</code> object
(‘ensemble NMA’).
</p>
<p>See documentation and examples for each corresponding function for
more details. 
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Skjaerven, L. et al. (2014) <em>BMC Bioinformatics</em> <b>15</b>, 399.
Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>nma.pdb</code>, <code>nma.pdbs</code>,
<code>pca</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
##- Singe structure NMA
## Fetch stucture
pdb &lt;- read.pdb( system.file("examples/1hel.pdb", package="bio3d") )

## Calculate normal modes
modes &lt;- nma(pdb)

## Print modes
print(modes)

## Plot modes
plot(modes)

## Visualize modes
#m7 &lt;- mktrj.nma(modes, mode=7, file="mode_7.pdb")


## Needs MUSCLE installed - testing excluded

##- Ensemble NMA
if(check.utility("muscle")) {

## Fetch PDB files and split to chain A only PDB files
ids &lt;- c("1a70_A", "1czp_A", "1frd_A", "1fxi_A", "1iue_A", "1pfd_A")
files &lt;- get.pdb(ids, split = TRUE, path = tempdir())

## Sequence Alignement
pdbs &lt;- pdbaln(files, outfile = tempfile())

## Normal mode analysis on aligned data
modes &lt;- nma(pdbs, rm.gaps=FALSE)

## Plot fluctuation data
plot(modes, pdbs=pdbs)

}

</code></pre>


</div>