<div class="container">

<table style="width: 100%;"><tr>
<td>filter.identity</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Percent Identity Filter </h2>

<h3>Description</h3>

<p>Identify and filter subsets of sequences at a given sequence identity cutoff.
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter.identity(aln = NULL, ide = NULL, cutoff = 0.6, verbose = TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> sequence alignment list, obtained from
<code>seqaln</code> or <code>read.fasta</code>, or an alignment
character matrix. Not used if ‘ide’ is given.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ide</code></td>
<td>
<p> an optional identity matrix obtained from
<code>seqidentity</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p> a numeric identity cutoff value ranging between 0 and 1. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE print details of the clustering
process. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function performs hierarchical cluster analysis of a given sequence
identity matrix ‘ide’, or the identity matrix calculated from a given
alignment ‘aln’, to identify sequences that fall below a given
identity cutoff value ‘cutoff’.  
</p>


<h3>Value</h3>

<p>Returns a list object with components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p>indices of the sequences below the cutoff value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tree</code></td>
<td>
<p>an object of class <code>"hclust"</code>, which describes the
tree produced by the clustering process. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ide</code></td>
<td>
<p>a numeric matrix with all pairwise identity values.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>seqaln</code>,
<code>seqidentity</code>, <code>entropy</code>, <code>consensus</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">attach(kinesin)

ide.mat &lt;- seqidentity(pdbs)

# Histogram of pairwise identity values
op &lt;- par(no.readonly=TRUE)
par(mfrow=c(2,1))
hist(ide.mat[upper.tri(ide.mat)], breaks=30,xlim=c(0,1),
     main="Sequence Identity", xlab="Identity")

k &lt;- filter.identity(ide=ide.mat, cutoff=0.6)
ide.cut &lt;- seqidentity(pdbs$ali[k$ind,])
hist(ide.cut[upper.tri(ide.cut)], breaks=10, xlim=c(0,1),
     main="Sequence Identity", xlab="Identity")

#plot(k$tree, axes = FALSE, ylab="Sequence Identity")
#print(k$ind) # selected
par(op)
detach(kinesin)
</code></pre>


</div>