<div class="container">

<table style="width: 100%;"><tr>
<td>bio3d-package</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Biological Structure Analysis </h2>

<h3>Description</h3>

<p>Utilities for the analysis of protein structure and sequence data.
</p>


<h3>Details</h3>


<table>
<tr>
<td style="text-align: left;">
Package: </td>
<td style="text-align: left;"> bio3d</td>
</tr>
<tr>
<td style="text-align: left;">
Type: </td>
<td style="text-align: left;"> Package</td>
</tr>
<tr>
<td style="text-align: left;">
Version: </td>
<td style="text-align: left;"> 2.4-4</td>
</tr>
<tr>
<td style="text-align: left;">
Date: </td>
<td style="text-align: left;"> 2022-10-20</td>
</tr>
<tr>
<td style="text-align: left;">
License: </td>
<td style="text-align: left;"> GPL version 2 or newer</td>
</tr>
<tr>
<td style="text-align: left;">
URL: </td>
<td style="text-align: left;"> <a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a>
</td>
</tr>
<tr>
<td style="text-align: left;">
</td>
</tr>
</table>
<p>Features include the ability to read and write structure
(<code>read.pdb</code>,  <code>write.pdb</code>,
<code>read.fasta.pdb</code>),  sequence (<code>read.fasta</code>,
<code>write.fasta</code>) and dynamics trajectory data
(<code>read.dcd</code>, <code>read.ncdf</code>, <code>write.ncdf</code>).
</p>
<p>Perform sequence and structure database searches (<code>blast.pdb</code>,
<code>hmmer</code>), atom summaries (<code>summary.pdb</code>),  atom selection
(<code>atom.select</code>), alignment (<code>pdbaln</code>,  <code>seqaln</code>,
<code>mustang</code>) superposition (<code>rot.lsq</code>,  <code>fit.xyz</code>),
<code>pdbfit</code>), rigid core identification (<code>core.find</code>,  <code>plot.core</code>,
<code>fit.xyz</code>), dynamic domain analysis (<code>geostas</code>), torsion/dihedral analysis
(<code>torsion.pdb</code>,  <code>torsion.xyz</code>), clustering (via
<code>hclust</code>), principal component analysis
(<code>pca.xyz</code>,  <code>pca.pdbs</code>, <code>pca.tor</code>,  <code>plot.pca</code>,
<code>plot.pca.loadings</code>,  <code>mktrj.pca</code>), dynamical
cross-correlation analysis (<code>dccm</code>, <code>plot.dccm</code>) and correlation network analysis (<code>cna</code>,  <code>plot.cna</code>, <code>cnapath</code>) of structure data.
</p>
<p>Perform conservation analysis of sequence (<code>seqaln</code>, <code>conserv</code>,
<code>seqidentity</code>,  <code>entropy</code>,  <code>consensus</code>)
and structural (<code>pdbaln</code>, <code>rmsd</code>,
<code>rmsf</code>, <code>core.find</code>) data.
</p>
<p>Perform normal mode analysis (<code>nma</code>, <code>build.hessian</code>), ensemble normal
mode analysis (<code>nma.pdbs</code>), mode comparison
(<code>rmsip</code>) and (<code>overlap</code>), atomic fluctuation
prediction (<code>fluct.nma</code>), cross-correlation analysis
(<code>dccm.nma</code>), cross-correlation visualization (<code>pymol.dccm</code>),
deformation analysis (<code>deformation.nma</code>), and mode visualization
(<code>pymol.modes</code>, <code>mktrj.nma</code>).
</p>
<p>In addition, various utility functions are provided to facilitate
manipulation and analysis of biological sequence and structural data
(e.g. <code>get.pdb</code>, <code>get.seq</code>, <code>aa123</code>,
<code>aa321</code>, <code>pdbseq</code>, <code>aln2html</code>, <code>atom.select</code>,
<code>rot.lsq</code>,  <code>fit.xyz</code>, <code>is.gap</code>, <code>gap.inspect</code>,
<code>orient.pdb</code>, <code>pairwise</code>, <code>plot.bio3d</code>, <code>plot.nma</code>, <code>plot.blast</code>, <code>biounit</code>, etc.).
</p>


<h3>Note</h3>

<p>The latest version, package vignettes and documentation with worked example
outputs can be obtained from the bio3d website:<br><a href="http://thegrantlab.org/bio3d/">http://thegrantlab.org/bio3d/</a>.<br><a href="http://thegrantlab.org/bio3d/reference/">http://thegrantlab.org/bio3d/reference/</a>.<br><a href="https://bitbucket.org/Grantlab/bio3d/">https://bitbucket.org/Grantlab/bio3d/</a>.
</p>


<h3>Author(s)</h3>

<p>Barry Grant &lt;bjgrant@ucsd.edu&gt;
Xin-Qiu Yao &lt;xinqiu.yao@gmail.com&gt;
Lars Skjaerven &lt;larsss@gmail.com&gt;
Julien Ide &lt;julien.ide.fr@gmail.com&gt;
</p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
Skjaerven, L. et al. (2014) <em>BMC Bioinformatics</em> <b>15</b>, 399.
Grant, B.J. et al. (2021) <em>Protein Science</em> <b>30</b>, 20–30.
</p>


<h3>Examples</h3>

<pre><code class="language-R">help(package="bio3d")     # list the functions within the package
#lbio3d()                  # list bio3d function names only

## Or visit:
##   http://thegrantlab.org/bio3d/reference/

## See the individual functions for further documentation and examples, e.g.
#help(read.pdb)

## Or online:
##    http://thegrantlab.org/bio3d/reference/read.pdb.html

## Not run: 
##-- See the list of Bio3D demos
demo(package="bio3d")

## Try some out, e.g:
demo(pdb) # PDB Reading, Manipulation, Searching and Alignment
demo(pca) # Principal Component Analysis
demo(md)  # Molecular Dynamics Trajectory Analysis
demo(nma) # Normal Mode Analysis

## See package vignettes and tutorals online:
##   http://thegrantlab.org/bio3d/articles/

## End(Not run)
</code></pre>


</div>