<div class="container">

<table style="width: 100%;"><tr>
<td>mustang</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Structure-based Sequence Alignment with MUSTANG </h2>

<h3>Description</h3>

<p>Create a multiple sequence alignment from a bunch of PDB files.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mustang(files, exefile="mustang", outfile="aln.mustang.fa",
        cleanpdb=FALSE, cleandir="mustangpdbs", verbose=TRUE)  
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p> a character vector of PDB file names. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exefile</code></td>
<td>
<p> file path to the ‘MUSTANG’ program on your system (i.e.
how is ‘MUSTANG’ invoked). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p> name of ‘FASTA’ output file to which alignment
should be written. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cleanpdb</code></td>
<td>
<p> logical, if TRUE iterate over the PDB files and map
non-standard residues to standard residues (e.g. SEP-&gt;SER..) to
produce ‘clean’ PDB files. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cleandir</code></td>
<td>
<p> character string specifying the directory in which
the ‘clean’ PDB files should be written. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE ‘MUSTANG’ warning and error
messages are printed. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Structure-based sequence alignment with ‘MUSTANG’ attempts to
arrange and align the sequences of proteins based on their 3D
structure.
</p>
<p>This function calls the ‘MUSTANG’ program, to perform a multiple
structure alignment, which MUST BE INSTALLED on your system and in the
search path for executables.
</p>
<p>Note that non-standard residues are mapped to “Z” in
MUSTANG. As a workaround the bio3d ‘mustang’ function will
attempt to map any non-standard residues to standard residues
(e.g. SEP-&gt;SER, etc). To avoid this behaviour use
‘cleanpdb=FALSE’.  
</p>


<h3>Value</h3>

<p>A list with two components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>A system call is made to the ‘MUSTANG’ program, which must be
installed on your system and in the search path for executables.
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>‘MUSTANG’ is the work of Konagurthu et al:
Konagurthu, A.S. et al. (2006) <em>Proteins</em> <b>64</b>(3):559–74.
</p>
<p>More details of the ‘MUSTANG’ algorithm, along with download and
installation instructions can be obtained from:<br><a href="https://lcb.infotech.monash.edu/mustang/">https://lcb.infotech.monash.edu/mustang/</a>.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code>,
<code>pdbaln</code>, <code>plot.fasta</code>,
<code>seqaln</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

if(!check.utility('mustang')) {
   message('Need MUSTANG installed to run this example')
} else {

## Fetch PDB files and split to chain A only PDB files
ids &lt;- c("1a70_A", "1czp_A", "1frd_A")
files &lt;- get.pdb(ids, split = TRUE, path = tempdir())

##-- Or, read a folder/directory of existing PDB files
#pdb.path &lt;- "my_dir_of_pdbs"
#files  &lt;- list.files(path=pdb.path ,
#                     pattern=".pdb",
#                     full.names=TRUE)

##-- Align these PDB sequences
aln &lt;- mustang(files)

##-- Read Aligned PDBs storing coordinate data
pdbs &lt;- read.fasta.pdb(aln) 

}

## End(Not run)
</code></pre>


</div>