<div class="container">

<table style="width: 100%;"><tr>
<td>pymol</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Biomolecular Visualization with PyMOL </h2>

<h3>Description</h3>

<p>Visualize Bio3D structure objects in PyMOL
</p>


<h3>Usage</h3>

<pre><code class="language-R">pymol(...)

## S3 method for class 'pdbs'
pymol(pdbs, col=NULL, as="ribbon", file=NULL, type="script",
                     exefile="pymol", user.vec=NULL, ...)

## S3 method for class 'nma'
pymol(...)

## S3 method for class 'pca'
pymol(...)

## S3 method for class 'modes'
pymol(modes, mode=NULL, file=NULL, scale=5, dual=FALSE,
                     type="script", exefile="pymol", ...)

## S3 method for class 'dccm'
pymol(dccm, pdb, file=NULL,
                     step=0.2, omit=0.2, radius = 0.15,
                     type="script", exefile="pymol", ...)


</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> aligned C-alpha Cartesian coordinates as obtained with 
<code>read.fasta.pdb</code> or <code>pdbaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p> a single element character vector specifying the coloring
of the structures. Options are: ‘index’, ‘index2’,
‘gaps’, ‘rmsf’, ‘user’.
</p>
<p>Special cases: Provide a ‘core’ object as obtained by
<code>core.find</code> to color on the invariant core. Alternatively,
provide a vector containing the color code for each structure in the
‘pdbs’ object. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>user.vec</code></td>
<td>
<p> User defined vector for coloring. Only used if <code>col="user"</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>as</code></td>
<td>
<p> show as ‘ribbon’, ‘cartoon’, ‘lines’, ‘putty’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> a single element character vector specifying the file name
of the PyMOL session/script file. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p> a single element character vector specifying the output
type: ‘script’ generates a .pml script; ‘session’
generates a .pse session file; ‘launch’ launches pymol. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exefile</code></td>
<td>
<p> file path to the ‘PYMOL’ program on your system (i.e.
how is ‘PYMOL’ invoked). If <code>NULL</code>, use OS-dependent 
default path to the program. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>modes</code></td>
<td>
<p> an object of class <code>nma</code> or <code>pca</code> as obtained
from functions <code>nma</code> or <code>pca.xyz</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mode</code></td>
<td>
<p> the mode number for which the vector field should be
made. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale</code></td>
<td>
<p> global scaling factor. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dual</code></td>
<td>
<p> logical, if TRUE mode vectors are also drawn in both
direction. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dccm</code></td>
<td>
<p> an object of class <code>dccm</code> as obtained from
function <code>dccm</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> an object of class <code>pdb</code> as obtained from
function <code>read.pdb</code> or a numerical vector of Cartesian
coordinates. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>step</code></td>
<td>
<p> binning interval of cross-correlation coefficents. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>omit</code></td>
<td>
<p> correlation coefficents with values (0-omit, 0+omit) will
be omitted from visualization. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>radius</code></td>
<td>
<p> numeric, radius of visualized correlation cylinders in
PyMol. Alternatively, a matrix with the same dimesions as
<code>dccm</code> can be provided, e.g. to draw cylinders with radii
associated to the pairwise correlation value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to function <code>pymol.modes</code> for
‘nma’ and ‘pca’ objects. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>These functions provides a convenient approach for the visualization of
Bio3D objects in PyMOL. See examples for more details.
</p>
<p>DCCM PyMOL visualization:
This function generates a PyMOL (python) script that will draw colored
lines between (anti)correlated residues. The PyMOL script file is
stored in the working directory with filename “R.py”.
PyMOL will only be launched (and opened)  when using argument
‘type='launch'’. Alternatively a PDB file with CONECT records
will be generated (when argument <code>type='pdb'</code>). 
</p>
<p>For the PyMOL version, PyMOL CGO objects are generated - each object
representing a range of correlation values (corresponding to the
actual correlation values as found in the correlation
matrix). E.g. the PyMOL object with name “cor_-1_-08” would
display all pairs of correlations with values between -1 and -0.8.
</p>
<p>NMA / PCA PyMOL vector field visualization:
This function generates a PyMOL (python) script for drawing
mode vectors on a PDB structure. The PyMOL script file is stored in
the working directory with filename “R.py”. 
</p>


<h3>Value</h3>

<p>Called for its action
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Skjaerven, L. et al. (2014) <em>BMC Bioinformatics</em> <b>15</b>, 399.
Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>view</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 

##- pymol with a 'pdbs' object
attach(transducin)

# build a pymol session containing all structures in the PDBs object
pymol(pdbs)

# color by invariant core (
# core &lt;- core.find(pdbs)
pymol(pdbs, col=core)

# color by RMSF
pymol(pdbs, col="rmsf")

# color by a user defined vector
# For example, colored by averaged contact density around each residue
cm &lt;- cmap(pdbs, binary=FALSE)
vec &lt;- rowSums(cm, na.rm=TRUE)
pymol(pdbs, col="user", user.vec=vec)

# color by clustering
rd &lt;- rmsd(pdbs$xyz)
hc &lt;- hclust(as.dist(rd))
grps &lt;- cutree(hc, k=3)
pymol(pdbs, col=grps)



##- pymol with a 'dccm' object
## Fetch stucture
pdb &lt;- read.pdb( system.file("examples/1hel.pdb", package="bio3d") )

## Calculate normal modes
modes &lt;- nma(pdb)

## Calculate correlation matrix
cm &lt;- dccm.nma(modes)

pymol(cm, modes$xyz)



##- pymol with a 'nma' or 'pca' object
pymol(modes, mode=7)

detach(transducin)

## End(Not run)
</code></pre>


</div>