<div class="container">

<table style="width: 100%;"><tr>
<td>seqaln</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Sequence Alignment with MUSCLE</h2>

<h3>Description</h3>

<p>Create multiple alignments of amino acid or nucleotide sequences
according to the method of Edgar.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqaln(aln, id=NULL, profile=NULL, exefile="muscle", outfile="aln.fa", 
       protein=TRUE, seqgroup=FALSE, refine=FALSE, extra.args="",
       verbose=FALSE, web.args = list(), ...)  
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> a sequence character matrix, as obtained from
<code>seqbind</code>, or an alignment list object as obtained from
<code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> a vector of sequence names to serve as sequence
identifers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>profile</code></td>
<td>
<p> a profile alignment of class ‘fasta’
(e.g. obtained from <code>read.fasta</code>). The alignment
<code>aln</code> will be added to the profile. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exefile</code></td>
<td>
<p> file path to the ‘MUSCLE’ program on your system (i.e.
how is ‘MUSCLE’ invoked). Alternatively, ‘CLUSTALO’ can
be used. Also supported is using the ‘msa’ package from Bioconductor
(need to install packages using <code>BiocManager::install()</code>). To do so, simply
set <code>exefile="msa"</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p> name of ‘FASTA’ output file to which alignment
should be written. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>protein</code></td>
<td>
<p> logical, if TRUE the input sequences are assumed to be
protein not DNA or RNA. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqgroup</code></td>
<td>
<p> logical, if TRUE similar sequences are grouped
together in the output. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>refine</code></td>
<td>
<p> logical, if TRUE the input sequences are assumed to already
be aligned, and only tree dependent refinement is performed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>extra.args</code></td>
<td>
<p> a single character string containing extra command
line arguments for the alignment program. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE ‘MUSCLE’ warning and error
messages are printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>web.args</code></td>
<td>
<p> a ‘list’ object containing arguments to perform
online sequence alignment using EMBL-EBI Web Services. 
See below for details. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to the function
<code>msa::msaMuscle()</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Sequence alignment attempts to arrange the sequences of protein,
DNA or RNA, to highlight regions of shared similarity that may
reflect functional, structural, and/or evolutionary relationships
between the sequences.
</p>
<p>Aligned sequences are represented as rows within a matrix. Gaps
(‘-’) are inserted between the aminoacids or nucleotides so that
equivalent characters are positioned in the same column.
</p>
<p>This function calls the ‘MUSCLE’ program to perform a multiple 
sequence alignment, which must be installed on your system and in the search 
path for executables. If local ‘MUSCLE’ can not be found, alignment
can still be performed via online web services (see below) with limited
features. 
</p>
<p>If you have a large number of input sequences (a few thousand), or they are
very long, the default settings may be too slow for practical
use. A good compromise between speed and accuracy is to run just the
first two iterations of the ‘MUSCLE’ algorithm by setting the
<code>extra.args</code> argument to “-maxiters 2”.
</p>
<p>You can set ‘MUSCLE’ to improve an existing alignment by setting
<code>refine</code> to TRUE.
</p>
<p>To inspect the sequence clustering used by ‘MUSCLE’ to produce
alignments, include “-tree2 tree.out” in the <code>extra.args</code>
argument. You can then load the “tree.out” file with the
‘read.tree’ function from the ‘ape’ package.
</p>
<p>‘CLUSTALO’ can be used as an alternative to ‘MUSCLE’ by
specifiying <code>exefile='clustalo'</code>. This might be useful e.g. when
adding several sequences to a profile alignment. 
</p>
<p>If local ‘MUSCLE’ or ‘CLUSTALO’ program is unavailable, the alignment
can be performed via the ‘msa’ package from the Bioconductor repository.
To do so, set <code>exefile="msa"</code>. Note that both ‘msa’ and 
‘Biostrings’ packages need to be installed properly using <code>BiocManager::install()</code>. 
</p>
<p>If the access to any method metioned above fails, 
the function will attempt to perform alignment via the EMBL-EBI Web Services 
(See <a href="https://www.ebi.ac.uk/">https://www.ebi.ac.uk/</a>). In this case, the argument <code>web.args</code> 
cannot be empty and must contain at least user's E-Mail address. 
Note that as stated by EBI, a fake email address may result 
in your jobs being killed and your IP, organisation or entire domain being 
black-listed (See FAQs on <a href="https://www.ebi.ac.uk/">https://www.ebi.ac.uk/</a>). 
Possible parameters to be passed via <code>web.args</code> include:
</p>

<dl>
<dt>email</dt>
<dd>
<p> a string containing a valid E-Mail address. Required. </p>
</dd>
<dt>title</dt>
<dd>
<p> a string for the title of the job to be submitted to
the remote server. Optional. </p>
</dd>
<dt>timeout</dt>
<dd>
<p> integer specifying the number of seconds to wait for
the response of the server before a time out occurs. Default: 90. </p>
</dd>
</dl>
<p>An example of usage is <code>web.args=list(email='user_id@email.provider')</code>.
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"fasta"</code> with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>A system call is made to the ‘MUSCLE’ program, which must be
installed on your system and in the search path for executables.
See <a href="http://thegrantlab.org/bio3d/articles/online/install_vignette/Bio3D_install.html">http://thegrantlab.org/bio3d/articles/online/install_vignette/Bio3D_install.html</a> for 
instructions of how to install this program.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>‘MUSCLE’ is the work of Edgar:
Edgar (2004) <em>Nuc. Acid. Res.</em> <b>32</b>, 1792–1797.
</p>
<p>Full details of the ‘MUSCLE’ algorithm, along with download and
installation instructions can be obtained from:<br><a href="http://www.drive5.com/muscle/">http://www.drive5.com/muscle/</a>.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code>,
<code>get.seq</code>, <code>seqbind</code>, 
<code>pdbaln</code>, <code>plot.fasta</code>,
<code>blast.pdb</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##-- Basic sequence alignemnt
seqs &lt;- get.seq(c("4q21_A", "1ftn_A"))
aln &lt;- seqaln(seqs)

##-- add a sequence to the (profile) alignment
seq &lt;- get.seq("1tnd_A")
aln &lt;- seqaln(seq, profile=aln)


##-- Read a folder/directory of PDB files
#pdb.path &lt;- "my_dir_of_pdbs"
#files  &lt;- list.files(path=pdb.path ,
#                     pattern=".pdb",
#                     full.names=TRUE)

##-- Use online files
files &lt;- get.pdb(c("4q21","1ftn"), URLonly=TRUE)

##-- Extract and store sequences
raw &lt;- NULL
for(i in 1:length(files)) {
  pdb &lt;- read.pdb(files[i])
  raw &lt;- seqbind(raw, pdbseq(pdb) )
}

##-- Align these sequences
aln &lt;- seqaln(raw, id=files, outfile="seqaln.fa")

##-- Read Aligned PDBs storing coordinate data
pdbs &lt;- read.fasta.pdb(aln) 

## Sequence identity
seqidentity(aln)

## Note that all the above can be done with the pdbaln() function:
#pdbs &lt;- pdbaln(files)


##-- For identical sequences with masking use a custom matrix
aa &lt;- seqbind(c("X","C","X","X","A","G","K"),
              c("C","-","A","X","G","X","X","K"))

aln &lt;- seqaln(aln=aln, id=c("a","b"), outfile="temp.fas", protein=TRUE,
               extra.args= paste("-matrix",
                system.file("matrices/custom.mat", package="bio3d"),
                "-gapopen -3.0 ",
                "-gapextend -0.5",
                "-center 0.0") )

## End(Not run)
</code></pre>


</div>