<div class="container">

<table style="width: 100%;"><tr>
<td>dm</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Distance Matrix Analysis </h2>

<h3>Description</h3>

<p>Construct a distance matrix for a given protein structure.
</p>


<h3>Usage</h3>

<pre><code class="language-R">dm(...)

## S3 method for class 'pdb'
dm(pdb, inds = NULL, grp = TRUE, verbose=TRUE, ...)
## S3 method for class 'pdbs'
dm(pdbs, rm.gaps=FALSE, all.atom=FALSE, 
                        aligned.atoms.only=NULL, ...)

## S3 method for class 'xyz'
dm(xyz, grpby = NULL, scut = NULL, mask.lower = TRUE, 
                  gc.first=FALSE, ncore=1, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> a <code>pdb</code> structure object as returned by
<code>read.pdb</code> or a numeric vector of ‘xyz’ coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inds</code></td>
<td>
<p> atom and xyz coordinate indices obtained from <code>atom.select</code>
that selects the elements of <code>pdb</code> upon which the calculation
should be based. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grp</code></td>
<td>
<p> logical, if TRUE atomic distances will be grouped according
to their residue membership. See ‘grpby’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE possible warnings are printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> a ‘pdbs’ object as returned by <code>read.fasta.pdb</code>,
<code>read.all</code>, or <code>pdbaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm.gaps</code></td>
<td>
<p> logical, if TRUE gapped positions are removed in the returned value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>all.atom</code></td>
<td>
<p> logical, if TRUE all-atom coordinates from <code>read.all</code>
are used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aligned.atoms.only</code></td>
<td>
<p> logical, if TRUE only equivalent (aligned) atoms are considered. 
Only meaningful when <code>all.atom=TRUE</code>. Default: FALSE. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p> a numeric vector or matrix of Cartesian coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grpby</code></td>
<td>
<p> a vector counting connective duplicated elements that
indicate the elements of <code>xyz</code> that should be considered as a group
(e.g. atoms from a particular residue). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scut</code></td>
<td>
<p> a cutoff neighbour value which has the effect of
excluding atoms, or groups, that are sequentially within this
value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mask.lower</code></td>
<td>
<p> logical, if TRUE the lower matrix elements
(i.e. those below the diagonal) are returned as NA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gc.first</code></td>
<td>
<p> logical, if TRUE will call gc() first before calculation of 
distance matrix. This is to solve the memory overload problem when <code>ncore &gt; 1</code>
and <code>xyz</code> has many rows/columns, with a bit sacrifice on speed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore </code></td>
<td>
<p> number of CPU cores used to do the calculation.
<code>ncore&gt;1</code> requires package ‘parallel’ installed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Distance matrices, also called distance plots or distance maps, are an
established means of describing and comparing protein conformations
(e.g. Phillips, 1970; Holm, 1993).
</p>
<p>A distance matrix is a 2D representation of 3D structure that is
independent of the coordinate reference frame and, ignoring
chirality, contains enough information to reconstruct the 3D Cartesian
coordinates (e.g. Havel, 1983).
</p>


<h3>Value</h3>

<p>Returns a numeric matrix of class <code>"dmat"</code>, with all N by N
distances, where N is the number of selected atoms. With multiple
frames the output is provided in a three dimensional array. 
</p>


<h3>Note</h3>

<p>The input <code>selection</code> can be any character string or pattern
interpretable by the function <code>atom.select</code>.  For example,
shortcuts <code>"calpha"</code>, <code>"back"</code>, <code>"all"</code> and selection
strings of the form <code>/segment/chain/residue number/residue
    name/element number/element name/</code>; see <code>atom.select</code>
for details.
</p>
<p>If a coordinate vector is provided as input (rather than a <code>pdb</code>
object) the <code>selection</code> option is redundant and the input vector
should be pruned instead to include only desired positions. 
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>Phillips (1970) <em>Biochem. Soc. Symp.</em> <b>31</b>, 11–28.
</p>
<p>Holm (1993) <em>J. Mol. Biol.</em> <b>233</b>, 123–138.
</p>
<p>Havel (1983) <em>Bull. Math. Biol.</em> <b>45</b>, 665–720.
</p>


<h3>See Also</h3>

 <p><code>plot.dmat</code>, <code>read.pdb</code>, <code>atom.select</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# PDB server connection required - testing excluded

##--- Distance Matrix Plot
pdb &lt;- read.pdb( "4q21" )
k &lt;- dm(pdb,inds="calpha")
filled.contour(k, nlevels = 10)

## NOTE: FOLLOWING EXAMPLE NEEDS MUSCLE INSTALLED
if(check.utility("muscle")) {

##--- DDM: Difference Distance Matrix
# Downlaod and align two PDB files
pdbs &lt;- pdbaln( get.pdb( c( "4q21", "521p"), path = tempdir() ), outfile = tempfile() )

# Get distance matrix
a &lt;- dm.xyz(pdbs$xyz[1,])
b &lt;- dm.xyz(pdbs$xyz[2,])

# Calculate DDM
c &lt;- a - b

# Plot DDM
plot(c,key=FALSE, grid=FALSE)

plot(c, axis.tick.space=10,
     resnum.1=pdbs$resno[1,],
     resnum.2=pdbs$resno[2,],
     grid.col="black",
     xlab="Residue No. (4q21)", ylab="Residue No. (521p)")
}


## Not run: 
##-- Residue-wise distance matrix based on the
##   minimal distance between all available atoms
l &lt;- dm.xyz(pdb$xyz, grpby=pdb$atom[,"resno"], scut=3)

## End(Not run)
</code></pre>


</div>