<div class="container">

<table style="width: 100%;"><tr>
<td>seqbind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Combine Sequences by Rows Without Recycling </h2>

<h3>Description</h3>

<p>Take vectors and/or matrices arguments and combine them row-wise without
recycling them (as is the case with <code>rbind</code>).
</p>


<h3>Usage</h3>

<pre><code class="language-R">seqbind(..., blank = "-")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> vectors, matrices, and/or alignment ‘fasta’
objects to combine. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blank</code></td>
<td>
<p> a character to add to short arguments, to achieve the
same length as the longer argument. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a list of class <code>"fasta"</code> with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>seqaln</code>, <code>read.fasta</code>,
<code>read.pdb</code>, <code>write.fasta</code>, <code>rbind</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Read two pdbs
a.pdb &lt;- read.pdb("1bg2")
b.pdb &lt;- read.pdb("1goj")

seqs &lt;- seqbind(aa321(a.pdb$atom[a.pdb$calpha,"resid"]),
                aa321(b.pdb$atom[b.pdb$calpha,"resid"]))

# seqaln(seqs)

## End(Not run)
</code></pre>


</div>