<div class="container">

<table style="width: 100%;"><tr>
<td>pdbaln</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Sequence Alignment of PDB Files </h2>

<h3>Description</h3>

<p>Create multiple sequences alignments from a list of PDB files
returning aligned sequence and structure records.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pdbaln(files, fit = FALSE, pqr = FALSE, ncore = 1,
       nseg.scale = 1, progress = NULL, ...) 
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>files</code></td>
<td>
<p> a character vector of PDB file names. Alternatively, a
list of <code>pdb</code> objects can be provided. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fit</code></td>
<td>
<p> logical, if TRUE coordinate superposition is performed on
the input structures. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pqr</code></td>
<td>
<p> logical, if TRUE the input structures are assumed to be in
PQR format. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore </code></td>
<td>
<p> number of CPU cores used to do the calculation.
<code>ncore&gt;1</code> requires package ‘parallel’ installed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nseg.scale </code></td>
<td>
<p> split input data into specified number of segments
prior to running multiple core calculation. See <code>fit.xyz</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p> progress bar for use with shiny web app. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> extra arguments passed to <code>seqaln</code> function. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This wrapper function calls the underlying functions <code>read.pdb</code>,
<code>pdbseq</code>, <code>seqaln</code> and <code>read.fasta.pdb</code> returning a
list of class <code>"pdbs"</code> similar to that returned by
<code>read.fasta.pdb</code>.
</p>
<p>As these steps are often error prone it is recomended for most cases that
the individual underlying functions are called in sequence with checks
made on the valadity of their respective outputs to ensure sensible
results.
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"pdbs"</code> with the following five
components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p>numeric matrix of aligned C-alpha coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p>character matrix of aligned residue numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>numeric matrix of aligned B-factor values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p>character matrix of aligned chain identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>character vector of PDB sequence/structure names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p>character matrix of aligned sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p> See recommendation in details section above. </p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>read.pdb</code>, <code>pdbseq</code>, <code>seqaln</code>,
<code>read.fasta</code>,<code>read.fasta.pdb</code>,
<code>core.find</code>, <code>fit.xyz</code>, <code>read.all</code>,
<code>pymol.pdbs</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##- Align PDBs (from vector of filenames)
#files &lt;- get.pdb(c("4q21","5p21"), URLonly=TRUE)
files &lt;- get.pdb(c("4q21","5p21"), path=tempdir(), overwrite=TRUE)
pdbaln(files)


##- Align PDBs (from list of existing PDB objects)
pdblist &lt;- list(read.pdb(files[1]), read.pdb(files[2]))
pdbaln(pdblist)


## End(Not run)
</code></pre>


</div>