<div class="container">

<table style="width: 100%;"><tr>
<td>pdbs2sse</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> SSE annotation for a PDBs Object </h2>

<h3>Description</h3>

<p>Returns secondary structure element (SSE) annotation (<code>"sse"</code> object) for a structure in the provided <code>"pdbs"</code> object.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pdbs2sse(pdbs, ind = NULL, rm.gaps = TRUE, resno = TRUE, pdb = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> a list of class <code>"pdbs"</code> containing PDB file
data, as obtained from <code>read.fasta.pdb</code> or
<code>pdbaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ind</code></td>
<td>
<p> numeric index pointing to the PDB in which the SSE should
be provided. If <code>ind=NULL</code>, then the consensus SSE is returned. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm.gaps</code></td>
<td>
<p> logical, if TRUE SSEs spanning gap containing columns are
omitted from the output in the resulting <code>sse</code> object. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p> logical, if TRUE output is in terms of residue numbers
rather than residue index (position in sequence). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> logical, if TRUE function <code>dssp</code> will be called on
the corresponding <code>pdb</code> object rather than to use <code>pdbs$sse</code> to
obtain the SSE object. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to function <code>dssp</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function provides a <code>"sse"</code> list object containing
secondary structure elements (SSE) annotation data for a particular
structure in the provided <code>"pdbs"</code> object. Residue numbers are
provided relative to the alignment in the <code>"pdbs"</code> object.
</p>
<p>When <code>ind=NULL</code> the function will attemt to return the consensus
SSE annotation, i.e. where there are SSEs across all structures. This
will only work SSE data is found in the <code>"pdbs"</code> object.
</p>
<p>See examples for more details.
</p>


<h3>Value</h3>

<p>Returns a list object of class <code>sse</code>.
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>dssp</code>, <code>pdbaln</code>,
<code>read.fasta.pdb</code>. </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
attach(transducin)

## calculate RMSF
rf &lt;- rmsf(pdbs$xyz)

## Fetch SSE annotation, output in terms of alignment index
sse &lt;- pdbs2sse(pdbs, ind=1, rm.gaps=FALSE, resno=FALSE)

## Add SSE annotation to plot
plotb3(rf, sse=sse)


## Calculate RMSF only for non-gap columns
gaps.pos &lt;- gap.inspect(pdbs$xyz)
rf &lt;- rmsf(pdbs$xyz[, gaps.pos$f.inds])

## With gap columns removed, output in terms of residue number
sse &lt;- pdbs2sse(pdbs, ind=1, rm.gaps=TRUE, resno=TRUE)
gaps.res &lt;- gap.inspect(pdbs$ali)
plotb3(rf, sse=sse, resno=pdbs$resno[1, gaps.res$f.inds])

detach(transducin)

## End(Not run)
</code></pre>


</div>