<div class="container">

<table style="width: 100%;"><tr>
<td>rmsf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Atomic RMS Fluctuations </h2>

<h3>Description</h3>

<p>Calculate atomic root mean squared fluctuations.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rmsf(xyz, grpby=NULL, average=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p> numeric matrix of coordinates with each row corresponding
to an individual conformer. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grpby</code></td>
<td>
<p> a vector counting connective duplicated elements that
indicate the elements of 'xyz' that should be considered as a
group (e.g. atoms from a particular residue). If provided a
'pdb' object, grouping is automatically set by amino acid residues. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>average</code></td>
<td>
<p> logical, if TRUE averaged over atoms. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>RMSF is an often used measure of conformational variance. It is calculated by
</p>
<p style="text-align: center;"><code class="reqn">f_i=\sqrt{\frac{1}{M-1}\sum_j \|r_i^j-r_i^0\|^2}</code>
</p>
<p>,
where <code class="reqn">f_i</code> is the RMSF value for the ith atom, M the total number of frames 
(total number of rows of <code>xyz</code>), <code class="reqn">r_i^j</code> the positional vector of the
ith atom in the jth frame, and <code class="reqn">r_i^0</code> the mean position of ith atom. 
||r|| denotes the Euclidean norm of the vector r.
</p>


<h3>Value</h3>

<p>Returns a numeric vector of RMSF values. If <code>average=TRUE</code> a single numeric value
representing the averaged RMSF value over all atoms will be returned. 
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>read.dcd</code>, <code>fit.xyz</code>,
<code>read.fasta.pdb</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">attach(transducin)

# Ignore Gaps
gaps &lt;- gap.inspect(pdbs$ali)

r &lt;- rmsf(pdbs$xyz)
plot(r[gaps$f.inds], typ="h", ylab="RMSF (A)")

detach(transducin)

## Not run: 

pdb &lt;- read.pdb("1d1d", multi=TRUE)
xyz &lt;- pdb$xyz

# superimpose trajectory
xyz &lt;- fit.xyz(xyz[1, ], xyz)

# select mainchain atoms
sele &lt;- atom.select(pdb, elety=c("CA", "C", "N", "O"))

# residue numbers to group by
resno &lt;- pdb$atom$resno[sele$atom]

# mean rmsf value of mainchain atoms of each residue
r &lt;- rmsf(xyz[, sele$xyz], grpby=resno)
plot.bio3d(r, resno=pdb, sse=pdb, ylab="RMSF (A)")


## End(Not run)

</code></pre>


</div>