<div class="container">

<table style="width: 100%;"><tr>
<td>read.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Read FASTA formated Sequences </h2>

<h3>Description</h3>

<p>Read aligned or un-aligned sequences from a FASTA format file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.fasta(file, rm.dup = TRUE, to.upper = FALSE, to.dash=TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> input sequence file. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>rm.dup</code></td>
<td>
<p> logical, if TRUE duplicate sequences (with the same
names/ids) will be removed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>to.upper</code></td>
<td>
<p> logical, if TRUE residues are forced to uppercase. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>to.dash</code></td>
<td>
<p> logical, if TRUE ‘.’ gap characters are
converted to ‘-’ gap characters. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list with two components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali </code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids </code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

  
<p>For a description of FASTA format see:
<a href="https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml">https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml</a>.
When reading alignment files, the dash ‘-’ is interpreted as
the gap character.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta.pdb</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># Read alignment
aln &lt;- read.fasta(system.file("examples/hivp_xray.fa",package="bio3d"))

# Print alignment overview
aln

# Sequence names/ids
head( aln$id )

# Alignment positions 335 to 339
head( aln$ali[,33:39] )

# Sequence d2a4f_b
aa123( aln$ali["d2a4f_b",] )

# Write out positions 33 to 45 only
#aln$ali=aln$ali[,30:45]
#write.fasta(aln, file="eg2.fa")

</code></pre>


</div>