<div class="container">

<table style="width: 100%;"><tr>
<td>is.gap</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Gap Characters </h2>

<h3>Description</h3>

<p>Test for the presence of gap characters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">is.gap(x, gap.char = c("-", "."))
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> an R object to be tested. Typically a sequence vector or 
sequence/structure alignment object as returned from <code>seqaln</code>, 
<code>pdbaln</code> etc. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap.char</code></td>
<td>
<p> a character vector containing the gap character types
to test for. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Returns a logical vector with the same length as the input vector, or 
the same length as the number of columns present in an alignment input 
object ‘x’. In the later case TRUE elements corresponding to 
‘gap.char’ matches in any alignment column (i.e. gap containing 
columns).
</p>


<h3>Note</h3>

<p>During alignment, gaps are introduced into sequences that are believed
to have undergone deletions or insertions with respect to other
sequences in the alignment.  These gaps, often referred to as
indels, can be represented with ‘NA’, ‘-’ or
‘.’ characters.
</p>
<p>This function provides a simple test for the presence of such
characters, or indeed any set of user defined characters set by
the ‘gap.char’ argument.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>gap.inspect</code>, <code>read.fasta</code>,
<code>read.fasta.pdb</code>, <code>seqaln</code>, <code>pdbaln</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
is.gap( c("G",".","X","-","G","K","S","T") )

## Not run: 
aln &lt;- read.fasta( system.file("examples/kif1a.fa",
                   package = "bio3d") )

##- Print only non-gap positions (i.e. no gaps in any sequence)
aln$ali[, !is.gap(aln) ]

##- Mask any existing gaps with an "X"
xaln &lt;- aln
xaln$ali[ is.gap(xaln$ali) ]="X"

##- Read a new PDB and align its sequence to the existing masked alignment
pdb &lt;- read.pdb( "1mkj" )
seq2aln(pdbseq(pdb), xaln, id = "1mkj")

## End(Not run)
</code></pre>


</div>