<div class="container">

<table style="width: 100%;"><tr>
<td>atom.select</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Atom Selection from PDB and PRMTOP Structure Objects </h2>

<h3>Description</h3>

<p>Return the ‘atom’ and ‘xyz’ coordinate indices of
‘pdb’ or ‘prmtop’ structure objects corresponding to the
intersection of a hierarchical selection.
</p>


<h3>Usage</h3>

<pre><code class="language-R">atom.select(...)

## S3 method for class 'pdb'
atom.select(pdb, string = NULL,
                          type  = NULL, eleno = NULL, elety = NULL,
                          resid = NULL, chain = NULL, resno = NULL,
                          insert = NULL, segid = NULL, 
                          operator = "AND", inverse = FALSE,
                          value = FALSE, verbose=FALSE, ...)
## S3 method for class 'pdbs'
atom.select(pdbs, string = NULL, 
                           resno = NULL, chain = NULL, resid = NULL,
                           operator="AND", inverse = FALSE,
                           value = FALSE, verbose=FALSE, ...)
## S3 method for class 'mol2'
atom.select(mol, string=NULL,
                           eleno = NULL, elena = NULL, elety = NULL,
                           resid = NULL, chain = NULL, resno = NULL,
                           statbit = NULL,
			   operator = "AND", inverse = FALSE,
                           value = FALSE, verbose=FALSE,  ...)

## S3 method for class 'prmtop'
atom.select(prmtop, ...)

## S3 method for class 'select'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> arguments passed to <code>atom.select.pdb</code>,
<code>atom.select.prmtop</code>, or <code>print</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> a structure object of class <code>"pdb"</code>, obtained from
<code>read.pdb</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> a numeric matrix of aligned C-alpha xyz Cartesian
coordinates as obtained with <code>read.fasta.pdb</code> or <code>pdbaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>string</code></td>
<td>
<p> a single selection keyword from <code>calpha</code>
<code>cbeta</code> <code>backbone</code> <code>sidechain</code> <code>protein</code>
<code>nucleic</code> <code>ligand</code> <code>water</code> <code>h</code> or <code>noh</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p> a single element character vector for selecting ‘ATOM’
or ‘HETATM’ record types. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eleno</code></td>
<td>
<p> a numeric vector of element numbers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>elena</code></td>
<td>
<p> a character vector of atom names. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>elety</code></td>
<td>
<p> a character vector of atom names. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resid</code></td>
<td>
<p> a character vector of residue name identifiers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p> a character vector of chain identifiers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p> a numeric vector of residue numbers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>insert</code></td>
<td>
<p> a character vector of insert identifiers. Non-insert 
residues can be selected with <code>NA</code> or ‘’ values. The default value 
of <code>NULL</code> will select both insert and non-insert residues. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>segid</code></td>
<td>
<p> a character vector of segment identifiers. Empty segid
values can be selected with <code>NA</code> or ‘’ values. The default value
of <code>NULL</code> will select both empty and non-empty segment identifiers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>operator</code></td>
<td>
<p> a single element character specifying either the AND
or OR operator by which individual selection components should be
combined. Allowed values are ‘"AND"’ and ‘"OR"’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE details of the selection are printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inverse</code></td>
<td>
<p> logical, if TRUE the inversed selection is retured
(i.e. all atoms NOT in the selection). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>value</code></td>
<td>
<p> logical, if FALSE, vectors containing the (integer) indices of
the matches determined by <code>atom.select</code> are returned, and if TRUE,
a <code>pdb</code> object containing the matching atoms themselves is
returned. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mol</code></td>
<td>
<p> a structure object of class <code>"mol2"</code>, obtained from
<code>read.mol2</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>statbit</code></td>
<td>
<p> a character vector of statbit identifiers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prmtop</code></td>
<td>
<p> a structure object of class <code>"prmtop"</code>, obtained from
<code>read.prmtop</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a atom.select object as obtained from
<code>atom.select</code>.  </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function allows for the selection of atom and coordinate data
corresponding to the intersection of various input criteria.
</p>
<p>Input selection criteria include selection <code>string</code> keywords (such as
<code>"calpha"</code>, <code>"backbone"</code>, <code>"sidechain"</code>,  <code>"protein"</code>,
<code>"nucleic"</code>, <code>"ligand"</code>, etc.) and individual named
selection components (including ‘chain’,  ‘resno’,
‘resid’, ‘elety’ etc.).
</p>
<p>For example, <code>atom.select(pdb, "calpha")</code> will return indices for
all C-alpha (CA) atoms found in protein residues in the <code>pdb</code>
object, <code>atom.select(pdb, "backbone")</code> will return indices
for all protein N,CA,C,O atoms, and <code>atom.select(pdb, "cbeta")</code>
for all protein N,CA,C,O,CB atoms. 
</p>
<p>Note that keyword <code>string</code> shortcuts can be combined with individual
selection components, e.g. <code>atom.select(pdb, "protein", chain="A")</code>
will select all protein atoms found in chain A.
</p>
<p>Selection criteria are combined according to the provided
<code>operator</code> argument. The default operator <code>AND</code> (or <code>&amp;</code>) will
combine by intersection while <code>OR</code> (or <code>|</code>) will take the union.
</p>
<p>For example, <code>atom.select(pdb, "protein", elety=c("N", "CA", "C"),
  resno=65:103)</code> will select the N, CA, C atoms in the protein residues
65 through 103, while <code>atom.select(pdb, "protein", resid="ATP",
  operator="OR")</code> will select all protein atoms as well as any ATP
residue(s).
</p>
<p>Other <code>string</code> shortcuts include:
<code>"calpha"</code>, <code>"back"</code>, <code>"backbone"</code>, <code>"cbeta"</code>,
<code>"protein"</code>, <code>"notprotein"</code>, <code>"ligand"</code>,
<code>"water"</code>, <code>"notwater"</code>, <code>"h"</code>, <code>"noh"</code>,
<code>"nucleic"</code>, and <code>"notnucleic"</code>.
</p>
<p>In addition, the <code>combine.select</code> function can further combine atom
selections using ‘AND’, ‘OR’, or ‘NOT’ logical
operations.
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"select"</code> with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>atom</code></td>
<td>
<p> a numeric matrix of atomic indices. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xyz </code></td>
<td>
<p> a numeric matrix of xyz indices. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call </code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Protein atoms are defined as any atom in a residue matching the
residue name in the attached <code>aa.table</code> data frame. See
<code>aa.table$aa3</code> for a complete list of residue names.
</p>
<p>Nucleic atoms are defined as all atoms found in residues with names A,
U, G, C, T, I, DA, DU, DG, DC, DT, or DI.
</p>
<p>Water atoms/residues are defined as those with residue names H2O, OH2,
HOH, HHO, OHH, SOL, WAT, TIP, TIP, TIP3, or TIP4.
</p>


<h3>Author(s)</h3>

<p> Barry Grant, Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>read.pdb</code>, <code>as.select</code>,
<code>combine.select</code>, <code>trim.pdb</code>,
<code>write.pdb</code>, <code>read.prmtop</code>,
<code>read.crd</code>, <code>read.dcd</code>,
<code>read.ncdf</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
##- PDB example
# Read a PDB file
pdb &lt;- read.pdb( system.file("examples/1hel.pdb", package="bio3d") )

# Select protein atoms of chain A
atom.select(pdb, "protein", chain="A")

# Select all atoms except from the protein
atom.select(pdb, "protein", inverse=TRUE, verbose=TRUE)

# Select all C-alpha atoms with residues numbers between 43 and 54
sele &lt;- atom.select(pdb, "calpha", resno=43:54, verbose=TRUE)

# Access the PDB data with the selection indices
print( pdb$atom[ sele$atom, "resid" ] )
print( pdb$xyz[ sele$xyz ] )

# Trim PDB to selection
ca.pdb &lt;- trim.pdb(pdb, sele)

## Not run: 

##- PRMTOP example
prmtop &lt;- read.prmtop(system.file("examples/crambin.prmtop", package="bio3d"))

## Atom selection
ca.inds &lt;- atom.select(prmtop, "calpha")


## End(Not run)
</code></pre>


</div>