<div class="container">

<table style="width: 100%;"><tr>
<td>aa123</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Convert Between 1-letter and 3-letter Aminoacid Codes </h2>

<h3>Description</h3>

<p>Convert between one-letter IUPAC aminoacid codes and three-letter PDB
style aminoacid codes.
</p>


<h3>Usage</h3>

<pre><code class="language-R">aa123(aa)
aa321(aa)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>aa</code></td>
<td>
<p> a character vector of individual aminoacid codes. </p>
</td>
</tr></table>
<h3>Details</h3>

<p>Standard conversions will map ‘A’ to ‘ALA’, ‘G’ to
‘GLY’, etc.
Non-standard codes in <code>aa</code> will generate a warning and return
‘UNK’ or ‘X’.
</p>


<h3>Value</h3>

<p>A character vector of aminoacid codes.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>For a description of IUPAC one-letter codes see:<br><a href="https://www.insdc.org/documents/feature_table.html#7.4.3">https://www.insdc.org/documents/feature_table.html#7.4.3</a>
</p>
<p>For more information on PDB residue codes see:<br><a href="http://ligand-expo.rcsb.org/ld-search.html">http://ligand-expo.rcsb.org/ld-search.html</a>
</p>


<h3>See Also</h3>

 <p><code>read.pdb</code>, <code>read.fasta</code>, <code>pdbseq</code> </p>


<h3>Examples</h3>

<pre><code class="language-R"># Simple conversion
aa123(c("D","L","A","G","S","H"))
aa321(c("ASP", "LEU", "ALA", "GLY", "SER", "HIS"))

## Not run: 
# Extract sequence from a PDB file's ATOM and SEQRES cards
pdb &lt;- read.pdb("1BG2") 
s &lt;- aa321(pdb$seqres)                   # SEQRES
a &lt;- aa321(pdb$atom[pdb$calpha,"resid"]) # ATOM

# Write both sequences to a fasta file
write.fasta(alignment=seqbind(s,a), id=c("seqres","atom"), file="eg2.fa")

# Alternative approach for ATOM sequence extraction
pdbseq(pdb)
pdbseq(pdb, aa1=FALSE )

## End(Not run)
</code></pre>


</div>