<div class="container">

<table style="width: 100%;"><tr>
<td>write.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Write FASTA Formated Sequences </h2>

<h3>Description</h3>

<p>Write aligned or un-aligned sequences to a FASTA format file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write.fasta(alignment=NULL, ids=NULL, seqs=alignment$ali, gap=TRUE, file, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p>  an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p> a vector of sequence names to serve as sequence identifers </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p> an sequence or alignment character matrix or vector with a row per sequence </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gap</code></td>
<td>
<p> logical, if FALSE gaps will be removed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> name of output file.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p> logical, if TRUE output will be appended to
<code>file</code>; otherwise, it will overwrite the contents of
<code>file</code>. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Called for its effect.
</p>


<h3>Note</h3>

<p>For a description of FASTA format see:
<a href="https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml">https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml</a>.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# PDB server connection required - testing excluded

## Read a PDB file
pdb &lt;- read.pdb("1bg2")

## Extract sequence from PDB file
s &lt;- aa321(pdb$seqres)                   # SEQRES
a &lt;- aa321(pdb$atom[pdb$calpha,"resid"]) # ATOM

## Write simple fasta file
#write.fasta( seqs=seqbind(s,a), file="eg.fa")
#write.fasta( ids=c("seqres","atom"), seqs=seqbind(s,a), file="eg.fa" )

outfile1 = file.path(tempdir(), "eg.fa")
write.fasta(list( id=c("seqres"),ali=s ), file = outfile1)
write.fasta(list( id=c("atom"),ali=a ), file = outfile1, append=TRUE)

## Align seqres and atom records
#seqaln(seqbind(s,a))

## Read alignment
aln&lt;-read.fasta(system.file("examples/kif1a.fa",package="bio3d"))

## Cut all but positions 130 to 245
aln$ali=aln$ali[,130:245]

outfile2 = file.path(tempdir(), "eg2.fa")
write.fasta(aln, file = outfile2)

invisible( cat("\nSee the output files:", outfile1, outfile2, sep="\n") )

</code></pre>


</div>