<div class="container">

<table style="width: 100%;"><tr>
<td>read.fasta.pdb</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Read Aligned Structure Data </h2>

<h3>Description</h3>

<p>Read aligned PDB structures and store their C-alpha atom data, including xyz
coordinates, residue numbers, residue type and B-factors.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read.fasta.pdb(aln, prefix = "", pdbext = "", fix.ali = FALSE,
             pdblist=NULL, ncore = 1, nseg.scale = 1, progress = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> an alignment data structure obtained with
<code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prefix</code></td>
<td>
<p> prefix to aln$id to locate PDB files. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdbext</code></td>
<td>
<p> the file name extention of the PDB files. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fix.ali</code></td>
<td>
<p> logical, if TRUE check consistence between <code>$ali</code> 
and <code>$resno</code>, and correct <code>$ali</code> if they don't match. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdblist</code></td>
<td>
<p> an optional list of <code>pdb</code> objects with sequence
corresponding to the alignments in <code>aln</code>. Primarily used
through function <code>pdbaln</code> when the PDB objects already exists
(avoids reading PDBs from file). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ncore </code></td>
<td>
<p> number of CPU cores used to do the calculation.
<code>ncore&gt;1</code> requires package ‘parallel’ installed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nseg.scale </code></td>
<td>
<p> split input data into specified number of segments
prior to running multiple core calculation. See <code>fit.xyz</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>progress</code></td>
<td>
<p> progress bar for use with shiny web app. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> other parameters for <code>read.pdb</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The input <code>aln</code>, produced with <code>read.fasta</code>, must
have identifers (i.e. sequence names) that match the PDB file
names. For example the sequence corresponding to the structure
“1bg2.pdb” should have the identifer ‘1bg2’. See
examples below.
</p>
<p>Sequence miss-matches will generate errors.  Thus, care should be taken
to ensure that the sequences in the alignment match the sequences in
their associated PDB files.
</p>


<h3>Value</h3>

<p>Returns a list of class <code>"pdbs"</code> with the following five
components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p>numeric matrix of aligned C-alpha coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p>character matrix of aligned residue numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>numeric matrix of aligned B-factor values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p>character matrix of aligned chain identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>character vector of PDB sequence/structure names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p>character matrix of aligned sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resid</code></td>
<td>
<p>character matrix of aligned 3-letter residue names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sse</code></td>
<td>
<p>character matrix of aligned helix and strand secondary
structure elements as defined in each PDB file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>The sequence character ‘X’ is useful for masking unusual
or unknown residues, as it can match any other residue type.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>read.fasta</code>, <code>read.pdb</code>,
<code>core.find</code>, <code>fit.xyz</code>,
<code>read.all</code>, <code>pymol.pdbs</code>  </p>


<h3>Examples</h3>

<pre><code class="language-R">
# Redundant testing excluded

# Read sequence alignment
file &lt;- system.file("examples/kif1a.fa",package="bio3d")
aln  &lt;- read.fasta(file)

# Read aligned PDBs
pdbs &lt;- read.fasta.pdb(aln)

# Structure/sequence names/ids
basename( pdbs$id )

# Alignment positions 335 to 339
pdbs$ali[,335:339]
pdbs$resid[,335:339]
pdbs$resno[,335:339]
pdbs$b[,335:339]

# Alignment C-alpha coordinates for these positions
pdbs$xyz[, atom2xyz(335:339)]

# See 'fit.xyz()' function for actual coordinate superposition
#  e.g. fit to first structure
# xyz &lt;- fit.xyz(pdbs$xyz[1,], pdbs)
# xyz[, atom2xyz(335:339)]


</code></pre>


</div>