<div class="container">

<table style="width: 100%;"><tr>
<td>pfam</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Download Pfam FASTA Sequence Alignment </h2>

<h3>Description</h3>

<p>Downloads FASTA sequence alignment from the Pfam database.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pfam(id, alignment = "seed", verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> the Pfam familiy identifier (e.g ‘Piwi’) or accession
(e.g. ‘PF02171’). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p> the alignment type. Allowed values are:
‘seed’, ‘ncbi’, ‘full’,
‘metagenomics’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE details of the download process
is printed. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is a basic function to download a multiple sequence alignment for
a protein family from the Pfam database. 
</p>


<h3>Value</h3>

<p>A ‘fasta’ object with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali </code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids </code></td>
<td>
<p> sequence names as identifiers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call </code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>Full more information on the Pfam database:<br><a href="http://pfam.xfam.org">http://pfam.xfam.org</a>
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>,
<code>hmmer</code>, <code>get.seq</code>,
<code>uniprot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# PFAM server connection required - testing excluded

aln &lt;- pfam("piwi")
aln &lt;- pfam("PF02171")

seq &lt;- get.seq("1rx2_A", outfile = tempfile())
hmm &lt;- hmmer(seq, type="hmmscan", db="pfam")
aln &lt;- pfam(hmm$hit.tbl$acc[1])

# Or much more simply for RCSB PDB entries:
acc &lt;- pdb.pfam("1rx2_A", compact=FALSE)$pfamAcc
aln &lt;- pfam(acc)

## End(Not run)
</code></pre>


</div>