<div class="container">

<table style="width: 100%;"><tr>
<td>pdb2aln</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Align a PDB structure to an existing alignment </h2>

<h3>Description</h3>

<p>Extract sequence from a PDB object and align it to 
an existing multiple sequence alignment that you wish 
keep intact.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pdb2aln(aln, pdb, id="seq.pdb", aln.id=NULL, file="pdb2aln.fa", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>,
<code>read.fasta.pdb</code>, and <code>seqaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> the PDB object to be added to <code>aln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> name for the PDB sequence in the generated new alignment. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aln.id</code></td>
<td>
<p> id of the sequence in <code>aln</code> that is close to the 
sequence from <code>pdb</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> output file name for writing the generated new alignment. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to <code>seqaln</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The basic effect of this function is to add a PDB sequence to an existing
alignement. In this case, the function is simply a wrapper of 
<code>seq2aln</code>.
</p>
<p>The more advanced (and also more useful) effect is giving complete mappings
from the column indices of the original alignment (<code>aln$ali</code>) to 
atomic indices of equivalent C-alpha atoms in the <code>pdb</code>. These mappings
are stored in the output list (see below 'Value' section). This feature
is better illustrated in the function <code>pdb2aln.ind</code>, which 
calls <code>pdb2aln</code> and directly returns atom selections given a set of 
alignment positions. (See <code>pdb2aln.ind</code> for details. ) 
</p>
<p>When <code>aln.id</code> is provided, the function will do pairwise alignment 
between the sequence from <code>pdb</code> and the sequence in <code>aln</code> 
with id matching <code>aln.id</code>. This is the best way to use the
function if the protein has an identical or very similar sequence 
to one of the sequences in <code>aln</code>.
</p>


<h3>Value</h3>

<p>Return a list object of the class 'fasta' containing three components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ref</code></td>
<td>
<p> an integer 2xN matrix, where N is the number of columns of
the new alignment <code>ali</code>. The first row contains the column indices of 
the original alignment <code>aln$ali</code>. The second row contains atomic 
indices of equivalent C-alpha atoms in <code>pdb</code>. Gaps in the new 
alignement are indicated by NAs. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Xin-Qiu Yao &amp; Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>seqaln</code>, <code>seq2aln</code>, 
<code>seqaln.pair</code>, <code>pdb2aln.ind</code></p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##--- Read aligned PDB coordinates (CA only)
aln  &lt;- read.fasta(system.file("examples/kif1a.fa",package="bio3d"))
pdbs &lt;- read.fasta.pdb(aln)

##--- Read PDB coordinate for a new structure (all atoms)
id &lt;- get.pdb("2kin", URLonly=TRUE)
pdb &lt;- read.pdb(id)

# add pdb to the alignment 
naln &lt;- pdb2aln(aln=pdbs, pdb=pdb, id=id)
naln


## End(Not run)

</code></pre>


</div>