<div class="container">

<table style="width: 100%;"><tr>
<td>conserv</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Score Residue Conservation At Each Position in an Alignment </h2>

<h3>Description</h3>

<p>Quantifies residue conservation in a given protein sequence alignment
by calculating the degree of amino acid variability in each column of
the alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">conserv(x, method = c("similarity","identity","entropy22","entropy10"),
        sub.matrix = c("bio3d", "blosum62", "pam30", "other"),
        matrix.file = NULL, normalize.matrix = TRUE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p> the conservation assesment method. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sub.matrix</code></td>
<td>
<p> a matrix to score conservation. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>matrix.file</code></td>
<td>
<p> a file name of an arbitary user matrix. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>normalize.matrix</code></td>
<td>
<p> logical, if TRUE the matrix is normalized
pior to assesing conservation. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>To assess the level of sequence conservation at each position in an
alignment, the “similarity”, “identity”, and
“entropy” per position can be calculated.
</p>
<p>The “similarity” is defined as the average of the similarity
scores of all pairwise residue comparisons for that position in the
alignment, where the similarity score between any two residues is the
score value between those residues in the chosen substitution matrix
“sub.matrix”.
</p>
<p>The “identity” i.e. the preference for a specific amino acid to
be found at a certain position, is assessed by averaging the identity
scores resulting from all possible pairwise comparisons at that position
in the alignment, where all identical residue comparisons are given a
score of 1 and all other comparisons are given a value of 0.
</p>
<p>“Entropy” is based on Shannons information entropy. See the
<code>entropy</code> function for further details.
</p>
<p>Note that the returned scores are normalized so that conserved columns
score 1 and diverse columns score 0.
</p>


<h3>Value</h3>

<p>Returns a numeric vector of scores
</p>


<h3>Note</h3>

<p>Each of these conservation scores has particular strengths and
weaknesses.  For example, entropy elegantly captures amino acid
diversity but fails to account for stereochemical similarities. By
employing a combination of scores and taking the union of their
respective conservation signals we expect to achieve a more
comprehensive analysis of sequence conservation (Grant, 2007).
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
Grant, B.J. et al. (2007) <em>J. Mol. Biol.</em> <b>368</b>, 1231–1248.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Read an example alignment
aln &lt;- read.fasta(system.file("examples/hivp_xray.fa",package="bio3d"))

## Score conservation
conserv(x=aln$ali, method="similarity", sub.matrix="bio3d")
##conserv(x=aln$ali,method="entropy22", sub.matrix="other")

</code></pre>


</div>