<div class="container">

<table style="width: 100%;"><tr>
<td>hmmer</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> HMMER Sequence Search </h2>

<h3>Description</h3>

<p>Perform a HMMER search against the PDB, NR, swissprot or other sequence
and structure databases.
</p>


<h3>Usage</h3>

<pre><code class="language-R">hmmer(seq, type="phmmer", db = NULL, verbose = TRUE, timeout = 90)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq</code></td>
<td>
<p> a multi-element character vector containing the query
sequence. Alternatively a ‘fasta’ object as obtained
from functions <code>get.seq</code> or <code>read.fasta</code> can be
provided. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p> character string specifying the ‘HMMER’ job
type. Current options are ‘phmmer’, ‘hmmscan’,
‘hmmsearch’, and ‘jackhmmer’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p> character string specifying the database to search.
Current options are ‘pdb’, ‘nr’, ‘swissprot’,
‘pfam’, etc. See ‘details’ for a complete list. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE details of the download process
is printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>timeout</code></td>
<td>
<p> integer specifying the number of seconds to wait for
the blast reply before a time out occurs. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function employs direct HTTP-encoded requests to the HMMER web
server. HMMER can be used to search sequence databases for homologous
protein sequences. The HMMER server implements methods using
probabilistic models called profile hidden Markov models (profile
HMMs).
</p>
<p>There are currently four types of HMMER search to perform:
</p>
<p>- ‘phmmer’: protein sequence vs protein sequence database.<br>
(input argument <code>seq</code> must be a sequence). 
</p>
<p>Allowed options for <code>type</code> includes: 
‘env_nr’, ‘nr’, ‘refseq’, ‘pdb’,
‘rp15’, ‘rp35’, ‘rp55’, ‘rp75’,
‘swissprot’, ‘unimes’, ‘uniprotkb’,
‘uniprotrefprot’, ‘pfamseq’. 
</p>
<p>- ‘hmmscan’: protein sequence vs profile-HMM database.<br>
(input argument <code>seq</code> must be a sequence). 
</p>
<p>Allowed options for <code>type</code> includes:
‘pfam’, ‘gene3d’,  ‘superfamily’, ‘tigrfam’.
</p>
<p>- ‘hmmsearch’: protein alignment/profile-HMM vs protein sequence
database.<br>
(input argument <code>seq</code> must be an alignment). 
</p>
<p>Allowed options for <code>type</code> includes:
‘pdb’, ‘swissprot’.
</p>
<p>- ‘jackhmmer’: iterative search vs protein sequence database.<br>
(input argument <code>seq</code> must be an alignment).
‘jackhmmer’ functionality incomplete!!
</p>
<p>Allowed options for <code>type</code> includes:
‘env_nr’, ‘nr’, ‘refseq’, ‘pdb’,
‘rp15’, ‘rp35’, ‘rp55’, ‘rp75’,
‘swissprot’, ‘unimes’, ‘uniprotkb’,
‘uniprotrefprot’, ‘pfamseq’. 
</p>
<p>More information can be found at the HMMER website:<br><a href="http://hmmer.org">http://hmmer.org</a>
</p>


<h3>Value</h3>

<p>A list object with components ‘hit.tbl’ and
‘url’. ‘hit.tbl’ is a data frame with multiple
components depending on the selected job ‘type’. Frequently
reported fields include: 
</p>
<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p> a character vector containing the name of the target. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acc</code></td>
<td>
<p> a character vector containing the accession identifier of
the target. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acc2</code></td>
<td>
<p> a character vector containing secondary accession of the
target. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb.id</code></td>
<td>
<p> same as ‘acc’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> a character vector containing Identifier of the target </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>desc</code></td>
<td>
<p> a character vector containing entry description. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>score</code></td>
<td>
<p> a numeric vector containing bit score of the sequence (all
domains, without correction). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bitscore</code></td>
<td>
<p> same as ‘score’. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pvalue</code></td>
<td>
<p> a numeric vector containing the P-value of the score. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>evalue</code></td>
<td>
<p> a numeric vector containing the E-value of the score. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mlog.evalue </code></td>
<td>
<p> a numeric vector containing minus the natural log
of the E-value. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nregions</code></td>
<td>
<p> a numeric vector containing Number of regions
evaluated. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nenvelopes</code></td>
<td>
<p> a numeric vector containing the number of envelopes
handed over for domain definition, null2, alignment, and scoring. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ndom</code></td>
<td>
<p> a numeric vector containing the total number of domains
identified in this sequence. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nreported</code></td>
<td>
<p> a numeric vector containing the number of domains
satisfying reporting thresholding. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nincluded</code></td>
<td>
<p> a numeric vector containing the number of domains
satisfying inclusion thresholding. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>taxid</code></td>
<td>
<p> a character vector containing The NCBI taxonomy identifier
of the target (if applicable). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>species</code></td>
<td>
<p> a character vector containing the species name. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kg</code></td>
<td>
<p> a character vector containing the kingdom of life that the
target belongs to - based on placing in the NCBI taxonomy tree. </p>
</td>
</tr>
</table>
<p>More details can be found at the HMMER website:<br><a href="https://www.ebi.ac.uk/Tools/hmmer/help/api">https://www.ebi.ac.uk/Tools/hmmer/help/api</a>
</p>


<h3>Note</h3>

<p>Note that the chained ‘pdbs’ HMMER field (used for redundant
PDBs) is included directly into the result list (applies only when
<code>db='pdb'</code>). In this case, the ‘name’ component of the
target contains the parent (non redundant) entry, and the ‘acc’
component the chained PDB identifiers. The search results will therefore
provide duplicated PDB identifiers for component <code>$name</code>, while
<code>$acc</code> should be unique. 
</p>


<h3>Note</h3>

<p>Online access is required to query HMMER services.
</p>


<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>Finn, R.D. et al. (2011) <em>Nucl. Acids Res.</em> <b>39</b>, 29–37.
Eddy, S.R. (2011) <em>PLoS Comput Biol</em> <b>7</b>(10): e1002195.
</p>
<p>See also the ‘HMMER’ website:<br><a href="http://hmmer.org">http://hmmer.org</a>
</p>


<h3>See Also</h3>

<p><code>blast.pdb</code>, <code>plot.blast</code>, 
<code>seqaln</code>, <code>get.seq</code>,
<code>pfam</code>, <code>uniprot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# HMMER server connection required - testing excluded

##- PHMMER
seq &lt;- get.seq("2abl_A", outfile=tempfile())
res &lt;- hmmer(seq, db="pdb")

##- HMMSCAN
fam &lt;- hmmer(seq, type="hmmscan", db="pfam")
pfam.aln &lt;- pfam(fam$hit.tbl$acc[1])

##- HMMSEARCH
hmm &lt;- hmmer(pfam.aln, type="hmmsearch", db="pdb")
unique(hmm$hit.tbl$species)
hmm$hit.tbl$acc


## End(Not run)
</code></pre>


</div>