<div class="container">

<table style="width: 100%;"><tr>
<td>write.pir</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Write PIR Formated Sequences </h2>

<h3>Description</h3>

<p>Write aligned or un-aligned sequences to a PIR format file.
</p>


<h3>Usage</h3>

<pre><code class="language-R">write.pir(alignment=NULL, ids=NULL, seqs=alignment$ali, 
   pdb.file = NULL, chain.first = NULL, resno.first = NULL,
   chain.last = NULL, resno.last = NULL, file, append = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p>  an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p> a vector of sequence names to serve as sequence identifers </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqs</code></td>
<td>
<p> an sequence or alignment character matrix or vector with a row per sequence </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb.file</code></td>
<td>
<p> a vector of pdb filenames; For sequence, provide "". </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain.first</code></td>
<td>
<p> a vector of chain id for the first residue. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno.first</code></td>
<td>
<p> a vector of residue number for the first residue. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain.last</code></td>
<td>
<p> a vector of chain id for the last residue. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno.last</code></td>
<td>
<p> a vector of residue number for the last residue. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> name of output file.  </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>append</code></td>
<td>
<p> logical, if TRUE output will be appended to
<code>file</code>; otherwise, it will overwrite the contents of
<code>file</code>. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Called for its effect.
</p>


<h3>Note</h3>

<p>PIR is required format for input alignment file to use Modeller. 
For a description of PIR format see:
<a href="https://salilab.org/modeller/manual/node501.html">https://salilab.org/modeller/manual/node501.html</a>.
</p>


<h3>Author(s)</h3>

<p> Xin-Qiu Yao </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code>, 
<code>write.fasta</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
# Needs MUSCLE installed - testing excluded

if(check.utility("muscle")) {

## Generate an input file for structural modeling of 
## transducin G-alpha subunit using the template 3SN6_A

## Read transducin alpha subunit sequence
seq &lt;- get.seq("P04695", outfile = tempfile())

## Read structure template
path = tempdir()
pdb.file &lt;- get.pdb("3sn6_A", path = path, split = TRUE)
pdb &lt;- read.pdb(pdb.file)

## Build an alignment between template and target
aln &lt;- seqaln(seqbind(pdbseq(pdb), seq), id = c("3sn6_A", seq$id), outfile = tempfile())

## Write PIR format alignment file
outfile = file.path(tempdir(), "eg.pir")
write.pir(aln, pdb.file = c(pdb.file, ""), file = outfile)

invisible( cat("\nSee the output file:", outfile, sep = "\n") )

}

</code></pre>


</div>