<div class="container">

<table style="width: 100%;"><tr>
<td>gap.inspect</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Alignment Gap Summary </h2>

<h3>Description</h3>

<p>Report the number of gaps per sequence and per position for a given alignment.
</p>


<h3>Usage</h3>

<pre><code class="language-R">gap.inspect(x)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a matrix or an alignment data structure obtained from
<code>read.fasta</code> or <code>read.fasta.pdb</code>.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>Reports the number of gap characters per row (i.e. sequence) and
per column (i.e. position) for a given <code>alignment</code>. In addition,
the indices for gap and non-gap containing coloums are returned along
with a binary matrix indicating the location of gap positions. 
</p>


<h3>Value</h3>

<p>Returns a list object with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>row </code></td>
<td>
<p>a numeric vector detailing the number of gaps per row
(i.e. sequence).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col </code></td>
<td>
<p>a numeric vector detailing the number of gaps per column
(i.e. position).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>t.inds </code></td>
<td>
<p> indices for gap containing coloums </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>f.inds </code></td>
<td>
<p> indices for non-gap containing coloums</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bin </code></td>
<td>
<p>a binary numeric matrix with the same dimensions as the
<code>alignment</code>, with 0 at non-gap positions and 1 at gap
positions.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>During alignment, gaps are introduced into sequences that are believed
to have undergone deletions or insertions with respect to other
sequences in the alignment.  These gaps, often referred to as
indels, can be represented with ‘NA’, a ‘-’ or
‘.’ character.
</p>
<p>This function gives an overview of gap occurrence and may be useful
when considering positions or sequences that could/should be excluded
from further analysis.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code>, <code>read.fasta.pdb</code></p>


<h3>Examples</h3>

<pre><code class="language-R">aln &lt;- read.fasta( system.file("examples/hivp_xray.fa",
                   package = "bio3d") )

gap.stats &lt;- gap.inspect(aln$ali)
gap.stats$row # Gaps per sequence
gap.stats$col # Gaps per position
##gap.stats$bin # Binary matrix (1 for gap, 0 for aminoacid)
##aln[,gap.stats$f.inds] # Alignment without gap positions 

plot(gap.stats$col, typ="h", ylab="No. of Gaps") 
</code></pre>


</div>