<div class="container">

<table style="width: 100%;"><tr>
<td>consensus</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Sequence Consensus for an Alignment </h2>

<h3>Description</h3>

<p>Determines the consensus sequence for a given alignment at a given
identity cutoff value.
</p>


<h3>Usage</h3>

<pre><code class="language-R">  consensus(alignment, cutoff = 0.6)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p> an <code>alignment</code> object created by the
<code>read.fasta</code> function or an alignment character matrix. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cutoff</code></td>
<td>
<p> a numeric value beteen 0 and 1, indicating the minimum
sequence identity threshold for determining a consensus amino acid.
Default is 0.6, or 60 percent residue identity. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A vector containing the consensus sequence, where ‘-’ represents
positions with no consensus (i.e. under the <code>cutoff</code>)
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>read.fasta</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">
#-- Read HIV protease alignment
aln &lt;- read.fasta(system.file("examples/hivp_xray.fa",package="bio3d"))

# Generate consensus
con &lt;- consensus(aln)
print(con$seq)

# Plot residue frequency matrix
##png(filename = "freq.png", width = 1500, height = 780)
col &lt;- mono.colors(32)
aa  &lt;- rev(rownames(con$freq))

image(x=1:ncol(con$freq),
      y=1:nrow(con$freq),
      z=as.matrix(rev(as.data.frame(t(con$freq)))),
      col=col, yaxt="n", xaxt="n",
      xlab="Alignment Position", ylab="Residue Type")

# Add consensus along the axis
axis(side=1, at=seq(0,length(con$seq),by=5))
axis(side=2, at=c(1:22), labels=aa)
axis(side=3, at=c(1:length(con$seq)), labels =con$seq)
axis(side=4, at=c(1:22), labels=aa)
grid(length(con$seq), length(aa))
box()

# Add consensus sequence
for(i in 1:length(con$seq)) {
  text(i, which(aa==con$seq[i]),con$seq[i],col="white")
}

# Add lines for residue type separation
abline(h=c(2.5,3.5, 4.5, 5.5, 3.5, 7.5, 9.5,
         12.5, 14.5, 16.5, 19.5), col="gray")


</code></pre>


</div>