<div class="container">

<table style="width: 100%;"><tr>
<td>trim.pdbs</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Filter or Trim a PDBs Object </h2>

<h3>Description</h3>

<p>Trim residues and/or filter out structures from a PDBs object. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'pdbs'
trim(pdbs, row.inds=NULL, col.inds=NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> an object of class <code>pdbs</code> as obtained from
function <code>pdbaln</code> or <code>read.fasta.pdb</code>; a xyz matrix
containing the cartesian coordinates of C-alpha atoms. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>row.inds </code></td>
<td>
<p> a numeric vector of indices pointing to the PDB
structures to keep (rows in the <code>pdbs$ali</code> matrix). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.inds </code></td>
<td>
<p> a numeric vector of indices pointing to the
alignment columns to keep (columns in the <code>pdbs$ali</code> matrix). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Utility function to remove structures, or trim off columns, in a   
‘pdbs’ object. 
</p>


<h3>Value</h3>

<p>Returns an updated ‘pdbs’ object with the following
components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p>numeric matrix of aligned C-alpha coordinates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p>character matrix of aligned residue numbers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p>numeric matrix of aligned B-factor values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chain</code></td>
<td>
<p>character matrix of aligned chain identifiers.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p>character vector of PDB sequence/structure names.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p>character matrix of aligned sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p> the matched call. </p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p> Lars Skjaerven </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

<p><code>pdbaln</code>, <code>gap.inspect</code>,
<code>read.fasta</code>,<code>read.fasta.pdb</code>,
<code>trim.pdb</code>,
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Fetch PDB files and split to chain A only PDB files
ids &lt;- c("1a70_A", "1czp_A", "1frd_A", "1fxi_A", "1iue_A", "1pfd_A")
raw.files &lt;- get.pdb(ids, path = "raw_pdbs")
files &lt;- pdbsplit(raw.files, ids, path = "raw_pdbs/split_chain")

## Sequence Alignement, and connectivity check
pdbs &lt;- pdbaln(files)

cons &lt;- inspect.connectivity(pdbs)

## omit files with missing residues
trim.pdbs(pdbs, row.inds=which(cons))

## End(Not run)
</code></pre>


</div>