<div class="container">

<table style="width: 100%;"><tr>
<td>get.seq</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Download FASTA Sequence Files </h2>

<h3>Description</h3>

<p>Downloads FASTA sequence files from the NCBI nr, SWISSPROT/UNIPROT,
OR RCSB PDB databases.
</p>


<h3>Usage</h3>

<pre><code class="language-R">get.seq(ids, outfile = "seqs.fasta", db = "nr", verbose = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ids</code></td>
<td>
<p> A character vector of one or more appropriate database
codes/identifiers of the files to be downloaded. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>outfile</code></td>
<td>
<p> A single element character vector specifying the name
of the local file to which sequences will be written. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p> A single element character vector specifying the database
from which sequences are to be obtained. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE URL details of the download process
are printed. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This is a basic function to automate sequence file download from the 
databases including NCBI nr, SWISSPROT/UNIPROT, and RCSB PDB.
</p>


<h3>Value</h3>

<p>If all files are successfully downloaded a list object with two
components is returned:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali </code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ids </code></td>
<td>
<p> sequence names as identifiers.</p>
</td>
</tr>
</table>
<p>This is similar to that returned by <code>read.fasta</code>. However,
if some files were not successfully downloaded then a vector detailing
which ids were not found is returned.
</p>


<h3>Note</h3>

  
<p>For a description of FASTA format see:
<a href="https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml">https://www.ncbi.nlm.nih.gov/BLAST/blastcgihelp.shtml</a>.
When reading alignment files, the dash ‘-’ is interpreted as
the gap character.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>


<h3>See Also</h3>

 <p><code>blast.pdb</code>, <code>read.fasta</code>,
<code>read.fasta.pdb</code>, <code>get.pdb</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
## Sequence identifiers (GI or PDB codes e.g. from blast.pdb etc.)
get.seq( c("P01112", "Q61411", "P20171") )

#aa &lt;-get.seq( c("4q21", "5p21") )
#aa$id
#aa$ali

## End(Not run)
</code></pre>


</div>