<div class="container">

<table style="width: 100%;"><tr>
<td>torsion.xyz</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Calculate Torsion/Dihedral Angles </h2>

<h3>Description</h3>

<p>Defined from the Cartesian coordinates of four successive atoms
(A-B-C-D) the torsion or dihedral angle is calculated about an axis
defined by the middle pair of atoms (B-C).
</p>


<h3>Usage</h3>

<pre><code class="language-R">torsion.xyz(xyz, atm.inc = 4)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p> a numeric vector of Cartisean coordinates. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>atm.inc</code></td>
<td>
<p> a numeric value indicating the number of atoms to
increment by between successive torsion evaluations (see below). </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The conformation of a polypeptide or nucleotide chain can be usefully
described in terms of angles of internal rotation around its
constituent bonds.
</p>
<p>If a system of four atoms A-B-C-D is projected onto a plane normal to
bond B-C, the angle between the projection of A-B and the projection
of C-D is described as the torsion angle of A and D about bond B-C.
</p>
<p>By convention angles are measured in the range -180 to +180, rather
than from 0 to 360, with positive values defined to be in the
clockwise direction.
</p>
<p>With <code>atm.inc=1</code>, torsion angles are calculated for each set of
four successive atoms contained in <code>xyz</code> (i.e. moving along one
atom, or three elements of <code>xyz</code>, between sucessive
evaluations).  With <code>atm.inc=4</code>, torsion angles are calculated
for each set of four successive non-overlapping atoms contained in
<code>xyz</code> (i.e. moving along four atoms, or twelve elements of
<code>xyz</code>, between sucessive evaluations).
</p>


<h3>Value</h3>

<p>A numeric vector of torsion angles.
</p>


<h3>Note</h3>

<p> Contributions from Barry Grant. </p>


<h3>Author(s)</h3>

<p> Karim ElSawy </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>torsion.pdb</code>, <code>pca.tor</code>,
<code>wrap.tor</code>, <code>read.pdb</code>, <code>read.dcd</code>.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Calculate torsions for cis &amp; trans conformers
xyz &lt;- rbind(c(0,-0.5,0,1,0,0,1,1,0,0,1.5,0),
             c(0,-0.5,0,1,0,0,1,1,0,2,1.5,0)-3)
cis.tor   &lt;- torsion.xyz( xyz[1,] )
trans.tor &lt;- torsion.xyz( xyz[2,] )
apply(xyz, 1, torsion.xyz)


plot(range(xyz), range(xyz), xlab="", ylab="", typ="n", axes=FALSE)
  apply(xyz, 1, function(x){
    lines(matrix(x, ncol=3, byrow=TRUE), lwd=4)
    points(matrix(x, ncol=3, byrow=TRUE), cex=2.5,
           bg="white", col="black", pch=21) } )
  
  text( t(apply(xyz, 1, function(x){
    apply(matrix(x, ncol=3, byrow=TRUE)[c(2,3),], 2, mean) })),
       labels=c(0,180), adj=-0.5, col="red")


# PDB server connection required - testing excluded

##-- PDB torsion analysis
pdb &lt;- read.pdb("1bg2")
tor &lt;- torsion.pdb(pdb)
## basic Ramachandran plot
plot(tor$phi, tor$psi)

## torsion analysis of a single coordinate vector 
inds &lt;- atom.select(pdb,"calpha")
tor.ca &lt;- torsion.xyz(pdb$xyz[inds$xyz], atm.inc=3)

##-- Compare two PDBs to highlight interesting residues
aln &lt;- read.fasta(system.file("examples/kif1a.fa",package="bio3d"))
m &lt;- read.fasta.pdb(aln)
a &lt;- torsion.xyz(m$xyz[1,],1)
b &lt;- torsion.xyz(m$xyz[2,],1)
## Note the periodicity of torsion angles
d &lt;- wrap.tor(a-b)
plot(m$resno[1,],d, typ="h")



</code></pre>


</div>