<div class="container">

<table style="width: 100%;"><tr>
<td>dssp</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Secondary Structure Analysis with DSSP or STRIDE </h2>

<h3>Description</h3>

<p>Secondary structure assignment according to the method of Kabsch and Sander (DSSP)
or the method of Frishman and Argos (STRIDE).
</p>


<h3>Usage</h3>

<pre><code class="language-R">dssp(...)

## S3 method for class 'pdb'
dssp(pdb, exefile = "dssp", resno=TRUE, full=FALSE, verbose=FALSE, ...)

## S3 method for class 'pdbs'
dssp(pdbs, ...)

## S3 method for class 'xyz'
dssp(xyz, pdb, ...)

stride(pdb, exefile = "stride", resno=TRUE)

## S3 method for class 'sse'
print(x, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> a structure object of class <code>"pdb"</code>, obtained from
<code>read.pdb</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>exefile</code></td>
<td>
<p> file path to the ‘DSSP’ or ‘STRIDE’ program 
on your system (i.e. how is ‘DSSP’ or ‘STRIDE’ invoked). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>resno</code></td>
<td>
<p> logical, if TRUE output is in terms of residue numbers
rather than residue index (position in sequence). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>full</code></td>
<td>
<p> logical, if TRUE bridge pairs and hbonds columns are
parsed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p> logical, if TRUE ‘DSSP’ warning and error
messages are printed. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdbs</code></td>
<td>
<p> a list object of class <code>"pdbs"</code> (obtained with
<code>pdbaln</code> or <code>read.fasta.pdb</code>). </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xyz</code></td>
<td>
<p> a trajectory object of class <code>"xyz"</code>, obtained from
<code>read.ncdf</code>, <code>read.dcd</code>,
<code>read.crd</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> an <code>sse</code> object obtained from <code>dssp.pdb</code> or 
<code>stride</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments to and from functions. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calls the ‘DSSP’ or ‘STRIDE’ program to define 
secondary structure and psi and phi torsion angles.
</p>


<h3>Value</h3>

<p>Returns a list with the following components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>helix</code></td>
<td>
 <p>‘start’, ‘end’, ‘length’,
‘chain’ and ‘type’ of helix, where start and end are
residue numbers or residue index positions depending on the value of
“resno” input argument. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sheet</code></td>
<td>
 <p>‘start’, ‘end’ and ‘length’ of E
type sse, where start and end are residue numbers “resno”. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>turn</code></td>
<td>
 <p>‘start’, ‘end’ and ‘length’ of T
type sse, where start and end are residue numbers “resno”. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phi</code></td>
<td>
<p> a numeric vector of phi angles. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>psi</code></td>
<td>
<p> a numeric vector of psi angles. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>acc</code></td>
<td>
<p> a numeric vector of solvent accessibility. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sse</code></td>
<td>
<p> a character vector of secondary structure type per residue. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>hbonds</code></td>
<td>
<p> a 10 or 16 column matrix containing the bridge pair
records as well as backbone NH–&gt;O and O–&gt;NH H-bond records. 
(Only available for <code>dssp</code> </p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>A system call is made to the ‘DSSP’ or ‘STRIDE’ program, 
which must be installed on your system and in the search path for executables.
See <a href="http://thegrantlab.org/bio3d/articles/online/install_vignette/Bio3D_install.html">http://thegrantlab.org/bio3d/articles/online/install_vignette/Bio3D_install.html</a> for 
instructions of how to install these programs.
</p>
<p>For the <code>hbonds</code> list component the column names can be
used as a convenient means of data access, namely:<br>
Bridge pair 1            “BP1”,<br>
Bridge pair 2            “BP2”,<br>
Backbone H-bond (NH–&gt;O) “NH-O.1”,<br>
H-bond energy of NH–&gt;O  “E1”,<br>
Backbone H-bond (O–&gt;NH) “O-HN.1”,<br>
H-bond energy of O–&gt;NH  “E2”,<br>
Backbone H-bond (NH–&gt;O) “NH-O.2”,<br>
H-bond energy of NH–&gt;O  “E3”,<br>
Backbone H-bond (O–&gt;NH) “O-HN.2”,<br>
H-bond energy of O–&gt;NH  “E4”.<br></p>
<p>If ‘resno=TRUE’ the following additional columns are included:<br>
Chain ID of resno “BP1”:    “ChainBP1”,<br>
Chain ID of resno “BP2”:    “ChainBP2”,<br>
Chain ID of resno “O-HN.1”: “Chain1”,<br>
Chain ID of resno “NH-O.2”: “Chain2”,<br>
Chain ID of resno “O-HN.1”: “Chain3”,<br>
Chain ID of resno “NH-O.2”: “Chain4”.
</p>


<h3>Author(s)</h3>

<p> Barry Grant, Lars Skjaerven (dssp.pdbs) </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>‘DSSP’ is the work of Kabsch and Sander:
Kabsch and Sander (1983) <em>Biopolymers.</em>  <b>12</b>, 2577–2637.
</p>
<p>For information on obtaining ‘DSSP’, see:<br><a href="https://swift.cmbi.umcn.nl/gv/dssp/">https://swift.cmbi.umcn.nl/gv/dssp/</a>.
</p>
<p>‘STRIDE’ is the work of Frishman and Argos:
Frishman and Argos (1995) <em>Proteins.</em>  <b>3</b>, 566–579.
</p>
<p>For information on obtaining the ‘STRIDE’ program, see:<br><a href="http://webclu.bio.wzw.tum.de/stride/">http://webclu.bio.wzw.tum.de/stride/</a>,
or copy it from an installation of VMD.
</p>


<h3>See Also</h3>

<p><code>read.pdb</code>,
<code>torsion.pdb</code>, <code>torsion.xyz</code>,
<code>plot.bio3d</code>,
</p>
<p><code>read.ncdf</code>, <code>read.dcd</code>,
<code>read.prmtop</code>, <code>read.crd</code>,
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##- PDB example
# Read a PDB file
pdb &lt;- read.pdb("1bg2")
sse &lt;- dssp(pdb)
sse2 &lt;- stride(pdb)

## Short summary
sse
sse2

# Helix data
sse$helix

# Precent SSE content
sum(sse$helix$length)/sum(pdb$calpha) * 100
sum(sse$sheet$length)/sum(pdb$calpha) * 100



##- PDBs example
aln  &lt;- read.fasta( system.file("examples/kif1a.fa",package="bio3d") )
pdbs &lt;- read.fasta.pdb( aln )

## Aligned PDB defined secondary structure
pdbs$sse

## Aligned DSSP defined secondary structure 
sse &lt;- dssp(pdbs)


##- XYZ Trajectory
pdb &lt;- read.pdb("2mda", multi=TRUE)
dssp.xyz(pdb$xyz, pdb)

## Note. for large MD trajectories you may want to skip some frames, e.g.
xyz &lt;- rbind(pdb$xyz, pdb$xyz)       ## dummy trajectory
frames &lt;- seq(1, to=nrow(xyz), by=4) ## frame numbers to examine
ss &lt;- dssp.xyz(xyz[frames, ], pdb)      ## matrix of sse frame x residue


## End(Not run)
</code></pre>


</div>