<div class="container">

<table style="width: 100%;"><tr>
<td>print.fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Printing Sequence Alignments </h2>

<h3>Description</h3>

<p>Print method for fasta and pdbs sequence alignment objects.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'fasta'
print(x, alignment=TRUE, ...)
.print.fasta.ali(x, width = NULL, col.inds = NULL, numbers = TRUE,
         conservation=TRUE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a sequence alignment object as obtained from the functions  
<code>read.fasta</code>, <code>read.fasta.pdb</code>,
<code>pdbaln</code>, <code>seqaln</code>, etc. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alignment</code></td>
<td>
<p> logical, if TRUE the sequence alignment will be
printed to screen. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>width</code></td>
<td>
<p> a single numeric value giving the number of residues per 
printed sequence block. By default this is determined from considering 
alignment identifier widths given a standard 85 column terminal
window. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col.inds</code></td>
<td>
<p> an optional numeric vector that can be used to select 
subsets of alignment positions/columns for printing. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numbers</code></td>
<td>
<p> logical, if TRUE position numbers and a tick-mark every 
10 positions are printed above and below sequence blocks. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>conservation</code></td>
<td>
<p> logical, if TRUE conserved and semi-conserved
columns in the alignment are marked with an ‘*’ and 
‘^’, respectively. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments to ‘.print.fasta.ali’. </p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Called mostly for its effect but also silently returns block
divided concatenated sequence strings as a matrix.  
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.</p>


<h3>See Also</h3>

<p><code>read.fasta</code>, <code>read.fasta.pdb</code>,
<code>pdbaln</code>, <code>seqaln</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
     file &lt;- system.file("examples/kif1a.fa",package="bio3d")
     aln  &lt;- read.fasta(file)
     print(aln)

     # print(aln, col.inds=30:100, numbers=FALSE)
</code></pre>


</div>