<div class="container">

<table style="width: 100%;"><tr>
<td>seq2aln</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Add a Sequence to an Existing Alignmnet </h2>

<h3>Description</h3>

<p>Add one or more sequences to an existing multiple alignment that you
wish to keep intact.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seq2aln(seq2add, aln, id = "seq", file = "aln.fa", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>seq2add</code></td>
<td>
<p> an sequence character vector or an alignment list
object with <code>id</code> and <code>ali</code> components, similar to that
generated by <code>read.fasta</code> and <code>seqaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>
and <code>seqaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> a vector of sequence names to serve as sequence
identifers. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p> name of ‘FASTA’ output file to which alignment
should be written. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to <code>seqaln</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function calls the ‘MUSCLE’ program, to perform a profile
profile alignment, which MUST BE INSTALLED on your system and in the
search path for executables.
</p>


<h3>Value</h3>

<p>A list with two components:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>ali</code></td>
<td>
<p> an alignment character matrix with a row per sequence and
a column per equivalent aminoacid/nucleotide. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>id</code></td>
<td>
<p> sequence names as identifers.</p>
</td>
</tr>
</table>
<h3>Note</h3>

<p>A system call is made to the ‘MUSCLE’ program, which must be
installed on your system and in the search path for executables.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695–2696.
</p>
<p>‘MUSCLE’ is the work of Edgar:
Edgar (2004) <em>Nuc. Acid. Res.</em> <b>32</b>, 1792–1797.
</p>
<p>Full details of the ‘MUSCLE’ algorithm, along with download and
installation instructions can be obtained from:<br><a href="http://www.drive5.com/muscle/">http://www.drive5.com/muscle/</a>.
</p>


<h3>See Also</h3>

 <p><code>seqaln</code>, <code>read.fasta</code>,
<code>read.fasta.pdb</code>, <code>seqbind</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
## Not run: 
aa.1 &lt;- pdbseq( read.pdb("1bg2") )
aa.2 &lt;- pdbseq( read.pdb("3dc4") )
aa.3 &lt;- pdbseq( read.pdb("1mkj") )

aln &lt;- seqaln( seqbind(aa.1,aa.2) )

seq2aln(aa.3, aln)

## End(Not run)

</code></pre>


</div>