<div class="container">

<table style="width: 100%;"><tr>
<td>pdb2aln.ind</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Mapping from alignment positions to PDB atomic indices </h2>

<h3>Description</h3>

<p>Find the best alignment between a PDB structure and an 
existing alignment. Then, given a set of column indices
of the original alignment, returns atom selections of 
equivalent C-alpha atoms in the PDB structure.
</p>


<h3>Usage</h3>

<pre><code class="language-R">pdb2aln.ind(aln, pdb, inds = NULL, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>aln</code></td>
<td>
<p> an alignment list object with <code>id</code> and <code>ali</code>
components, similar to that generated by <code>read.fasta</code>,
<code>read.fasta.pdb</code>, <code>pdbaln</code>, and 
<code>seqaln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pdb</code></td>
<td>
<p> the PDB object to be aligned to <code>aln</code>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inds</code></td>
<td>
<p> a numeric vector containing a subset of column indices of
<code>aln</code>. If NULL, non-gap positions of <code>aln$ali</code> are used. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p> additional arguments passed to <code>pdb2aln</code>. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Call <code>pdb2aln</code> to align the sequence of <code>pdb</code> to <code>aln</code>.
Then, find the atomic indices of C-alpha atoms in <code>pdb</code> that are 
equivalent to <code>inds</code>, the subset of column indices of <code>aln$ali</code>. 
</p>
<p>The function is a rountine utility in a combined analysis of 
molecular dynamics (MD) simulation trajectories and crystallographic 
structures. For example, a typical post-analysis of MD simulation is to 
compare the principal components (PCs) derived from simulation trajectories 
with those derived from crystallographic structures. The C-alpha atoms used
to fit trajectories and do PCA must be the same (or equivalent) to those
used in the analysis of crystallographic structures, e.g. the 'non-gap'  
alignment positions. Call <code>pdb2aln.ind</code> with providing relevant
alignment positions, one can easily get equivalent atom selections 
('select' class objects) for the simulation topology (PDB) file and then
do proper trajectory analysis. 
</p>


<h3>Value</h3>

<p>Returns a list containing two "select" objects:
</p>
<table>
<tr style="vertical-align: top;">
<td><code>a</code></td>
<td>
<p> atom and xyz indices for the alignment. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>b</code></td>
<td>
<p> atom and xyz indices for the PDB. </p>
</td>
</tr>
</table>
<p>Note that if any element of <code>inds</code> has no corresponding CA atom in the 
PDB, the output <code>a$atom</code> and <code>b$atom</code> will be shorter than 
<code>inds</code>, i.e. only indices having equivalent CA atoms are returned. 
</p>


<h3>Author(s)</h3>

<p> Xin-Qiu Yao, Lars Skjaerven &amp; Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>seq2aln</code>, <code>seqaln.pair</code>,
<code>pdb2aln</code> </p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
##--- Read aligned PDB coordinates (CA only)
aln  &lt;- read.fasta(system.file("examples/kif1a.fa",package="bio3d"))
pdbs &lt;- read.fasta.pdb(aln)

##--- Read the topology file of MD simulations
##--- For illustration, here we read another pdb file (all atoms)
pdb &lt;- read.pdb("2kin")

#--- Map the non-gap positions to PDB C-alpha atoms
#pc.inds &lt;- gap.inspect(pdbs$ali)
#npc.inds &lt;- pdb2aln.ind(aln=pdbs, pdb=pdb, inds=pc.inds$f.inds)

#npc.inds$a
#npc.inds$b

#--- Or, map the non-gap positions with a known close sequence in the alignment
#npc.inds &lt;- pdb2aln.ind(aln=pdbs, pdb=pdb, aln.id="1bg2", inds=pc.inds$f.inds)

#--- Map core positions
core &lt;- core.find(pdbs)
core.inds &lt;- pdb2aln.ind(aln=pdbs, pdb=pdb, inds = core$c1A.atom)

core.inds$a
core.inds$b

##--- Fit simulation trajectories to one of the X-ray structures based on
##--- core positions
#xyz &lt;- fit.xyz(pdbs$xyz[1,], pdb$xyz, core.inds$a$xyz, core.inds$b$xyz)

##--- Do PCA of trajectories based on non-gap positions
#pc.traj &lt;- pca(xyz[, npc.inds$b$xyz])


## End(Not run)

</code></pre>


</div>