<div class="container">

<table style="width: 100%;"><tr>
<td>motif.find</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2> Find Sequence Motifs. </h2>

<h3>Description</h3>

<p>Return Position Indices of a Short Sequence Motif Within a Larger Sequence.
</p>


<h3>Usage</h3>

<pre><code class="language-R">motif.find(motif, sequence)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>motif</code></td>
<td>
<p> a character vector of the short sequence motif. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p> a character vector of the larger sequence. </p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The sequence and the motif can be given as a either a multiple or
single element character vector. The dot character and other valid
<code>regexpr</code> characters are allowed in the motif, see examples.
</p>


<h3>Value</h3>

<p>Returns a vector of position indices within the sequence where the
motif was found, see examples.
</p>


<h3>Author(s)</h3>

<p> Barry Grant </p>


<h3>References</h3>

<p>Grant, B.J. et al. (2006) <em>Bioinformatics</em> <b>22</b>, 2695â€“2696.
</p>


<h3>See Also</h3>

 <p><code>regexpr</code>, <code>read.fasta</code>, <code>pdbseq</code> 
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# PDB server connection required - testing excluded

aa.seq &lt;- pdbseq( read.pdb( get.pdb("4q21", URLonly=TRUE) ) )
motif = c("G....GKS")
motif.find(motif, aa.seq)

</code></pre>


</div>