<div class="container">

<table style="width: 100%;"><tr>
<td>BioCircosSNPTrack</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create a track with SNPs to be added to a BioCircos tracklist</h2>

<h3>Description</h3>

<p>SNPs are defined by genomic coordinates and associated with a numerical value
</p>


<h3>Usage</h3>

<pre><code class="language-R">BioCircosSNPTrack(trackname, chromosomes, positions, values,
  colors = "#40B9D4", labels = "", size = 2, shape = "circle",
  opacities = 1, maxRadius = 0.9, minRadius = 0.5, range = 0, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>trackname</code></td>
<td>
<p>The name of the new track.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chromosomes</code></td>
<td>
<p>A vector containing the chromosomes on which each SNP are found.
Values should match the chromosome names given in the genome parameter of the BioCircos function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>positions</code></td>
<td>
<p>A vector containing the coordinates on which each SNP are found.
Values should be inferior to the chromosome lengths given in the genome parameter of the BioCircos function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>values</code></td>
<td>
<p>A vector of numerical values associated with each SNPs, used to determine the 
radial coordinates of each point on the visualization.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>colors</code></td>
<td>
<p>The colors for each point. Can be a RColorBrewer palette name used to
generate one color per point, or a character object or vector of character objects stating RGB values in hexadecimal
format or base R colors. If the vector is shorter than the number of points, values will be repeated.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>labels</code></td>
<td>
<p>One or multiple character objects to label each point.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>size</code></td>
<td>
<p>The size of each point.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shape</code></td>
<td>
<p>Shape of the points. Can be "circle" or "rect".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>opacities</code></td>
<td>
<p>One or multiple opacity values for the points, between 0 and 1.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minRadius, maxRadius</code></td>
<td>
<p>Where the track should begin and end, in proportion of the inner radius of the plot.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>range</code></td>
<td>
<p>Vector of values to be mapped to the minimum and maximum radii of the track.
Default to 0, mapping the minimal and maximal values input in the values parameter.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Ignored</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">BioCircos(BioCircosSNPTrack('SNPTrack', chromosomes = 1:3, positions = 1e+7*2:4, 
  values = 1:3, colors = "Accent", labels = c('A', 'B', 'C')) + BioCircosBackgroundTrack('BGTrack'))

</code></pre>


</div>