<div class="container">

<table style="width: 100%;"><tr>
<td>compose_description_v1.4.2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Compose BioCompute Object - Description Domain (v1.4.2)</h2>

<h3>Description</h3>

<p>Compose BioCompute Object - Description Domain (v1.4.2)
</p>


<h3>Usage</h3>

<pre><code class="language-R">compose_description_v1.4.2(
  keywords = NULL,
  xref = NULL,
  platform = list("Seven Bridges Platform"),
  pipeline_meta = NULL,
  pipeline_prerequisite = NULL,
  pipeline_input = NULL,
  pipeline_output = NULL
)

compose_description(
  keywords = NULL,
  xref = NULL,
  platform = list("Seven Bridges Platform"),
  pipeline_meta = NULL,
  pipeline_prerequisite = NULL,
  pipeline_input = NULL,
  pipeline_output = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>keywords</code></td>
<td>
<p>Character vector. A list of keywords to aid in
searchability and description of the experiment.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>xref</code></td>
<td>
<p>Data frame. A list of the databases and/or ontology IDs
that are cross-referenced in the BCO.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>platform</code></td>
<td>
<p>Character string or list. Reference to a particular deployment
of an existing platform where this BCO can be reproduced.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pipeline_meta</code></td>
<td>
<p>Data frame. Pipeline metadata.
Variables include <code>step_number</code>, <code>name</code>, <code>description</code>,
and <code>version</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pipeline_prerequisite</code></td>
<td>
<p>Data frame. Packages or prerequisites
for running the tools used. Variables include <code>step_number</code>,
<code>name</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pipeline_input</code></td>
<td>
<p>Data frame. Input files for the tools.
Variables include <code>step_number</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pipeline_output</code></td>
<td>
<p>Data frame. Output files for the tools.
Variables include <code>step_number</code>, <code>uri</code>, and <code>access_time</code>.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list of class <code>bco.domain</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">keywords &lt;- c("HCV1a", "Ledipasvir", "antiviral resistance", "SNP", "amino acid substitutions")
xref &lt;- data.frame(
  "namespace" = c("pubchem.compound", "pubmed", "so", "taxonomy"),
  "name" = c("PubChem-compound", "PubMed", "Sequence Ontology", "Taxonomy"),
  "ids" = I(list(
    "67505836",
    "26508693",
    c("SO:000002", "SO:0000694", "SO:0000667", "SO:0000045"),
    "31646"
  )),
  "access_time" = c(
    as.POSIXct("2017-01-20T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-21T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-22T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-23T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

platform &lt;- "Seven Bridges Platform"

pipeline_meta &lt;- data.frame(
  "step_number" = c("1"),
  "name" = c("HIVE-hexagon"),
  "description" = c("Alignment of reads to a set of references"),
  "version" = c("1.3"),
  stringsAsFactors = FALSE
)

pipeline_prerequisite &lt;- data.frame(
  "step_number" = rep("1", 5),
  "name" = c(
    "Hepatitis C virus genotype 1",
    "Hepatitis C virus type 1b complete genome",
    "Hepatitis C virus (isolate JFH-1) genomic RNA",
    "Hepatitis C virus clone J8CF, complete genome",
    "Hepatitis C virus S52 polyprotein gene"
  ),
  "uri" = c(
    "https://www.ncbi.nlm.nih.gov/nuccore/22129792",
    "https://www.ncbi.nlm.nih.gov/nuccore/5420376",
    "https://www.ncbi.nlm.nih.gov/nuccore/13122261",
    "https://www.ncbi.nlm.nih.gov/nuccore/386646758",
    "https://www.ncbi.nlm.nih.gov/nuccore/295311559"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

pipeline_input &lt;- data.frame(
  "step_number" = rep("1", 2),
  "uri" = c(
    "https://example.com/dna.cgi?cmd=objFile&amp;ids=514683",
    "https://example.com/dna.cgi?cmd=objFile&amp;ids=514682"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

pipeline_output &lt;- data.frame(
  "step_number" = rep("1", 2),
  "uri" = c(
    "https://example.com/data/514769/allCount-aligned.csv",
    "https://example.com/data/514801/SNPProfile*.csv"
  ),
  "access_time" = c(
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST"),
    as.POSIXct("2017-01-24T09:40:17", format = "%Y-%m-%dT%H:%M:%S", tz = "EST")
  ),
  stringsAsFactors = FALSE
)

compose_description(
  keywords, xref, platform,
  pipeline_meta, pipeline_prerequisite, pipeline_input, pipeline_output
) %&gt;% convert_json()
</code></pre>


</div>