<div class="container">

<table style="width: 100%;"><tr>
<td>meta.retrieval</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform Meta-Genome Retrieval</h2>

<h3>Description</h3>

<p>Download genomes, proteomes, cds, gff, rna, or assembly stats
files of all species within a kingdom of life.
</p>


<h3>Usage</h3>

<pre><code class="language-R">meta.retrieval(
  db = "refseq",
  kingdom,
  group = NULL,
  type = "genome",
  restart_at_last = TRUE,
  reference = FALSE,
  combine = FALSE,
  path = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "emsembl"</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>kingdom</code></td>
<td>
<p>a character string specifying the kingdom of the organisms
of interest, e.g.
</p>

<ul>
<li>
<p> For <code>NCBI RefSeq</code>:
</p>

<ul>
<li> <p><code>kingdom = "archaea"</code>
</p>
</li>
<li> <p><code>kingdom = "bacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "fungi"</code>
</p>
</li>
<li> <p><code>kingdom = "invertebrate"</code>
</p>
</li>
<li> <p><code>kingdom = "plant"</code>
</p>
</li>
<li> <p><code>kingdom = "protozoa"</code>
</p>
</li>
<li> <p><code>kingdom = "viral"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_mammalian"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_other"</code>
</p>
</li>
</ul>
</li>
<li>
<p> For <code>NCBI Genbank</code>:
</p>

<ul>
<li> <p><code>kingdom = "archaea"</code>
</p>
</li>
<li> <p><code>kingdom = "bacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "fungi"</code>
</p>
</li>
<li> <p><code>kingdom = "invertebrate"</code>
</p>
</li>
<li> <p><code>kingdom = "plant"</code>
</p>
</li>
<li> <p><code>kingdom = "protozoa"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_mammalian"</code>
</p>
</li>
<li> <p><code>kingdom = "vertebrate_other"</code>
</p>
</li>
</ul>
</li>
<li>
<p> For <code>ENSEMBL</code>:
</p>

<ul>
<li> <p><code>kingdom = "EnsemblVertebrates"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblPlants"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblFungi"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblMetazoa"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblBacteria"</code>
</p>
</li>
<li> <p><code>kingdom = "EnsemblProtists"</code>
</p>
</li>
</ul>
</li>
</ul>
<p>Available kingdoms can be retrieved with <code>getKingdoms</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>group</code></td>
<td>
<p>only species belonging to this subgroup will be downloaded.
Groups can be retrieved with <code>getGroups</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>type of sequences that shall be retrieved. Options are:
</p>

<ul>
<li> <p><code>type = "genome"</code> :
(for genome assembly retrieval; see also <code>getGenome</code>),
</p>
</li>
<li> <p><code>type = "proteome"</code> :
(for proteome retrieval; see also <code>getProteome</code>),
</p>
</li>
<li> <p><code>type = "cds"</code> :
(for coding sequence retrieval; see also <code>getCDS</code>),
</p>
</li>
<li> <p><code>type = "gff"</code> :
(for annotation file retrieval in gff format; see also <code>getGFF</code>),
</p>
</li>
<li> <p><code>type = "gtf"</code> :
(for annotation file retrieval in gtf format (only for ensembl and
ensemblgenomes); see also <code>getGTF</code>)
</p>
</li>
<li> <p><code>type = "rna"</code> :
(for RNA file retrieval in fasta format; see also <code>getRNA</code>),
</p>
</li>
<li> <p><code>type = "rm"</code> :
(for Repeat Masker output file retrieval; see also
<code>getRepeatMasker</code>),
</p>
</li>
<li> <p><code>type = "assemblystats"</code> :
(for genome assembly quality stats file retrieval;
see also <code>getAssemblyStats</code>).
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>restart_at_last</code></td>
<td>
<p>a logical value indicating whether or not <code>meta.retrieval</code> should pick up at the last species when re-running the function.
</p>

<ul>
<li>
<p> If <code>restart_at_last = TRUE</code> (Default) then <code>meta.retrieval</code> will skip all organisms that are already present in the folder
and will start downloading all remaining species. However, this way <code>meta.wretrieval</code> will not be able to check whether
already downloaded organism files are corrupted or not by checking the md5 checksum.
</p>
</li>
<li>
<p> If <code>restart_at_last = FALSE</code> then <code>meta.retrieval</code> will start from the beginning and crawl through already downloaded
organism files and check whether already downloaded organism files are corrupted or not by checking the md5 checksum.
After checking existing files the function will start downloading all remaining organisms.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database
as either a reference genome or a representative genome. Options are:
</p>

<ul>
<li> <p><code>reference = FALSE</code> (Default): all organisms (reference, representative, and non-representative genomes) are downloaded.
</p>
</li>
<li> <p><code>reference = TRUE</code>: organisms that are downloaded must be either a reference or representative genome. Thus, most genomes which are usually non-reference genomes
will not be downloaded.
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>combine</code></td>
<td>
<p>just in case <code>type = "assemblystats"</code> is specified, shall
assemby stats of individual species be imported and combined to a
<code>data.frame</code>?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>path to the folder in which downloaded genomes shall be stored.
By default the kingdom name is used to name the output folder.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function aims to perform bulk retrieval of the genomes,
proteomes, cds, etc. of species that belong to the same kingdom of life or
to the same subgroup.
</p>


<h3>Value</h3>

<p>a character vector storing the file paths of the retrieved files.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other meta_retrival: 
<code>meta.retrieval.all()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# get all available kingdoms for refseq
getKingdoms(db = "refseq")
# download all vertebrate genomes from refseq
meta.retrieval(kingdom = "vertebrate_mammalian",
               db = "refseq",
               type = "genome")

# get all available kingdoms for genbank
getKingdoms(db = "genbank")
# download all vertebrate genomes from genbank
meta.retrieval(kingdom = "vertebrate_mammalian",
               db = "genbank",
               type = "genome")


# In case users do not wish to retrieve genomes from an entire kingdom,
# but rather from a subgoup (e.g. from species belonging to the
# Gammaproteobacteria class, a subgroup of the bacteria kingdom),
# they can use the following workflow"
# First, users can again consult the getKingdoms() function to retrieve
# kingdom information.
getKingdoms(db = "refseq")

# In this example, we will choose the bacteria kingdom.
# Now, the getGroups() function allows users to obtain available
# subgroups of the bacteria kingdom.
getGroups(db = "refseq", kingdom = "bacteria")

# Now we choose the group Gammaproteobacteria and specify
# the group argument in the meta.retrieval() function
meta.retrieval(kingdom = "bacteria",
   roup = "Gammaproteobacteria",
   db = "refseq",
   type = "genome")

## End(Not run)
</code></pre>


</div>