<div class="container">

<table style="width: 100%;"><tr>
<td>getGFF</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Genome Annotation Retrieval (GFF3)</h2>

<h3>Description</h3>

<p>Main retrieval function for GFF files of an
organism of interest. By specifying the scientific name of an organism of
interest the corresponding gff file storing the annotation  for the organism
of interest can be downloaded and stored locally. GFF files can be retrieved
from several databases.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getGFF(
  db = "refseq",
  organism,
  reference = FALSE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  remove_annotation_outliers = FALSE,
  path = file.path("_ncbi_downloads", "annotation"),
  mute_citation = FALSE,
  format = "gff3"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li>
<p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li>
<p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li>
<p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>remove_annotation_outliers</code></td>
<td>
<p>shall outlier lines be removed from the input <code>annotation_file</code>?
If yes, then the initial <code>annotation_file</code> will be overwritten and the removed outlier lines will be stored at <code>tempdir</code>
for further exploration.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding annotation file shall be stored. Default is
<code>path = file.path("_ncbi_downloads","annotation")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>"gff3", alternative "gtf" for ensembl.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory relative to file type, if you get fasta genomes it will be
_ncbi_downloads/genomes'.
In case the corresponding fasta file already exists within the
'_ncbi_downloads/genomes' folder and is accessible within the workspace,
no download process will be performed.
For other file types the same rule applies.
</p>


<h3>Value</h3>

<p>File path to downloaded genome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code>getBio()</code>,
<code>getCDS()</code>,
<code>getCollection()</code>,
<code>getGenome()</code>,
<code>getProteome()</code>,
<code>getRNA()</code>
</p>
<p>Other gff: 
<code>getGFFSet()</code>,
<code>read_gff()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download the annotation of Arabidopsis thaliana from refseq
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Athal_gff &lt;- getGFF( db       = "refseq",
               organism = "Arabidopsis thaliana",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Athal_gff_import &lt;- read_gff(Athal_gff)


# download the genome of Arabidopsis thaliana from genbank
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Athal_gff &lt;- getGFF( db       = "genbank",
               organism = "Arabidopsis thaliana",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Athal_gff_import &lt;- read_gff(Athal_gff)

# download the genome of Homo sapiens from ensembl
# and store the corresponding genome file in '_ncbi_downloads/annotation'
Hsap_gff &lt;- getGFF( db       = "ensembl",
               organism = "Homo sapiens",
               path = file.path("_ncbi_downloads","annotation"),
               remove_annotation_outliers = TRUE)
Hsap_gff_import &lt;- read_gff(Hsap_gff)


## End(Not run)

</code></pre>


</div>