<div class="container">

<table style="width: 100%;"><tr>
<td>read_assemblystats</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import Genome Assembly Stats File</h2>

<h3>Description</h3>

<p>This function reads an organism specific Genome Assembly
Stats file that was retrieved with <code>getAssemblyStats</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_assemblystats(file, type = "raw")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a character string specifying the path to the file storing
the Genome Assembly Stats file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>a tibble object, either <code>type = "raw"</code> to import the entire genome assembly
stats file or <code>type = "stats"</code> to import overall statistics including
all chromosomes, mitochondria and plastids.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function takes a string specifying the path to the Genome
Assembly Stats file of interest (e.g. the path returned by
<code>getAssemblyStats</code>) and imports it.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code>getAssemblyStats</code>, <code>read_genome</code>,
<code>read_proteome</code>, <code>read_cds</code>, <code>read_gff</code>
</p>


</div>