<div class="container">

<table style="width: 100%;"><tr>
<td>meta.retrieval.all</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Perform Meta-Genome Retrieval of all organisms in all kingdoms of life</h2>

<h3>Description</h3>

<p>Download genomes, proteomes, cds, gff, rna, or assembly stats
files of individual species of all kingdoms of life.
</p>


<h3>Usage</h3>

<pre><code class="language-R">meta.retrieval.all(db = "refseq", type = "genome", reference = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "emsembl"</code>
</p>
</li>
<li> <p><code>db = "ensemblgenomes"</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>type</code></td>
<td>
<p>type of sequences that shall be retrieved. Options are:
</p>

<ul>
<li> <p><code>type = "genome"</code> :
for genome assembly retrieval; see also <code>getGenome</code>),
</p>
</li>
<li> <p><code>type = "proteome"</code> :
(for proteome retrieval; see also <code>getProteome</code>),
</p>
</li>
<li> <p><code>type = "cds"</code> :
(for coding sequence retrieval; see also <code>getCDS</code>),
</p>
</li>
<li> <p><code>type = "gff"</code> :
(for annotation file retrieval in gff format; see also <code>getGFF</code>),
</p>
</li>
<li> <p><code>type = "gtf"</code> :
(for annotation file retrieval in gtf format
(only for ensembl and ensemblgenomes); see also <code>getGTF</code>),
</p>
</li>
<li> <p><code>type = "rna"</code> :
(for RNA file retrieval in fasta format; see also <code>getRNA</code>),
</p>
</li>
<li> <p><code>type = "rm"</code> :
(for Repeat Masker output file retrieval; see also
<code>getRepeatMasker</code>),
</p>
</li>
<li> <p><code>type = "assemblystats"</code> (for genome assembly quality stats
file retrieval; see also <code>getAssemblyStats</code>).
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database
as either a reference genome or a representative genome. Options are:
</p>

<ul>
<li> <p><code>reference = FALSE</code> (Default): all organisms (reference, representative, and non-representative genomes) are downloaded.
</p>
</li>
<li> <p><code>reference = TRUE</code>: organisms that are downloaded must be either a reference or representative genome. Thus, most genomes which are usually non-reference genomes
will not be downloaded.
</p>
</li>
</ul>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function aims to perform bulk retrieval of all genomes
of species for all kingdoms of life.
</p>


<h3>Value</h3>

<p>a character vector storing the file paths of the retrieved files.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other meta_retrival: 
<code>meta.retrieval()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download all genomes from refseq
meta.retrieval.all(db = "refseq", type = "genome")
# download all vertebrate genomes from genbank
meta.retrieval.all(db = "genbank", type = "genome")
# download all vertebrate genomes from ensemblgenomes
meta.retrieval.all(db = "genbank", type = "ensemblgenomes")

## End(Not run)
</code></pre>


</div>