<div class="container">

<table style="width: 100%;"><tr>
<td>getCDSSet</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>CDS retrieval of multiple species</h2>

<h3>Description</h3>

<p>Main CDS retrieval function for a set of organism of interest.
By specifying the scientific names of the organisms of interest the corresponding fasta-files storing the CDS of the organisms of interest
will be downloaded and stored locally. CDS files can be retrieved from several databases.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getCDSSet(
  db = "refseq",
  organisms,
  reference = FALSE,
  release = NULL,
  gunzip = TRUE,
  update = FALSE,
  path = "set_CDS"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organisms</code></td>
<td>
<p>a character vector storing the names of the organisms than shall be retrieved.
There are three available options to characterize an organism:</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>update</code></td>
<td>
<p>logical, default FALSE. Updated backend cached files needed.
Usually keep this false, to make ut run much faster. Only set to TRUE, if you
believe you cache is outdated (Species only exist in newest release etc)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>character, default location is paste0("set_", toupper(set_type))</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory 'set_CDSs' to store
the CDSs of interest as fasta files for future processing.
In case the corresponding fasta file already exists within the
'set_CDSs' folder and is accessible within the workspace,
no download process will be performed.
</p>


<h3>Value</h3>

<p>File path to downloaded genomes
(names are identifiers: 'new' (file was downloaded now),
'old' files did already exist)
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBioSet: 
<code>getBioSet()</code>,
<code>getCollectionSet()</code>,
<code>getGFFSet()</code>,
<code>getGenomeSet()</code>,
<code>getProteomeSet()</code>,
<code>getRNASet()</code>
</p>
<p>Other cds: 
<code>getCDS()</code>,
<code>read_cds()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
getBioSet("refseq", organisms = c("Arabidopsis thaliana",
                                  "Arabidopsis lyrata",
                                  "Capsella rubella"),
                                  set_type = "cds")

## End(Not run)
</code></pre>


</div>