<div class="container">

<table style="width: 100%;"><tr>
<td>read_rna</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Import RNA as Biostrings or data.table object</h2>

<h3>Description</h3>

<p>This function reads an organism specific RNA stored in a
defined file format.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_rna(file, format = "fasta", obj.type = "Biostrings", ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>a character string specifying the path to the file
storing the RNA.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>format</code></td>
<td>
<p>a character string specifying the file format used to store the
genome, e.g. <code>format = "fasta"</code> (default) or <code>format = "gbk"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obj.type</code></td>
<td>
<p>a character string specifying the object stype in which the
genomic sequence shall be represented.
Either as <code>obj.type = "Biostrings"</code> (default) or
as <code>obj.type = "data.table"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>additional arguments that are used by
<code>read.fasta</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function takes a string specifying the path to the RNA file
of interest as first argument. It is possible to read in different proteome
file standards such as <em>fasta</em> or <em>genebank</em>.
</p>


<h3>Value</h3>

<p>A data.table storing the gene id in the first column and the
corresponding sequence as string in the second column.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other rna: 
<code>getRNASet()</code>,
<code>getRNA()</code>
</p>
<p>Other readers: 
<code>read_cds()</code>,
<code>read_genome()</code>,
<code>read_gff()</code>,
<code>read_proteome()</code>
</p>


</div>