<div class="container">

<table style="width: 100%;"><tr>
<td>getMetaGenomeAnnotations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Retrieve annotation *.gff files for metagenomes from NCBI Genbank</h2>

<h3>Description</h3>

<p>Retrieve available annotation *.gff files for metagenomes 
from NCBI Genbank. NCBI Genbank allows users
to download entire metagenomes and their annotations of several metagenome 
projects. This function downloads available metagenomes that can then be 
downloaded via <code>getMetaGenomes</code>.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getMetaGenomeAnnotations(
  name,
  path = file.path("_ncbi_downloads", "metagenome", "annotations")
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>name</code></td>
<td>
<p>metagenome name retrieved by <code>listMetaGenomes</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>a character string specifying the location (a folder) 
in which the corresponding metagenome annotations shall be stored. 
Default is
<code>path</code> = <code>file.path("_ncbi_downloads","metagenome","annotations")</code>.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p><code>getMetaGenomes</code>, <code>listMetaGenomes</code>, 
<code>getGFF</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# Frist, retrieve a list of available metagenomes
listMetaGenomes()

# Now, retrieve the 'human gut metagenome'
getMetaGenomeAnnotations(name = "human gut metagenome")

## End(Not run)
</code></pre>


</div>