<div class="container">

<table style="width: 100%;"><tr>
<td>getProteome</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Proteome Retrieval</h2>

<h3>Description</h3>

<p>Main proteome retrieval function for an organism of interest.
By specifying the scientific name of an organism of interest the
corresponding fasta-file storing the proteome of the organism of interest
can be downloaded and stored locally. Proteome files can be retrieved from
several databases.
</p>


<h3>Usage</h3>

<pre><code class="language-R">getProteome(
  db = "refseq",
  organism,
  reference = TRUE,
  skip_bacteria = TRUE,
  release = NULL,
  gunzip = FALSE,
  update = TRUE,
  path = file.path("_ncbi_downloads", "proteomes"),
  mute_citation = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>a character string specifying the database from which the genome
shall be retrieved:
</p>

<ul>
<li> <p><code>db = "refseq"</code>
</p>
</li>
<li> <p><code>db = "genbank"</code>
</p>
</li>
<li> <p><code>db = "ensembl"</code>
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>organism</code></td>
<td>
<p>Organism selector id,
there are three options to characterize an organism:
</p>

<ul>
<li>
<p> by <code>scientific name</code>: e.g. <code>organism = "Homo sapiens"</code>
</p>
</li>
<li>
<p> by <code>database specific accession identifier</code>: e.g. <code>organism = "GCF_000001405.37"</code> (= NCBI RefSeq identifier for <code>Homo sapiens</code>)
</p>
</li>
<li>
<p> by <code>taxonomic identifier from NCBI Taxonomy</code>: e.g. <code>organism = "9606"</code> (= taxid of <code>Homo sapiens</code>)
</p>
</li>
</ul>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference</code></td>
<td>
<p>a logical value indicating whether or not a genome shall be downloaded if it isn't marked in the database as either a reference genome or a representative genome.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>skip_bacteria</code></td>
<td>
<p>Due to its enormous dataset size (&gt; 700MB as of July 2023),
the bacterial summary file will not be loaded by default anymore. If users
wish to gain insights for the bacterial kingdom they needs to actively specify <code>skip_bacteria = FALSE</code>. When <code>skip_bacteria = FALSE</code> is set then the
bacterial summary file will be downloaded.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>release</code></td>
<td>
<p>a numeric, the database release version of ENSEMBL (<code>db = "ensembl"</code>). Default is <code>release = NULL</code> meaning
that the most recent database version is used. <code>release = 75</code> would for human would give the stable
GRCh37 release in ensembl. Value must be &gt; 46, since ensembl did not structure their data
if the standard format before that.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gunzip</code></td>
<td>
<p>a logical, indicating whether or not files should be unzipped.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>update</code></td>
<td>
<p>logical, default TRUE. (Uniprot only for now!)
If species info file exists already,
do not re download, makes it faster but the file can be old, i.e. no longer
as complete as it could be.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>a character string specifying the location (a folder) in which
the corresponding proteome shall be stored. Default is
<code>path</code> = <code>file.path("_ncbi_downloads","proteomes")</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mute_citation</code></td>
<td>
<p>logical, default FALSE, indicating whether citation message should be muted.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Internally this function loads the the overview.txt file from NCBI:
</p>
<p>refseq: ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/
</p>
<p>genbank: ftp://ftp.ncbi.nlm.nih.gov/genomes/genbank/
</p>
<p>and creates a directory '_ncbi_downloads/proteomes' to store
the proteome of interest as fasta file for future processing.
</p>


<h3>Value</h3>

<p>File path to downloaded proteome.
</p>


<h3>Author(s)</h3>

<p>Hajk-Georg Drost
</p>


<h3>See Also</h3>

<p>Other getBio: 
<code>getBio()</code>,
<code>getCDS()</code>,
<code>getCollection()</code>,
<code>getGFF()</code>,
<code>getGenome()</code>,
<code>getRNA()</code>
</p>
<p>Other proteome: 
<code>getProteomeSet()</code>,
<code>read_proteome()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: 
# download the proteome of Arabidopsis thaliana from NCBI RefSeq
# and store the corresponding proteome file in '_ncbi_downloads/refseq/proteomes'
file_path &lt;- getProteome( db       = "refseq",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","refseq","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# download the proteome of Arabidopsis thaliana from NCBI Genbank
# and store the corresponding proteome file in '_ncbi_downloads/genbank/proteomes'
file_path &lt;- getProteome( db       = "genbank",
             organism = "Arabidopsis thaliana",
             path     = file.path("_ncbi_downloads","genbank","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# and store the corresponding proteome file in '_downloads/uniprot/proteomes'
file_path &lt;- getProteome( db       = "uniprot",
             organism = "Arabidopsis thaliana",
             path     = file.path("_downloads","uniprot","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

# download the proteome of Arabidopsis thaliana from ENSEMBL
# and store the corresponding proteome file in '_downloads/ensembl/proteomes'
file_path &lt;- getProteome( db       = "ensembl",
             organism = "Arabidopsis thaliana",
             path     = file.path("_downloads","ensembl","proteomes") )
# import proteome into R session
Ath_proteome &lt;- read_proteome(file_path, format = "fasta")

## End(Not run)
</code></pre>


</div>