<div class="container">

<table style="width: 100%;"><tr>
<td>seq_rev_translate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reverse translate amino acid sequences</h2>

<h3>Description</h3>

<p>The function perform reverse translation of amino acid sequences.
Such operation does not exist in nature but is provided for completeness.
Because of codon degeneracy it is expected
to produce many ambiguous nucleotides.
</p>


<h3>Usage</h3>

<pre><code class="language-R">seq_rev_translate(x, code = 1)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an amino acid sequence (<code>bioseq_aa</code>)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>code</code></td>
<td>
<p>an integer indicating the genetic code to use for
reverse translation (default 1 uses the Standard genetic code). See Details.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Gaps (-) are interpreted as unknown amino acids (X) but can be
removed prior to the translation with the function <code>seq_remove_gap</code>.
</p>


<h3>Value</h3>

<p>a vector of DNA sequences.
</p>


<h3>See Also</h3>

<p>Other biological operations: 
<code>rev_complement</code>,
<code>seq_translate()</code>,
<code>transcription</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">
x &lt;- dna("ACTTTGGCTAAG")
y &lt;- seq_translate(x)
z &lt;- seq_rev_translate(y)
z
# There is a loss of information during the reverse translation
all.equal(x, z)

</code></pre>


</div>