<div class="container">

<table style="width: 100%;"><tr>
<td>sHe</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Spatial Analysis of Gene Diversity
</h2>

<h3>Description</h3>

<p>Estimate spatial gene diversity (expected heterozygozity - <em>He</em>) through the individual-centred approach by Manel et al. (2007).
<code>sHe()</code> calculates the unbiased estimate of <em>He</em> based on the information of allele frequency obtained from codominant or
dominant markers in individuals within a circular moving windows of known radius over the sampling area.
</p>


<h3>Usage</h3>

<pre><code class="language-R">sHe(x, coord.cols = 1:2, marker.cols = 3:4, 
	marker.type = c("codominant", "dominant"), 
	grid = NULL, latlong2km = TRUE, radius, nmin = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p> a data frame or numeric matrix containing columns with coordinates of individuals and marker genotyping</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coord.cols</code></td>
<td>
<p> a vector of integer giving the columns of coordinates in <code>x</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.cols</code></td>
<td>
<p> a vector of integer giving the columns of markers in <code>x</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>marker.type</code></td>
<td>
<p> a character; the type of molecular marker</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>grid</code></td>
<td>
<p> optional; a two-column matrix containing coordinates over which to predict <em>He</em></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>latlong2km</code></td>
<td>
<p> logical; should coordinates be converted from lat/long format into kilometer-grid based?</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>radius</code></td>
<td>
<p> the radius of the moving window. It must be in the same format as sampling coordinates</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>nmin</code></td>
<td>
<p> optional; a numeric value indicating the minimum number of individuals used to calculate <em>He</em>. If is
the number of individuals in a certain location is less then <code>nmin</code>, <code>sHe</code> will consider <em>He</em> as zero.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The unbiased estimate of expected heterogygozity (Nei, 1978) is given by:
</p>
<p style="text-align: center;"><code class="reqn"> He = (1 - \sum_{i=1}^{n} p_{i}^{2}) \frac{2n}{2n - 1} </code>
</p>

<p>where <code class="reqn">p_{i}</code> is the frequency of the i-th allele per locus considering the <code class="reqn">n</code> individuals in a certain location.
</p>


<h3>Value</h3>

<p>A list of
</p>
<table>
<tr style="vertical-align: top;">
<td><code>diversity</code></td>
<td>
<p> a data frame with the following columns: <em>coord.x</em> - the x-axis coordinates of the predicion grid,
<em>coord.y</em> - the y-axis coordinates of the predicion grid, <em>n</em> - the number of individuals in a certain points in the grid,
<em>MaxDist</em> - the maximum observed distance among these individuals, <em>uHe</em> - the unbiased estimate
of gene diversity (as expressed above), and <em>SE</em> - the standard error of <em>uHe</em>. </p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mHe</code></td>
<td>
<p> a matrix containing the estimates of <em>He</em> for every marker, on each point of the <code>grid</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>locations</code></td>
<td>
<p> a numeric matrix containing the sampling coordinates, as provides as input.</p>
</td>
</tr>
</table>
<h3>Warning </h3>

<p>Depending on the dimension of <code>x</code> and/or <code>grid</code>, <code>sHe()</code> can be time demanding.
</p>


<h3>Author(s)</h3>

<p>Anderson Rodrigo da Silva &lt;anderson.agro@hotmail.com&gt;
</p>
<p>Ivandilson Pessoa Pinto de Menezes &lt;ivan.menezes@ifgoiano.edu.br&gt;
</p>


<h3>References</h3>

<p>da Silva, A.R.; Malafaia, G.; Menezes, I.P.P. (2017) biotools: an R function to predict 
spatial gene diversity via an individual-based approach. <em>Genetics and Molecular Research</em>, 
<b>16</b>: gmr16029655.
</p>
<p>Manel, S., Berthoud, F., Bellemain, E., Gaudeul, M., Luikart, G., Swenson, J.E., Waits, L.P., 
Taberlet, P.; Intrabiodiv Consortium. (2007) A new individual-based spatial approach for 
identifying genetic discontinuities in natural populations. <em>Molecular Ecology</em>, <b>16</b>:2031-2043.
</p>
<p>Nei, M. (1978) Estimation of average heterozygozity and genetic distance from a small number of individuals. 
<em>Genetics</em>, <b>89</b>: 583-590.
</p>


<h3>See Also</h3>

<p><code>levelplot</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">data(moco)
data(brazil)

# check points
plot(brazil, cex = 0.1, col = "gray")
points(Lat ~ Lon, data = moco, col = "blue", pch = 20)

# using a retangular grid (not passed as input!)
# ex &lt;- sHe(x = moco, coord.cols = 1:2,
#	marker.cols = 3:20, marker.type = "codominant",
#	grid = NULL, radius = 150)
#ex
# plot(ex, xlab = "Lon", ylab = "Lat")

# A FANCIER PLOT...
# using Brazil's coordinates as prediction grid
# ex2 &lt;- sHe(x = moco, coord.cols = 1:2,
#	marker.cols = 3:20, marker.type = "codominant",
#	grid = brazil, radius = 150)
# ex2
#
# library(maps)
# borders &lt;- data.frame(x = map("world", "brazil")$x, 
#	y = map("world", "brazil")$y)
#
# library(latticeExtra)
# plot(ex2, xlab = "Lon", ylab = "Lat", 
#	xlim = c(-75, -30), ylim = c(-35, 10), aspect = "iso") +
#   latticeExtra::as.layer(xyplot(y ~ x, data = borders, type = "l")) +
#   latticeExtra::as.layer(xyplot(Lat ~ Lon, data = moco))

# End (not run)
</code></pre>


</div>