<div class="container">

<table style="width: 100%;"><tr>
<td>BIGEE</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>BIGEE: Bimodal Gene Expression Exclusivity.
</h2>

<h3>Description</h3>

<p>Part of the Synthetic Lethality detection in Genomics toolkit.  Detects bimodality and non-normality in all genes across the dataset.  Compares all pairwise combinations of bimodal genes and searches for mutually exclusive low expression as evidence of potential synthetic lethality.  Scores gene-pairs based on the presence of mutual exclusive bimodality and the distribution of signal intensity across the rest of the dataset.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BIGEE(
	bisepData=data, 
	sampleType=c("cell_line", "cell_line_low", "patient", "patient_low")
	)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bisepData</code></td>
<td>
<p>This should be the output from the BISEP function.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sampleType</code></td>
<td>
<p>The type of sample being analysed.  Select 'cell_line' or 'patient' for datasets with greater than ~200 samples.  For datasets with less than ~200 samples, use 'cell_line_low' or 'patient_low'.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Lower sample numbers have more stringent bimodality hurdles to clear in order to keep the false positive rate lower.  The tool returns a percentage complete text window so the user can observe the status of the job. 
</p>


<h3>Value</h3>

<p>A matrix containing three columns. Columns 1 and 2 are the gene symbols that make up the candidate synthetic lethal gene pairs.  Column 3 is the score calculated the tool to rank the statistical significance of the gene pairs.
</p>


<h3>Author(s)</h3>

<p>Mark Wappett
</p>


</div>