<div class="container">

<table style="width: 100%;"><tr>
<td>binarize.BASC</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Binarization Across Multiple Scales
</h2>

<h3>Description</h3>

<p>Binarizes real-valued data using the multiscale BASC methods.</p>


<h3>Usage</h3>

<pre><code class="language-R">binarize.BASC(vect, 
              method = c("A","B"), 
              tau = 0.01, 
              numberOfSamples = 999, 
              sigma = seq(0.1, 20, by=.1),
              na.rm=FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>

<p>Chooses the BASC method to use (see details), i.e. either  "A" or "B". 
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>vect</code></td>
<td>

<p>A real-valued vector of data to binarize.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tau</code></td>
<td>

<p>This parameter adjusts the sensitivity and the specificity of the statistical testing procedure that rates the quality of the binarization. Defaults to 0.01.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>numberOfSamples</code></td>
<td>

<p>The number of samples for the bootstrap test. Defaults to 999.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sigma</code></td>
<td>

<p>If <code>method="B"</code>, this specifies a vector of different sigma values for the convolutions with the Bessel function. Ignored for <code>method="A"</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>na.rm</code></td>
<td>

<p>If set to <code>TRUE</code>, <code>NA</code> values are removed from the input. Otherwise, binarization will fail in the presence of <code>NA</code> values.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The two BASC methods can be subdivided into three steps:
</p>
	
<dl>
<dt>Compute a series of step functions:</dt>
<dd>
<p>An initial step function is obtained by rearranging the original time series measurements in increasing order. Then, step functions with fewer discontinuities are calculated. BASC A calculates these step functions in such a way that each minimizes the Euclidean distance to the initial step function. BASC B obtains step functions from smoothened versions of the input function in a scale-space manner.</p>
</dd>
<dt>Find strongest discontinuity in each step function:</dt>
<dd>
<p>A strong discontinuity is a high jump size (derivative) in combination with a low approximation error.</p>
</dd>
<dt>Estimate location and variation of the strongest discontinuities:</dt>
<dd>
<p>Based on these estimates, data values can be excluded from further analyses.
</p>
</dd>
</dl>
<h3>Value</h3>

<p>Returns an object of class <code>BASCResult</code>.
</p>


<h3>References</h3>

<p>M. Hopfensitz, C. MÃ¼ssel, C. Wawra, M. Maucher, M. Kuehl, H. Neumann, and H. A. Kestler. Multiscale Binarization of Gene Expression Data for Reconstructing Boolean Networks. IEEE/ACM Transactions on Computational Biology and Bioinformatics 9(2):487-498, 2012.).
</p>


<h3>See Also</h3>

<p><code>BinarizationResult</code>, 
<code>BASCResult</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">par(mfrow=c(2,1))
result &lt;- binarize.BASC(iris[,"Petal.Length"], method="A", tau=0.15)
print(result)
plot(result)

result &lt;- binarize.BASC(iris[,"Petal.Length"], method="B", tau=0.15)
print(result)
plot(result)
</code></pre>


</div>