<div class="container">

<table style="width: 100%;"><tr>
<td>blast</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Runs BLAST sequence comparison algorithm.</h2>

<h3>Description</h3>

<p>Runs BLAST sequence comparison algorithm.
</p>


<h3>Usage</h3>

<pre><code class="language-R">blast(
  query,
  db,
  maxAccepts = 1,
  maxRejects = 16,
  minIdentity = 0.75,
  alphabet = "nucleotide",
  strand = "both",
  output_to_file = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>query</code></td>
<td>
<p>A dataframe of the query sequences (containing Id and Seq columns)
or a string specifying the FASTA file of the query sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>db</code></td>
<td>
<p>A dataframe of the database sequences (containing Id and Seq columns)
or a string specifying the FASTA file of the database sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxAccepts</code></td>
<td>
<p>A number specifying the maximum accepted hits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxRejects</code></td>
<td>
<p>A number specifying the maximum rejected hits.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minIdentity</code></td>
<td>
<p>A number specifying the minimal accepted sequence
similarity between the query and hit sequences.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabet</code></td>
<td>
<p>A string specifying the query and database alphabet:
'nucleotide' or 'protein'. Defaults to 'nucleotide'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>strand</code></td>
<td>
<p>A string specifying the strand to search: 'plus', 'minus' or
'both'. Defaults to 'both'. Only affects nucleotide searches.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output_to_file</code></td>
<td>
<p>A boolean specifying the output type. If TRUE, the
results are written into a temporary file a string
containing the file name and location is returned.
Otherwise a dataframe of the results is returned.
Defaults to FALSE.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe or a string. A dataframe is returned by default, containing
the BLAST output in columns QueryId, TargetId, QueryMatchStart, QueryMatchEnd,
TargetMatchStart, TargetMatchEnd, QueryMatchSeq, TargetMatchSeq, NumColumns,
NumMatches, NumMismatches, NumGaps, Identity and Alignment. A string is returned
if 'output_to_file' is set to TRUE. This string points to the file
containing the output table.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
query &lt;- system.file("extdata", "query.fasta", package = "blaster")
db &lt;- system.file("extdata", "db.fasta", package = "blaster")

blast_table &lt;- blast(query = query, db = db)

query &lt;- read_fasta(filename = query)
db &lt;- read_fasta(filename = db)
blast_table &lt;- blast(query = query, db = db)

prot &lt;- system.file("extdata", "prot.fasta", package = "blaster")
prot_blast_table &lt;- blast(query = prot, db = prot, alphabet = "protein")

</code></pre>


</div>