<div class="container">

<table style="width: 100%;"><tr>
<td>read_fasta</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Reads the contents of nucleotide or protein FASTA file into a dataframe.</h2>

<h3>Description</h3>

<p>Reads the contents of nucleotide or protein FASTA file into a dataframe.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_fasta(
  filename,
  filter = "",
  non_standard_chars = "error",
  alphabet = "nucleotide"
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>filename</code></td>
<td>
<p>A string specifying the name of the FASTA file to be imported.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter</code></td>
<td>
<p>An optional string specifying a sequence motif for sequence filtering.
Only keeps those sequences containing this motif. Also splits the
matched sequences and provides the split parts in two additional columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>non_standard_chars</code></td>
<td>
<p>A string specifying instructions for handling non-standard
nucleotide or amino acid characters. Options include 'remove',
'ignore' or throw an 'error'. Defaults to 'error'.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alphabet</code></td>
<td>
<p>A string specifying the query and database alphabet:
'nucleotide' or 'protein'. Defaults to 'nucleotide'.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A dataframe containing FASTA ids (Id column) and sequences (Seq column).
If 'filter' is specified, the split sequences are stored in additional columns
Part1 and Part2.
</p>


<h3>Examples</h3>

<pre><code class="language-R">
query &lt;- system.file("extdata", "query.fasta", package = "blaster")

query &lt;- read_fasta(filename = query)

</code></pre>


</div>