<div class="container">

<table style="width: 100%;"><tr>
<td>blockForest</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>blockForest</h2>

<h3>Description</h3>

<p>Block forests without parameter tuning. 
Use <code>blockfor</code> for standard interface.
This function is called by <code>blockfor</code>
and will rarely be considered directly by the user (since parameter tuning
is required in applications).
</p>


<h3>Usage</h3>

<pre><code class="language-R">blockForest(
  formula = NULL,
  data = NULL,
  num.trees = 500,
  mtry = NULL,
  importance = "none",
  write.forest = TRUE,
  probability = FALSE,
  min.node.size = NULL,
  replace = TRUE,
  sample.fraction = ifelse(replace, 1, 0.632),
  case.weights = NULL,
  splitrule = NULL,
  num.random.splits = 1,
  alpha = 0.5,
  minprop = 0.1,
  split.select.weights = NULL,
  always.split.variables = NULL,
  blocks = NULL,
  block.method = "BlockForest",
  block.weights = NULL,
  respect.unordered.factors = NULL,
  scale.permutation.importance = FALSE,
  keep.inbag = FALSE,
  holdout = FALSE,
  quantreg = FALSE,
  num.threads = NULL,
  save.memory = FALSE,
  verbose = TRUE,
  seed = NULL,
  dependent.variable.name = NULL,
  status.variable.name = NULL,
  classification = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>formula</code></td>
<td>
<p>Object of class <code>formula</code> or <code>character</code> describing the model to fit. Interaction terms supported only for numerical variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Training data of class <code>data.frame</code>, <code>matrix</code>, <code>dgCMatrix</code> (Matrix) or <code>gwaa.data</code> (GenABEL).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.trees</code></td>
<td>
<p>Number of trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mtry</code></td>
<td>
<p>This is either a number specifying the number of variables sampled for each
split from all variables (for variants "VarProb" and "SplitWeights")
or a vector of length equal to the number of blocks, where the m-th entry of the
vector gives the number of variables to sample from block m (for variants "BlockForest", "RandomBlock", and "BlockVarSel").
The default values are sqrt(p_1) + sqrt(p_2) + ... sqrt(p_M) and (sqrt(p_1), sqrt(p_2), ..., sqrt(p_M)), respectively,
where p_m denotes the number of variables in the m-th block (m = 1, ..., M) and sqrt() denoted the square root function.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>importance</code></td>
<td>
<p>Variable importance mode, one of 'none', 'impurity', 'impurity_corrected', 'permutation'. The 'impurity' measure is the Gini index for classification, the variance of the responses for regression and the sum of test statistics (see <code>splitrule</code>) for survival.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>write.forest</code></td>
<td>
<p>Save <code>blockForest.forest</code> object, required for prediction. Set to <code>FALSE</code> to reduce memory usage if no prediction intended.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>probability</code></td>
<td>
<p>Grow a probability forest as in Malley et al. (2012).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.node.size</code></td>
<td>
<p>Minimal node size. Default 1 for classification, 5 for regression, 3 for survival, and 10 for probability.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>replace</code></td>
<td>
<p>Sample with replacement.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.fraction</code></td>
<td>
<p>Fraction of observations to sample. Default is 1 for sampling with replacement and 0.632 for sampling without replacement. For classification, this can be a vector of class-specific values.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>case.weights</code></td>
<td>
<p>Weights for sampling of training observations. Observations with larger weights will be selected with higher probability in the bootstrap (or subsampled) samples for the trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>splitrule</code></td>
<td>
<p>Splitting rule, default "extratrees". Other options are "gini" for classification and probability estimation, "variance", or "maxstat" for regression and "logrank", "C" or "maxstat" for survival.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.random.splits</code></td>
<td>
<p>For "extratrees" splitrule.: Number of random splits to consider for each candidate splitting variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>For "maxstat" splitrule: Significance threshold to allow splitting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>minprop</code></td>
<td>
<p>For "maxstat" splitrule: Lower quantile of covariate distribution to be considered for splitting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>split.select.weights</code></td>
<td>
<p>Numeric vector with weights between 0 and 1, representing the probability to select variables for splitting. Alternatively, a list of size num.trees, containing split select weight vectors for each tree can be used. Use this for the "VarProb" variant.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>always.split.variables</code></td>
<td>
<p>Character vector with variable names to be always selected in addition to the <code>mtry</code> variables tried for splitting.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>blocks</code></td>
<td>
<p>Block memberships of the variables. See <code>blockfor</code> for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block.method</code></td>
<td>
<p>Variant to use. Options are: "BlockForest" (default), "RandomBlock", "BlockVarSel", "SplitWeights".</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>block.weights</code></td>
<td>
<p>Tuning parameter values for the blocks in the variants. A vector of length equal to the number of blocks or a list with vectors containing tree-wise values. For block.method='RandomBlock' these are the block sample probabilities.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>respect.unordered.factors</code></td>
<td>
<p>Handling of unordered factor covariates. One of 'ignore', 'order' and 'partition'. For the "extratrees" splitrule the default is "partition" for all other splitrules 'ignore'. Alternatively TRUE (='order') or FALSE (='ignore') can be used. See below for details.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>scale.permutation.importance</code></td>
<td>
<p>Scale permutation importance by standard error as in (Breiman 2001). Only applicable if permutation variable importance mode selected.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>keep.inbag</code></td>
<td>
<p>Save how often observations are in-bag in each tree.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>holdout</code></td>
<td>
<p>Hold-out mode. Hold-out all samples with case weight 0 and use these for variable importance and prediction error.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quantreg</code></td>
<td>
<p>Prepare quantile prediction as in quantile regression forests (Meinshausen 2006). Regression only. Set <code>keep.inbag = TRUE</code> to prepare out-of-bag quantile prediction.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.threads</code></td>
<td>
<p>Number of threads. Default is number of CPUs available.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.memory</code></td>
<td>
<p>Use memory saving (but slower) splitting mode. No effect for survival and GWAS data. Warning: This option slows down the tree growing, use only if you encounter memory problems.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>Show computation status and estimated runtime.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>Random seed. Default is <code>NULL</code>, which generates the seed from <code>R</code>. Set to <code>0</code> to ignore the <code>R</code> seed.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>dependent.variable.name</code></td>
<td>
<p>Name of dependent variable, needed if no formula given. For survival forests this is the time variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>status.variable.name</code></td>
<td>
<p>Name of status variable, only applicable to survival data and needed if no formula given. Use 1 for event and 0 for censoring.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>classification</code></td>
<td>
<p>Only needed if data is a matrix. Set to <code>TRUE</code> to grow a classification forest.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>See <code>blockfor</code> and the <code>ranger</code> package.
</p>


<h3>Value</h3>

<p>Object of class <code>blockForest</code> with elements
</p>
<table>
<tr style="vertical-align: top;">
<td><code>forest</code></td>
<td>
<p>Saved forest (If write.forest set to TRUE). Note that the variable IDs in the <code>split.varIDs</code> object do not necessarily represent the column number in R.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>predictions</code></td>
<td>
<p>Predicted classes/values, based on out of bag samples (classification and regression only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variable.importance</code></td>
<td>
<p>Variable importance for each independent variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>prediction.error</code></td>
<td>
<p>Overall out of bag prediction error. For classification this is the fraction of missclassified samples, for probability estimation and regression the mean squared error and for survival one minus Harrell's C-index.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>r.squared</code></td>
<td>
<p>R squared. Also called explained variance or coefficient of determination (regression only). Computed on out of bag data.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>confusion.matrix</code></td>
<td>
<p>Contingency table for classes and predictions based on out of bag samples (classification only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>unique.death.times</code></td>
<td>
<p>Unique death times (survival only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>chf</code></td>
<td>
<p>Estimated cumulative hazard function for each sample (survival only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>survival</code></td>
<td>
<p>Estimated survival function for each sample (survival only).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>call</code></td>
<td>
<p>Function call.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.trees</code></td>
<td>
<p>Number of trees.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.independent.variables</code></td>
<td>
<p>Number of independent variables.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mtry</code></td>
<td>
<p>Value of mtry used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min.node.size</code></td>
<td>
<p>Value of minimal node size used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>treetype</code></td>
<td>
<p>Type of forest/tree. classification, regression or survival.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>importance.mode</code></td>
<td>
<p>Importance mode used.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>num.samples</code></td>
<td>
<p>Number of samples.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>inbag.counts</code></td>
<td>
<p>Number of times the observations are in-bag in the trees.</p>
</td>
</tr>
</table>
<h3>Author(s)</h3>

<p>Marvin N. Wright
</p>


<h3>References</h3>


<ul>
<li>
<p> Hornung, R. &amp; Wright, M. N. (2019) Block Forests: random forests for blocks of clinical and omics covariate data. BMC Bioinformatics 20:358. <a href="https://doi.org/10.1186/s12859-019-2942-y">doi:10.1186/s12859-019-2942-y</a>.
</p>
</li>
<li>
<p> Wright, M. N. &amp; Ziegler, A. (2017). ranger: A Fast Implementation of Random Forests for High Dimensional Data in C++ and R. J Stat Softw 77:1-17. <a href="https://doi.org/10.18637/jss.v077.i01">doi:10.18637/jss.v077.i01</a>.
</p>
</li>
<li>
<p> Schmid, M., Wright, M. N. &amp; Ziegler, A. (2016). On the use of Harrell's C for clinical risk prediction via random survival forests. Expert Syst Appl 63:450-459. <a href="https://doi.org/10.1016/j.eswa.2016.07.018">doi:10.1016/j.eswa.2016.07.018</a>. 
</p>
</li>
<li>
<p> Wright, M. N., Dankowski, T. &amp; Ziegler, A. (2017). Unbiased split variable selection for random survival forests using maximally selected rank statistics. Stat Med. <a href="https://doi.org/10.1002/sim.7212">doi:10.1002/sim.7212</a>.
</p>
</li>
<li>
<p> Breiman, L. (2001). Random forests. Mach Learn, 45(1), 5-32. <a href="https://doi.org/10.1023/A%3A1010933404324">doi:10.1023/A:1010933404324</a>. 
</p>
</li>
<li>
<p> Ishwaran, H., Kogalur, U. B., Blackstone, E. H., &amp; Lauer, M. S. (2008). Random survival forests. Ann Appl Stat 2:841-860. <a href="https://doi.org/10.1097/JTO.0b013e318233d835">doi:10.1097/JTO.0b013e318233d835</a>. 
</p>
</li>
<li>
<p> Malley, J. D., Kruppa, J., Dasgupta, A., Malley, K. G., &amp; Ziegler, A. (2012). Probability machines: consistent probability estimation using nonparametric learning machines. Methods Inf Med 51:74-81. <a href="https://doi.org/10.3414/ME00-01-0052">doi:10.3414/ME00-01-0052</a>.
</p>
</li>
<li>
<p> Hastie, T., Tibshirani, R., Friedman, J. (2009). The Elements of Statistical Learning. Springer, New York. 2nd edition.
</p>
</li>
<li>
<p> Geurts, P., Ernst, D., Wehenkel, L. (2006). Extremely randomized trees. Mach Learn 63:3-42. <a href="https://doi.org/10.1007/s10994-006-6226-1">doi:10.1007/s10994-006-6226-1</a>.
</p>
</li>
<li>
<p> Meinshausen (2006). Quantile Regression Forests. J Mach Learn Res 7:983-999. <a href="https://www.jmlr.org/papers/v7/meinshausen06a.html">https://www.jmlr.org/papers/v7/meinshausen06a.html</a>.  
</p>
</li>
</ul>
<h3>See Also</h3>

<p><code>predict.blockForest</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">require(blockForest)

# Standard Block Forest
blockForest(Species ~ ., iris, 
            blocks = list(1:2, 3:4), 
            mtry = c(1, 2), 
            block.weights = c(0.1, 0.9), 
            block.method = "BlockForest")

# Without blocks, grow standard random forest
blockForest(Species ~ ., iris)

</code></pre>


</div>