<div class="container">

<table style="width: 100%;"><tr>
<td>bmass</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bayesian multivariate analysis of summary statistics 
(<code>bmass</code>)</h2>

<h3>Description</h3>

<p>Run <code>bmass</code> on a set of phenotypes that each have
univariate GWAS statistics on the same set of SNPs
</p>


<h3>Usage</h3>

<pre><code class="language-R">bmass(DataSources, GWASsnps = NULL,
  SNPMarginalUnivariateThreshold = 1e-06,
  SNPMarginalMultivariateThreshold = 1e-06, GWASThreshFlag = TRUE,
  GWASThreshValue = 5e-08, NminThreshold = 0,
  PrintMergedData = FALSE, PrintProgress = FALSE, ...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>DataSources</code></td>
<td>
<p>A string indicating the variable names of the
input datafiles and phenotypes. No default value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GWASsnps</code></td>
<td>
<p>A data.table containing rows of SNPs that were
univariate genome-wide significant in the phenotypes being used for
analysis; <code>GWASsnps</code> input file should have two columns, one 
for chromosome and another for basepair position (with column headers
of <code>Chr</code> and <code>BP</code>). No default value.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPMarginalUnivariateThreshold</code></td>
<td>
<p>A numerical value indicating
the univariate p-value threshold to use when collecting marginally
significant SNPs for final <code>bmass</code> analysis. Default is 
<code>1e-6</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SNPMarginalMultivariateThreshold</code></td>
<td>
<p>A numerical value
indicating the basic multivariate p-value threshold to use when
collecting marginally significant SNPs for final <code>bmass</code> 
analysis. Default is <code>1e-6</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GWASThreshFlag</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether to threshold input <code>GWASsnps</code> list by a 
univariate GWAS p-value or not (eg the input <code>GWASsnps</code> list 
contains variants that are significant from discovery + replication 
data, but the input summary statistics are just from the discovery 
cohort). Default is <code>TRUE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>GWASThreshValue</code></td>
<td>
<p>A numerical value indicating the univariate
p-value threshold to use in conjunction with the <code>GWASThreshFlag</code>.
Default is <code>5e-8</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>NminThreshold</code></td>
<td>
<p>A numerical value that indicates a sample size
threshold to use where SNPs below which are removed. Default is
<code>0</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PrintMergedData</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether the intermediary 'merged datafile' should be included 
in the final <code>bmass</code> output; this file combines all the phenotypes 
for every SNP provided just prior to thresholding for marginally
significant SNPs. Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>PrintProgress</code></td>
<td>
<p>A logical <code>TRUE</code>/<code>FALSE</code> flag that 
indicates whether progress statements should be printed to 
<code>stderr</code> during the course of running <code>bmass</code> or not.
Default is <code>FALSE</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Additional optional arguments.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A list containing model, SNP, and posterior information for
both the previously significant univariate SNPs (<code>PreviousSNPs</code>)
and the newly significant multivariate SNPs (<code>NewSNPs</code>). For a 
full breakdown of the <code>bmass</code> output list structure, please see 
the associated vignettes.
</p>


<h3>Other Examples</h3>

<p><code>bmass(c("HDL","LDL","TG","TC"), GWASsnps, NminThreshold = 50000)</code>
<code>bmass(c("HDL","LDL","TG","TC"), GWASsnps, GWASThreshValue = 1e-8,
  NminThreshold = 50000, PrintProgress = TRUE)</code> 
<code>bmass(c("HDL", "LDL", "TG", "TC"), GWASsnps, GWASThreshFlag = FALSE,
  SNPMarginalUnivariateThreshold = 1e-4,
  SNPMarginalMultivariateThreshold = 1e-4,
  PrintMergedData = TRUE)</code> 
<code>bmassOutput &lt;- bmass(c("HDL","LDL","TG","TC"),
  GWASsnps, NminThreshold = 50000)</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">Phenotypes &lt;- c("bmass_SimulatedData1", "bmass_SimulatedData2")
bmassOutput &lt;- bmass(Phenotypes, bmass_SimulatedSigSNPs)
summary(bmassOutput)
bmassOutput$NewSNPs$SNPs

</code></pre>


</div>