<div class="container">

<table style="width: 100%;"><tr>
<td>fixed_loci</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Identify fixed loci</h2>

<h3>Description</h3>

<p>A locus is "fixed" if the non-missing sub-vector contains all 0's or all 2's (the locus is completely homozygous for one allele or completely homozygous for the other allele).
This function tests each locus, returning a vector that is <code>TRUE</code> for each fixed locus, <code>FALSE</code> otherwise.
Loci with only missing elements (<code>NA</code>) are treated as fixed.
The parameter <code>maf_min</code> extends the "fixed" definition to loci whose minor allele frequency is smaller or equal than this value.
Below <code>m</code> is the number of loci, and <code>n</code> is the number of individuals.
</p>


<h3>Usage</h3>

<pre><code class="language-R">fixed_loci(X, maf_min = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>The <code>m</code>-by-<code>n</code> genotype matrix</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maf_min</code></td>
<td>
<p>The minimum minor allele frequency (default zero), to extend the working definition of "fixed" to include rare variants.
Loci with minor allele frequencies less than or <em>equal</em> to this value are marked as fixed.
Must be a scalar between 0 and 0.5.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A length-<code>m</code> boolean vector where the <code>i</code> element is <code>TRUE</code> if locus <code>i</code> is fixed or completely missing, <code>FALSE</code> otherwise.
If <code>X</code> had row names, they are copied to the names of this output vector.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># here's a toy genotype matrix
X &lt;- matrix(
       data = c(
              2, 2, NA,  # fixed locus (with one missing element)
              0, NA, 0,  # another fixed locus, for opposite allele
              1, 1, 1,   # NOT fixed (heterozygotes are not considered fixed)
              0, 1, 2,   # a completely variable locus
              0, 0, 1,   # a somewhat "rare" variant
              NA, NA, NA # completely missing locus (will be treated as fixed)
             ),
       ncol = 3, byrow = TRUE)

# test that we get the desired values
stopifnot(
  fixed_loci(X) == c(TRUE, TRUE, FALSE, FALSE, FALSE, TRUE)
)

# the "rare" variant gets marked as "fixed" if we set `maf_min` to its frequency
stopifnot(
  fixed_loci(X, maf_min = 1/6) == c(TRUE, TRUE, FALSE, FALSE, TRUE, TRUE)
)

</code></pre>


</div>