<div class="container">

<table style="width: 100%;"><tr>
<td>simulateNetwork</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Simulate Boolean Network
</h2>

<h3>Description</h3>

<p>Simulates a Boolean network and generates noise of user-defined model preference and paramters.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulateNetwork(net, n.data, p, obsModel)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>net</code></td>
<td>

<p>A boolean Network object (specified in BoolNet vernacular) to be simulated
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n.data</code></td>
<td>

<p>Length of time-series to simulate
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>p</code></td>
<td>

<p>Process noise to build into the simulation. Should be 0&lt;p&lt;0.5.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>obsModel</code></td>
<td>

<p>Parameters for the chosen observation model.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The <code>simulateNetwork</code> function can simulate many observation models to create observational data. An overvoew of how to use the various types of observation models present in <code>BoolFilter</code> is given below:
</p>

<ul>
<li>
<p>Bernoulli observation model requires only one parameter, aside from declaring the type, e.g. <br><br><code>obsModel = list(type = 'Bernoulli', q = 0.05)</code>
<br></p>
</li>
<li> <p>Gaussian observation model requires a vector of the observation parameters, which include the mean and standard deviation of Boolean variables in inactivated and activated states. This will be defined as a vector, e.g. <br><br><code>mu0 = 1</code> <br><code>sigma0 = 2</code><br><code>mu1 = 5</code><br><code>sigma1 = 2</code><br><code>obsModel = list(type = 'Gaussian', model = c(mu0, sigma0, mu1, sigma1))</code>
<br></p>
</li>
<li>
<p>Poisson observation model requires a list of parameters. This list will have 3 entries in addition to the type definition, for a total of 4 entries:
</p>

<ul>
<li>
<p>Sequencing depth <code>s</code>
</p>
</li>
<li>
<p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li>
<p>The differential expression, referred to as <code>delta</code>, which must be input as a vector of the same length as the number of genes in the network.
</p>
</li>
</ul>
<p>In this way, the user can define the exact observation parameter for each individual gene. For a 4-gene network, a potential <code>obsModel</code> parameter for a Poisson distribution could be defined as:<br><br><code>obsModel = list(type = 'Poisson', s = 10.875, mu = 0.01, delta = c(2, 2, 2, 2))</code>
<br></p>
</li>
<li>
<p>Negative-Binomial observation models also require a list of parameters. This list will have 4 entries in addition to the type definition, for a total of 5 entries:
</p>

<ul>
<li>
<p>Sequencing depth <code>s</code>
</p>
</li>
<li>
<p>Baseline espression in inactivated state, referred to as <code>mu</code>
</p>
</li>
<li>
<p>Differential expression, referred to as <code>delta</code>,  which must be input as a vector of the same length as the number of genes in the network.
</p>
</li>
<li>
<p>Inverse Dispersion, referred to as <code>phi</code>, which must also be input as a vector of the same length as the number of genes in the network.
</p>
</li>
</ul>
<p>For a 4-gene network, a potential <code>obsModel</code> parameter for a Negative-Binomial observation model could be defined as:<br><br><code>delta = c(2, 2, 2, 2)</code> <br><code>phi = c(3, 3, 3, 3)</code> <br><code>obsModel = list(type = 'NB', s = 10.875, mu = 0.01, delta, phi)</code>
<br></p>
</li>
</ul>
<h3>Value</h3>

<table>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>
<p>Original Boolean state trajectory, <b>without</b> observation noise</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Y</code></td>
<td>
<p>Observation trajectory</p>
</td>
</tr>
</table>
<h3>Examples</h3>

<pre><code class="language-R">data(p53net_DNAdsb1)

#generate data from poisson observation model
dataPoisson &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = 0.02, 
                          obsModel = list(type = 'Poisson',
                                          s = 10.875, 
                                          mu = 0.01, 
                                          delta = c(2,2,2,2)))

#generate data from Bernoulli observation model
dataBernoulli &lt;- simulateNetwork(p53net_DNAdsb1, n.data = 100, p = 0.02, 
                          obsModel = list(type = 'Bernoulli',
                                          q = 0.05))
</code></pre>


</div>