<div class="container">

<table style="width: 100%;"><tr>
<td>BootstrapQTL</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Bootstrap QTL analysis for accurate effect size estimation</h2>

<h3>Description</h3>

<p>Performs cis-QTL mapping using MatrixEQTL then performs a bootstrap
analysis to obtain unbiased effect size estimates for traits with
significant evidence of genetic regulation correcting for the
"Winner's Curse" arising from lead-SNP selection.
</p>


<h3>Usage</h3>

<pre><code class="language-R">BootstrapQTL(snps, gene, snpspos, genepos, cvrt = SlicedData$new(),
  n_bootstraps = 200, n_cores = 1, eGene_detection_file_name = NULL,
  bootstrap_file_directory = NULL, cisDist = 1e+06,
  local_correction = "bonferroni", global_correction = "fdr",
  correction_type = "shrinkage", errorCovariance = numeric(),
  useModel = modelLINEAR, eigenMT_tests_per_gene = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>snps</code></td>
<td>
<p><code>SlicedData</code> object containing genotype
information used as input into <code>Matrix_eQTL_main</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gene</code></td>
<td>
<p><code>SlicedData</code> object containing gene expression
information used as input into <code>Matrix_eQTL_main</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>snpspos</code></td>
<td>
<p><code>data.frame</code> object with information about SNP locations.
Used in conjunction with <code>'genespos'</code> and <code>'cisDist'</code> to
determine SNPs in <em>cis</em> of each gene. Must have three columns: </p>

<ol>
<li>
<p> 'snpid' describing the name of the SNP and corresponding to rows in
the 'snps' matrix.
</p>
</li>
<li>
<p> 'chr' describing the chromosome for each SNP.
</p>
</li>
<li>
<p> 'pos' describing the position of the SNP on the chromosome.
</p>
</li>
</ol>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genepos</code></td>
<td>
<p><code>data.frame</code> object with information about transcript locations.
Used in conjunction with <code>'snpspos'</code> and <code>'cisDist'</code> to
determine SNPs in <em>cis</em> of each gene. Must have four columns: </p>

<ol>
<li>
<p> 'geneid' describing the name of the gene and corresponding to rows in
the 'gene' matrix.
</p>
</li>
<li>
<p> 'chr' describing the chromosome for each SNP.
</p>
</li>
<li>
<p> 'left' describing the start position of the transcript.
</p>
</li>
<li>
<p> 'right' describing the end position of the transcript.
</p>
</li>
</ol>
<p>Note that <code>Matrix_eQTL_main</code> tests all
variants within <code>cisDist</code> of the start or end of the gene.
If you wish instead to test all variants within <code>cisDist</code> of
the transcription start site, you should specify this location in
both the 'left' and 'right' columns of the <code>genepos</code>
data.frame. Similarly, when analysing a molecular phenotype that
have a single chromosomal position then the 'left' and 'right'
columns should both contain the same position.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cvrt</code></td>
<td>
<p><code>SlicedData</code> object containing covariate
information used as input into <code>Matrix_eQTL_main</code>.
Argument can be ignored in the case of no covariates.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_bootstraps</code></td>
<td>
<p>number of bootstraps to run.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_cores</code></td>
<td>
<p>number of cores to parallise the bootstrap procedure
over.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eGene_detection_file_name</code></td>
<td>
<p><code>character</code>, <code>connection</code> or <code>NULL</code>.
File to save local <em>cis</em> associations to in the eGene detection analysis. Corresponds
to the <code>output_file_name.cis</code> argument in <code>Matrix_eQTL_main</code>.
If a file with this name exists it is overwritten, if <code>NULL</code> output is not saved
to file.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bootstrap_file_directory</code></td>
<td>
<p><code>character</code> or <code>NULL</code>. If not <code>NULL</code>,
files will be saved in this directory storing local <em>cis</em> associations for
the bootstrap eGene detection group (detection_bootstrapnumber.txt) and local
<em>cis</em> associations the bootstrap left-out eGene effect size estimation
group (estimation_bootstrapnumber.txt). Estimation group files will only be saved
where signficant eGenes are also significant in the bootstrap detection group
(see Details). Corresponds to the <code>output_file_name.cis</code> argument in the
respective calls to <code>Matrix_eQTL_main</code>. Files in this
directory will be overwritten if they already exist.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>cisDist</code></td>
<td>
<p><code>numeric</code>. Argument to <code>Matrix_eQTL_main</code>
controlling the maximum distance from a gene to consider tests for
eQTL mapping.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>local_correction</code></td>
<td>
<p>multiple testing correction method to use when
correcting p-values across all SNPs at each gene (see EQTL mapping
section in Details). Can be a method specified in <code>p.adjust.methods</code>,
"qvalue" for the <code>qvalue</code> package, or "eigenMT"
if EigenMT has been used to estimate the number effective independent 
tests (see <code>eigenMT_tests_per_gene</code>).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>global_correction</code></td>
<td>
<p>multiple testing correction method to use when
correcting p-values across all genes after performing local correction
(see EQTL mapping section in Details). Must be a method specified in
<code>p.adjust.methods</code> or "qvalue" for the
<code>qvalue</code> package.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>correction_type</code></td>
<td>
<p><code>character</code>. One of "shrinkage", "out_of_sample"
or "weighted". Determines which Winner's Curse correction method is
used (see Details).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>errorCovariance</code></td>
<td>
<p><code>numeric matrix</code> argument to <code>Matrix_eQTL_main</code>
specifying the error covariance.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>useModel</code></td>
<td>
<p><code>integer</code> argument to <code>Matrix_eQTL_main</code>
specifying the type of model to fit between each SNP and gene. Should be one of
<code>modelLINEAR</code>, <code>modelANOVA</code>, or
<code>modelLINEAR_CROSS</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>eigenMT_tests_per_gene</code></td>
<td>
<p><code>data.frame</code> containing the number of effective
independent tests for each gene estimated by the EigenMT (<a href="https://github.com/joed3/eigenMT">https://github.com/joed3/eigenMT</a>).
Ignore unless <code>'local_correction="eigenMT"'</code>.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Although the package interface and documentation describe the use of
<code>BootstrapQTL</code> for <em>cis</em>-eQTL mapping, the package can be
applied to any QTL study of quantitative traits with chromosomal
positions, for example <em>cis</em>-QTL mapping of epigenetic
modifications. Any matrix of molecular trait data can be provided
to the <code>'gene'</code> argument provided a corresponding <code>'genepos'</code>
'data.frame' detailing the chromosomal positions of each trait is
provided.
</p>


<h4>Cis-eQTL mapping:</h4>

<p>EQTL mapping is performed using the
<code>MatrixEQTL</code> package. A three step
hieararchical multiple testing correction procedure is used to
determine significant eGenes and eSNPs. At the first step, nominal
p-values from <code>MatrixEQTL</code> for all
<em>cis</em>-SNPs are adjusted for each gene separately using the
method specified in the <code>'local_correction'</code> argument
(Bonferroni correction by default). In the second step, the best
adjusted p-value is taken for each gene, and this set of locally
adjusted p-values is corrected for multiple testing across all genes
using the methods pecified in the <code>'global_correction'</code> argument
(FDR correction by default). In the third step, an eSNP significance
threshold on the locally corrected p-values is determined as the
locally corrected p-value corresponding to the globally corrected
p-value threshold of 0.05.
</p>
<p>A gene is considered a significant eGene if its globally corrected
p-value is &lt; 0.05, and a SNP is considered a significant eSNP for
that eGene if its locally corrected p-value &lt; the eSNP significance
threshold.
</p>
<p>The default settings for <code>'local_correction'</code> and
<code>'global_correction'</code> were found to best control eGene false
discovery rate without sacrificing sensitivity (see citation).
</p>



<h4>Winner's Curse correction:</h4>

<p>EQTL effect sizes of significant eSNPs on significant eGenes are
typically overestimated when compared to replication datasets
(see citation). <code>BootstrapEQTL</code> removes this overestimation by
performing a bootstrap procedure after eQTL mapping.
</p>
<p>Three Winner's Curse correction methods are available: the Shrinkage
method, the Out of Sample method, and the Weighted Estimator method.
All three methods work on the same basic principle of performing
repeated sample bootstrapping to partition the dataset into two
groups: an eQTL detection group comprising study samples select via
random sampling with replacement, and an eQTL effect size estimation
group comprising the remaining samples not selected via the random
sampling. The default estimator, <code>'correction_type = "shrinkage"'</code>,
provided the most accurate corrected effect sizes in our simulation
study (see citation).
</p>
<p>The <strong>shrinkage method</strong> ("shrinkage" in
<code>'correction_type'</code>) corrects for the Winner's Curse by
measuring the average difference between the eQTL effect size
in the bootstrap detection group and the bootstrap estimation group,
then subtracting this difference from the naive eQTL effect size
estimate obtained from the eGene detection analysis prior to the
bootstrap procedure.
</p>
<p>The <strong>out of sample method</strong> ("out_of_sample" in
<code>'correction_type'</code>) corrects for the Winner's Curse by taking
the average eQTL effect size across bootstrap estimation groups as
an unbiased effect size estimate.
</p>
<p>The <strong>weighted estimator method</strong> ("weighted" in
<code>'correction_type'</code>) corrects for the Winner's Curse by taking
a weighted average of the nominal estimate of the eQTL effect size
and the average of eQTL effect sizes across the bootstrap estimation
groups: <code class="reqn">0.368 * naive_estimate + 0.632 *
 mean(bootstrap estimation group effect sizes)</code>.
</p>
<p>In all three methods bootstrap effect sizes only contribute to
the Winner's Curse correction if the corresponding eSNP is
significantly associated with the eGene in the bootstrap detection
group (locally corrected bootstrap P-value &lt; eSNP significance
threshold determing in the eQTL mapping step).
</p>
<p>Note that eQTLs may not remain significant in all bootstraps, so the
effective number of bootstraps used to obtain the Winner's Curse
estimate will typically be lower than the number of bootstraps
specified in <code>'n_bootstraps'</code>. The number of bootstraps that
were significant for each eQTL are reported in the
<code>'correction_boots'</code> column of the returned table.
</p>



<h4>Winner's Curse corrected effect sizes</h4>

<p>The user should be aware that ability to correct for Winner's Curse
can vary across significant eQTLs depending on their statistical
power (<em>i.e. minor allele frequency, true effect size, and
study sample size</em>). Users should be skeptical of corrected effect
sizes that are larger than the nominal effect sizes estimated by
<code>MatrixEQTL</code>, which likely reflects low
power for eQTL detection rather than an underestimated effect size.
</p>



<h4>Bootstrap warning messages:</h4>

<p>It is possible for bootstrap analyses to fail due to the reduced
sample sizes of the bootstrap detection and bootstrap estimation
groups. For example, the bootstrap resampling may lead to an
detection or estimation groups in which all individuals are
homozygous for an eSNP or have no variance in their supplied
covariates (<em>e.g.</em> the estimation group may comprise
individuals all of the same sex). In this case the bootstrap will
fail for all eQTLs since <code>MatrixEQTL</code> will
be unable to perform the model fitting.
</p>
<p>Failed bootstraps are reported after the bootstrap procedure in
a series of warning messages indicating the number of bootstrap
failures grouped by the reason for the bootstrap failure.
</p>



<h3>Value</h3>

<p>A <code>data.frame</code> (or <code>data.table</code> if the
user has the library loaded) containing the results for each significant eQTL:
</p>

<dl>
<dt><code>'eGene':</code></dt>
<dd>
<p>The eQTL eGene.</p>
</dd>
<dt><code>'eSNP':</code></dt>
<dd>
<p>The eQTL eSNP.</p>
</dd>
<dt><code>'statistic':</code></dt>
<dd>
<p>The test statistic for the association between the eGene and eSNP.</p>
</dd>
<dt><code>'nominal_beta':</code></dt>
<dd>
<p>The eQTL effect size for the <code>eGene</code>-<code>eSNP</code>
pair estimated by <code>MatrixEQTL</code>.</p>
</dd>
<dt><code>'corrected_beta':</code></dt>
<dd>
<p>The eQTL effect size after adjustment for the <code>winners_curse</code>.</p>
</dd>
<dt><code>'winners_curse':</code></dt>
<dd>
<p>The amount of effect size overestimation determined by the
bootstrap analysis (See Details).</p>
</dd>
<dt><code>'correction_boots':</code></dt>
<dd>
<p>The number of bootstraps that contributed to the estimation of
the <code>winners_curse</code>, <em>i.e.</em> the number of bootstraps in which the <code>eSNP</code>
remained significantly associated with the <code>eGene</code> (see Details).</p>
</dd>
<dt><code>'nominal_pval':</code></dt>
<dd>
<p>The p-value for the <code>eGene</code>-<code>eSNP</code> pair
from the <code>MatrixEQTL</code> analysis.</p>
</dd>
<dt><code>'eSNP_pval':</code></dt>
<dd>
<p>The locally corrected p-value for the <code>eGene</code>-<code>eSNP</code> pair (see Details).</p>
</dd>
<dt><code>'eGene_pval':</code></dt>
<dd>
<p>The globally corrected p-value for the <code>eGene</code> based on its top eSNP (see Details).</p>
</dd>
</dl>
<h3>Examples</h3>

<pre><code class="language-R"># Locations for example data from the MatrixEQTL package
base.dir = find.package('MatrixEQTL');
SNP_file_name = paste(base.dir, "/data/SNP.txt", sep="");
snps_location_file_name = paste(base.dir, "/data/snpsloc.txt", sep="");
expression_file_name = paste(base.dir, "/data/GE.txt", sep="");
gene_location_file_name = paste(base.dir, "/data/geneloc.txt", sep="");
covariates_file_name = paste(base.dir, "/data/Covariates.txt", sep="");

# Load the SNP data
snps = SlicedData$new();
snps$fileDelimiter = "\t";      # the TAB character
snps$fileOmitCharacters = "NA"; # denote missing values;
snps$fileSkipRows = 1;          # one row of column labels
snps$fileSkipColumns = 1;       # one column of row labels
snps$fileSliceSize = 2000;      # read file in slices of 2,000 rows
snps$LoadFile(SNP_file_name);

# Load the data detailing the position of each SNP
snpspos = read.table(snps_location_file_name, header = TRUE, stringsAsFactors = FALSE);

# Load the gene expression data
gene = SlicedData$new();
gene$fileDelimiter = "\t";      # the TAB character
gene$fileOmitCharacters = "NA"; # denote missing values;
gene$fileSkipRows = 1;          # one row of column labels
gene$fileSkipColumns = 1;       # one column of row labels
gene$fileSliceSize = 2000;      # read file in slices of 2,000 rows
gene$LoadFile(expression_file_name);

# Load the data detailing the position of each gene
genepos = read.table(gene_location_file_name, header = TRUE, stringsAsFactors = FALSE);

# Load the covariates data
cvrt = SlicedData$new();
cvrt$fileDelimiter = "\t";      # the TAB character
cvrt$fileOmitCharacters = "NA"; # denote missing values;
cvrt$fileSkipRows = 1;          # one row of column labels
cvrt$fileSkipColumns = 1;       # one column of row labels
if(length(covariates_file_name)&gt;0) {
  cvrt$LoadFile(covariates_file_name);
}

# Run the BootstrapQTL analysis
eQTLs &lt;- BootstrapQTL(snps, gene, snpspos, genepos, cvrt, n_bootstraps=10, n_cores=2)

</code></pre>


</div>