<div class="container">

<table style="width: 100%;"><tr>
<td>Weights.mrmr</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Computation of MRMR weights for gene selection</h2>

<h3>Description</h3>

<p>The MRMR weights associated with each gene in the dataset are computed by using the MRMR technique for informative gene selection.</p>


<h3>Usage</h3>

<pre><code class="language-R">Weights.mrmr(x, y)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>x is a N by p dataframe of gene expression, where, rows as genes and columns as samples (with row names as gene names/ids)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>This function returns a vector of MRMR weights for all genes in the dataset.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>References</h3>

<p>Ding, C and Peng, H (2005). Minimum redundancy feature selection from microarray gene expression data. J. Bioinformatics Comput Biol 3(2):185-205.</p>


<h3>Examples</h3>

<pre><code class="language-R">
data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
Weights.mrmr(x, y)
</code></pre>


</div>