<div class="container">

<table style="width: 100%;"><tr>
<td>mrmr.cutoff</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Informative geneset selection using MRMR weights</h2>

<h3>Description</h3>

<p>The function returns the informative genes/ geneset for the particular trait/condition under investigation using Maximum Relevance and Minimum Redundancy (MRMR) technique.</p>


<h3>Usage</h3>

<pre><code class="language-R">mrmr.cutoff(x, y, n)</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>x is a N by p data frame of gene expression values where rows represent genes and columns represent samples/subject/time point. Each cell entry represents the expression level of a gene in a sample/subject (row names of x as gene names/gene ids).</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>
<p>y is a p by 1 numeric vector with entries 1 and -1 representing sample labels, where 1 and -1 represents the sample label of subjects/ samples for stress and control condition respectively.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>
<p>n is a numeric constant represents the number of informative genes to be selected.</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>An informative geneset is obtained, which is relevanit to the particular trait/condition and the genes within the selected geneset are minimum redundant using MRMR technique.</p>


<h3>Author(s)</h3>

<p>Samarendra Das</p>


<h3>References</h3>

<p>Ding, C and Peng, H (2005). Minimum redundancy feature selection from microarray gene expression data. J. Bioinformatics Comput Biol 3(2):185-205.</p>


<h3>Examples</h3>

<pre><code class="language-R">data(rice_salt)
x=as.data.frame(rice_salt[-1,])
y=as.numeric(rice_salt[1,])
n=20
mrmr.cutoff(x, y, n)
</code></pre>


</div>