<div class="container">

<table style="width: 100%;"><tr>
<td>SNV</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Single nucleotide variant (SNV) data from colon cancer patients</h2>

<h3>Description</h3>

<p>SNV calls from 30 genes across 58 colon cancer patients.
Additional data on the patient samples is found in in the "<code>patient</code>" dataset.
The same patient samples are described in the "<code>microarray</code>" and "<code>CNA</code>" 
datasets.
</p>


<h3>Usage</h3>

<pre><code class="language-R">SNV</code></pre>


<h3>Format</h3>

<p>A data frame with 58 columns and 30 rows. The columns indicate the patient sample,
and the rows indicate the gene. The contents of the data frame are either NA (indicating no
SNV call was made) or one of: 
</p>

<ul>
<li>
<p>1 - nonsynonymous SNV
</p>
</li>
<li>
<p>2 - stopgain SNV
</p>
</li>
<li>
<p>3 - frameshift insertion
</p>
</li>
<li>
<p>4 - frameshift deletion
</p>
</li>
<li>
<p>5 - nonframeshift insertion
</p>
</li>
<li>
<p>6 - nonframeshift deletion
</p>
</li>
<li>
<p>7 - splicing
</p>
</li>
<li>
<p>8 - unknown
</p>
</li>
</ul>
<h3>Author(s)</h3>

<p>Christine P'ng</p>


<h3>Examples</h3>

<pre><code class="language-R">len &lt;- apply(SNV[1:15], 2, function(x){mutation.count &lt;- length(which(x == 1))});

create.barplot(
    # filename = tempfile(pattern = 'Using_SNV_dataset', fileext = '.tiff'),
    formula = len ~ colnames(SNV[1:15]) ,
    data = SNV,
    main = 'SNV dataset',
    xaxis.rot = 45, 
    ylimits = c(0,30),
    yat = seq(0,30,5),
    description = 'Barplot created by BoutrosLab.plotting.general'
    );
</code></pre>


</div>