<div class="container">

<table style="width: 100%;"><tr>
<td>simulate_microbiome_counts</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Simulate microbiome counts</h2>

<h3>Description</h3>

<p>Each bacteria's absolute abundance is simulated from a lognormal distribution.
Then, convert each sample to relative abundance, and simulate sequencing counts
using a multinomial distribution, based on the desired number of reads and the
simulated relative abundances. This also simulates an environmental variable that
is produced by some of the bacteria.
</p>


<h3>Usage</h3>

<pre><code class="language-R">simulate_microbiome_counts(
  df,
  n_contributors = 20,
  coeff_contributor = 0,
  min_ab_contributor = -9,
  sd_Y = 1,
  n_reads = 1e+05,
  var_contributor = 5,
  use_uniform = TRUE,
  n_samples_use = NULL,
  corr_value = NULL,
  return_absolute = FALSE,
  seed = NULL
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>A dataframe of OTU counts that is a model for data simulation. Samples are rows and bacteria are columns.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_contributors</code></td>
<td>
<p>the number of bacteria that are to contribute to your environmental variable.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coeff_contributor</code></td>
<td>
<p>the average of the distribution used to simulate the contribution coefficient.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>min_ab_contributor</code></td>
<td>
<p>The minimum log relative abundance, averaged across samples, to include a bacteria</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sd_Y</code></td>
<td>
<p>the standard deviation of the simulated environmental variable</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_reads</code></td>
<td>
<p>the number of reads to be simulated per sample</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>var_contributor</code></td>
<td>
<p>If you use a uniform distribution, this is the range of the distribution, with a normal distribution it is the variance used to simulate the contribution coefficient.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>use_uniform</code></td>
<td>
<p>use a uniform distribution to simulate the contribution coefficient. Alternative is the normal distribution.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n_samples_use</code></td>
<td>
<p>number of microbiome samples to simulate. If NULL, uses the same number of samples as in your dataframe</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>corr_value</code></td>
<td>
<p>the bacteria-bacteria correlation value you want to include in the simulation</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>return_absolute</code></td>
<td>
<p>returns the abosulte abundance values instead of the simulated microbiome counts</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seed</code></td>
<td>
<p>random seed for reproducibility</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a list containing 1) the simulated count data 2) the simulated environmental variable and 3) the simulated contribution coefficients
</p>


</div>