<div class="container">

<table style="width: 100%;"><tr>
<td>WGR2 (EM)</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Expectation-Maximization WGR
</h2>

<h3>Description</h3>

<p>Univariate models to find breeding values through regression fitted via expectation-maximization implemented in C++.
</p>


<h3>Usage</h3>

<pre><code class="language-R">emRR(y, gen, df = 10, R2 = 0.5)
emBA(y, gen, df = 10, R2 = 0.5)
emBB(y, gen, df = 10, R2 = 0.5, Pi = 0.75)
emBC(y, gen, df = 10, R2 = 0.5, Pi = 0.75)
emBCpi(y, gen, df = 10, R2 = 0.5, Pi = 0.75)
emBL(y, gen, R2 = 0.5, alpha = 0.02)
emEN(y, gen, R2 = 0.5, alpha = 0.02)
emDE(y, gen, R2 = 0.5)
emML(y, gen, D = NULL)
lasso(y, gen)

emCV(y, gen, k = 5, n = 5, Pi = 0.75, alpha = 0.02,
     df = 10, R2 = 0.5, avg=TRUE, llo=NULL, tbv=NULL, ReturnGebv = FALSE)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>Numeric vector of response variable (<code class="reqn">n</code>). <code>NA</code> is not allowed.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>gen</code></td>
<td>

<p>Numeric matrix containing the genotypic data. A matrix with <code class="reqn">n</code>
rows of observations and <code class="reqn">m</code> columns of molecular markers.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>

<p>Hyperprior degrees of freedom of variance components.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>R2</code></td>
<td>

<p>Expected R2, used to calculate the prior shape (de los Campos et al. 2013).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Pi</code></td>
<td>

<p>Value between 0 and 1. Expected probability pi of having null effect (or 1-Pi if Pi&gt;0.5).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>

<p>Value between 0 and 1. Intensity of L1 variable selection.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>D</code></td>
<td>

<p>NULL or numeric vector with length p. Vector of weights for markers.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>k</code></td>
<td>

<p>Integer. Folding of a k-fold cross-validation.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>n</code></td>
<td>

<p>Integer. Number of cross-validation to perform.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>avg</code></td>
<td>

<p>Logical. Return average across CV, or correlations within CV.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>llo</code></td>
<td>

<p>NULL or a vector (numeric or factor) with the same length as y. If provided, the cross-validations are performed as Leave a Level Out (LLO). This argument allows the user to predefine the splits. This argument overrides <code>k</code> and <code>n</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>tbv</code></td>
<td>

<p>NULL or numeric vector of 'true breeding values' (<code class="reqn">n</code>) to use to compare cross-validations to. If NULL, the cross-validations will have the phenotypes as prediction target.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ReturnGebv</code></td>
<td>

<p>Logical. If TRUE, it returns a list with the average marker values and fitted values across all cross-validations, in addition to the regular output.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The model for the whole-genome regression is as follows:
</p>
<p style="text-align: center;"><code class="reqn">y = mu + Xb + e</code>
</p>

<p>where <code class="reqn">y</code> is the response variable, <code class="reqn">mu</code> is the intercept, <code class="reqn">X</code> is the genotypic matrix, <code class="reqn">b</code> is the effect of an allele substitution (or regression coefficient) and <code class="reqn">e</code> is the residual term. A k-fold cross-validation for model evaluation is provided by <code class="reqn">emCV</code>.
</p>


<h3>Value</h3>

<p>The EM functions returns a list with the intercept (<code class="reqn">mu</code>), the regression coefficient (<code class="reqn">b</code>), the fitted value (<code class="reqn">hat</code>), and the estimated intraclass-correlation (<code class="reqn">h2</code>). 
</p>
<p>The function emCV returns the predictive ability of each model, that is, the correlation between the predicted and observed values from <code class="reqn">k</code>-fold cross-validations repeated <code class="reqn">n</code> times.
</p>


<h3>Author(s)</h3>

<p>Alencar Xavier
</p>


<h3>Examples</h3>

<pre><code class="language-R">     ## Not run: 

data(tpod)
emCV(y,gen,3,3)
          
 
## End(Not run)
</code></pre>


</div>