<div class="container">

<table style="width: 100%;"><tr>
<td>XTRA 2</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Additional tools
</h2>

<h3>Description</h3>

<p>Complementary functions that may help with handling parameters and routine operations.
</p>


<h3>Details</h3>

<p><code>emGWA(y,gen)</code> # Simple MLM for association analysis
</p>
<p><code>markov(gen,chr=NULL)</code> # Markovian imputation of genotypes coded as 012
</p>
<p><code>IMP(X)</code> # Imputes genotypes with SNP expectation (column average)
</p>
<p><code>CNT(X)</code> # Recodes SNPs by centralizing columns in a matrix
</p>
<p><code>GAU(X)</code> # Creates Gaussian kernel as exp(-Dist2/mean(Dist2))
</p>
<p><code>GRM(X,Code012=FALSE)</code> # Creates additive kinship matrix VanRaden 2008
</p>
<p><code>SPC(y,blk,row,col,rN=3,cN=1)</code> # Spatial covariate
</p>
<p><code>SPM(blk,row,col,rN=3,cN=1)</code> # Spatial design matrix
</p>
<p><code>SibZ(id,p1,p2)</code> # Pedigree design matrix compatible to regression methods
</p>
<p><code>Hmat(ped,gen=NULL)</code> # Kinship combining pedigree and genomics
</p>
<p><code>EigenGRM(X, centralizeZ = TRUE, cores = 1)</code> # GRM using Eigen library
</p>
<p><code>EigenARC(X, centralizeZ = TRUE, cores = 1)</code> # ArcCosine kernel
</p>
<p><code>EigenGAU(X, phi = 1.0, cores = 1)</code> # Gaussian kernel using Eigen library
</p>
<p><code>EigenCNT(X, cores = 1)</code> # Center SNPs without missing Eigen library
</p>
<p><code>EigenEVD(A, cores = 1)</code> # Eigendecomposition from Eigen library
</p>
<p><code>EigenBDCSVD(X, cores = 1)</code> # BDC single value composition from Eigen
</p>
<p><code>EigenJacobiSVD(X, cores = 1)</code> # Jacobi single value composition from Eigen
</p>
<p><code>EigenAcc(X1, X2, h2 = 0.5, cores = 1)</code> # Deterministic accuracy X1 -&gt; X2 via V
</p>
<p><code>AccByC(X1, X2, h2 = 0.5, cores = 1)</code> # Deterministic accuracy X1 -&gt; X2 via C
</p>
<p><code>EigenArcZ(Zfndr, Zsamp, cores = 1)</code> # Reduced rank ArcCos kernel PCs with founder rotation
</p>
<p><code>EigenGauZ(Zfndr, Zsamp, phi=1, cores = 1)</code> # Reduced rank Gaussian kernel PCs with founder rotation 
</p>
<p><code>K2X(K, MinEV = 1e-8, cores = 1)</code> # Reparametrize kernel to PCs to run regression models 
</p>
<p><code>SimY(Z,k=5,h2=0.5,GC=0.5,seed=123,unbalanced=FALSE,PercMiss=0,BlkMiss=FALSE)</code> # Simulate phenotypes 
</p>
<p><code>SimZ(ind=500,snp=500,chr=2,F2=TRUE,rec=0.01)</code> # Simulate genome
</p>
<p><code>SimGC(k=50,...)</code> # Simulate genetic correlation matrix
</p>
<p><code>MvSimY(Ufndr,Zfndr,Zsamp,GxY,GxL,H2plot,nLoc=20,Seed=123)</code> # Simulate phenotypes given founders
</p>


<h3>Author(s)</h3>

<p>Alencar Xavier
</p>


</div>