<div class="container">

<table style="width: 100%;"><tr>
<td>XTRA 1</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Mixed model solver
</h2>

<h3>Description</h3>

<p>Function to solve univariate mixed models with or without the usage of omic information. This function allows single-step modeling of replicated observations with marker information available through the usage of a linkage function to connect to a whole-genome regression method. Genomic estimated values can be optionally deregressed (no shrinkage) while fitting the model.
</p>


<h3>Usage</h3>

<pre><code class="language-R">mixed(y,random=NULL,fixed=NULL,data=NULL,X=list(),
      alg=emML,maxit=10,Deregress=FALSE,...)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>y</code></td>
<td>

<p>Response variable from the data frame containg the dataset.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>random</code></td>
<td>

<p>Formula. Right-hand side formula of random effects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fixed</code></td>
<td>

<p>Formula. Right-hand side formula of fixed effects.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>

<p>Data frame containing the response variable, random and fixed terms.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>X</code></td>
<td>

<p>List of omic incidence matrix. Row names of these matrices connect the omic information to the levels of the indicated random terms (eg. <code>X=list("ID"=gen)</code>).
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alg</code></td>
<td>

<p>Function. Whole-genome regression algorithm utilized to solve link functions. These include MCMC (<code>wgr</code>, <code>BayesB</code>, etc) and EM (<code>emEN</code>, <code>emDE</code>, etc) algorithms. By default, it runs maximum likelihood <code>emML</code>.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>maxit</code></td>
<td>

<p>Integer. Maximum number of iterations.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>Deregress</code></td>
<td>

<p>Logical. Deregress (unshrink) coefficients while fitting the model?
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>

<p>Additional arguments to be passed to the whole-genome regression algorithms especified on <code>alg</code>.
</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The model for the whole-genome regression is as follows:
</p>
<p style="text-align: center;"><code class="reqn">y = Xb + Zu + Wa + e</code>
</p>

<p>where <code class="reqn">y</code> is the response variable, <code class="reqn">Xb</code> corresponds to the fixed effect term, <code class="reqn">Zu</code> corresponds to one or more random effect terms, <code class="reqn">W</code> is the incidence matrix of terms with omic information and <code class="reqn">a</code> is omic values by <code class="reqn">a=Mg</code>, where <code class="reqn">M</code> is the genotypic matrix and <code class="reqn">g</code> are marker effects. Here, <code class="reqn">e</code> is the residual term. An example is provided using the data from the NAM package with: <code>demo(mixedmodel)</code>.
</p>
<p>Alterinative (and updated) implementations have similar syntax:
</p>
<p>01) <code>mm(y,random=NULL,fixed=NULL,data=NULL,
              M=NULL,bin=FALSE,AM=NULL,it=10,verb=TRUE,
              FLM=TRUE,wgtM=TRUE,cntM=TRUE,nPc=3)</code>
</p>
<p>02) <code>mtmixed = function(resp, random=NULL, fixed=NULL,
              data, X=list(), maxit=10, init=10, regVC=FALSE)</code>
</p>


<h3>Value</h3>

<p>The function wgr returns a list with Fitness values (<code>Fitness</code>) containing observation <code>obs</code>, fitted values <code>hat</code>, residuals <code>res</code>, and fitted values by model term <code>fits</code>; Estimated variance components (<code>VarComp</code>) containing the variance components per se (<code>VarComponents</code>) and variance explained by each model term (<code>VarExplained</code>), regression coefficients by model term (<code>Coefficients</code>), and the effects of structured terms (<code>Structure</code>) containing the marker effects of each model term where markers were provided.
</p>


<h3>Author(s)</h3>

<p>Alencar Xavier
</p>


<h3>References</h3>

<p>Xavier, A. (2019). Efficient Estimation of Marker Effects in Plant Breeding. G3: Genes, Genomes, Genetics, DOI: 10.1534/g3.119.400728      
</p>


<h3>Examples</h3>

<pre><code class="language-R"> ## Not run: 
 demo(mixedmodel)
 
## End(Not run)
</code></pre>


</div>